---
title: "1. Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{1. Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```


```{r setup}
library(CNAqc)
```

We discuss the input format and references used in `CNAqc`.

# Input format

You need to provide data for _somatic mutations_, _copy number segments_ and _tumour purity_. The package comes with a template dataset available.

```{r}
# Load template data
data('example_dataset_CNAqc', package = 'CNAqc')
```

These fields are required for **mutation** calls:

* the mutation location, as `chr`, `from`, `to`. 
* the mutation reference and alternative alleles `ref` and `alt`;
* the total number of reads covering the mutated base(s), `DP` (depth);
* the total number of reads covering only the mutant allele(s), `NV` (number of reads with variant);
* the Variant Allele Frequency, `VAF`, defined as `NV/DP`.

Chromosome names and alleles should be in character format; chromosomes must be in the format `chr1`, `chr2`, etc..

```{r}
# Example input SNVs
print(
  example_dataset_CNAqc$snvs %>% 
        select(
          chr, from, to, # Genomic coordinates
          ref, alt,      # Alleles (reference and alternative)
          DP, NV, VAF    # Read counts (depth, number of variant reads, tumour VAF)
          )
  )
```

Optionally, you can annotate also _driver mutations_:

* whether the mutation is a driver, using a boolean flag `is_driver`;
* a `driver_label` reportin the label to show in the plots with drivers.

```{r}
print(
  example_dataset_CNAqc$snvs %>% 
        select(
          chr, from, to, ref, alt, is_driver, driver_label, everything()
        )
  )
```

These fields are required for **copy number segments**:

* the segment location, as `chr`, `from` and `to`;
* the segment absolute number of copies of the major and minor alleles, as `Major` and `minor`;

In `CNAqc` format, a diploid balanced segment has `Major = minor = 1` and will be shortly denoted as *karyotype* `1:1`. Similarly, a tetraploid segment has `Major = minor = 2` and *karyotype* `2:2`. A fixed set of colours is used to identify the main karyotypes supported by `CNAqc`: `"1:0"`, `"2:0"`, `"1:1"`, `"2:1"` and `"2:2"`. 

Optionally, you can annotate also annotate the Cancer Cell Fraction (CCF) `CCF` for each segment:

* segments with `CCF = 1` are _clonal_, otherwise _subclonal_;
* by default, if missing, all segments are assumed to be clonal and assigned `CCF = 1`. Some computations subset only clonal copy number segments (see the manual of each function).


```{r}
# Example input CNA
print(
  example_dataset_CNAqc$cna %>% 
        select(
          chr, from, to, # Genomic coordinates
          Major, minor  # Number of copies of major/ and minor allele (B-allele)
        )
  )
```
Tumour **purity**, defined as the percentage of reads coming from tumour cells), which must be a value in $[0, 1]$.

```{r}
# Example purity
print(example_dataset_CNAqc$purity)
```

**Note** In the current implementation no use of subclonal copy number segments is done. Most function compute statics only with clonal copy number segments.

## Initialisation of a new dataset

To use `CNAqc`, you need to initialize a `cnaqc` S3 object with the initialisation function `init`.

This function will check input formats, and will map mutations to CNA segments. This call does not subset the data, unless there are errors with respect to the required formats. It is required to explicit the reference genome for the
assembly (see below).

```{r}
# Use SNVs, CNAs and tumour purity (hg19 reference, see below)
x = init(
  example_dataset_CNAqc$snvs, 
  example_dataset_CNAqc$cna,
  example_dataset_CNAqc$purity,
  ref = 'hg19'
  )
```

The summary of `x` can be `print` to see the number of mutations per karyotype, sorted by size.
```{r}
print(x)
```


## Subsetting data

You can _subset randomly_ the data; if drivers are annotated, they can be forced to say in.

```{r, fig.width=4, fig.height=4}
y_5000 = subsample(x, N = 5000, keep_drivers = TRUE)

# 5000 + the ranomd entries that we sampled before
print(y_5000)
```

You can also subset data by _karyotype of the segments_, and by _total copy number_ of the segment. Both subset do not keep drivers that map off the selected segments.
```{r, fig.width=4, fig.height=4}
# Triploid and copy-neutral LOH segments 
y_tripl_cnloh = subset_by_segment_karyotype(x, karyotypes = c('2:1', '2:0'))

print(y_tripl_cnloh)

# Two and four copies
y_2_4 = subset_by_segment_totalcn(x, totalcn = c(2, 4))

print(y_2_4)
```


## Reference genome coordinates

`CNAqc` contains two coordinates for two reference genomes, `hg19` and `GRCh38` (default). For each reference genome the package stores the location of the chromosomes, and their centromers; you can access these data.

```{r, fig.width=9.5, fig.height=3}
CNAqc:::get_reference("hg19")

CNAqc:::get_reference("GRCh38")
```

The reference genomes can be specified as a parameter once you create a `CNAqc` object. Coordinates are used to translate relative to absolute genome coordinates of somatic mutations and CNA segments; this is done only 
when plotting the data. Note that mapping of mutations to CNA segments is independent of the reference genome, and it will work as far as both mutation and CNA segments are mapped to the same reference.

You can use a hidden function to plot a reference, and make it explicit with a title.

```{r, fig.width=9.5, fig.height=3}
print(CNAqc:::blank_genome(ref = 'hg19') + labs(title = "HG19 genome reference"))
```



