---
title: "4. Computation of Cancer Cell Fractions"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{4. Computation of Cancer Cell Fractions}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```


```{r setup}
library(CNAqc)
```

`CNAqc` provides a function to compute mutation's Cancer Cell Fractions (CCFs): `compute_CCF`. We suggest to use these to compare CCF estimates obtained with other tools, for instance, or to compute de novo CCF values.

The process of computing CCF values requires the computation of the *"mutation multiplicity"*, namely the number of copies of a mutation in a certain copy number segment. This is a difficult task that harmonize the data accounting for tumour purity, and CNAs. Errors in this task propagate errors in CCF estimates, and downstream analyses.

For this reason, `CNAqc`:

* limits the CCF computations to karyotypes that are usually "easier". These are `'2:1'`, `'2:0'` and `'2:2'` (Major:minor, e.g., `'2:1`' could be written as `AAB`), where a mutation is present in one or two copies if we assume that the aneuploidy state is directly achieved from a diploid genome;
* limits the CCF computations to clonal copy number segments.

The computation can be done in two different ways:

* from the entropy of a simplified model for the data distribution that seek to ensure low-uncertainty CCF estimates
* or just using a dynamic cut computed from the data.

By design, the latter is more rough but can work also in conditions of low-quality data, where the former method would have no power to detect strongly supported CCF estimates

# Mutation multiplicity using entropy

To get CCF values, we use the [relation between VAF, mutation multiplicity and tumour purity](https://caravagn.github.io/CNAqc/articles/CNAqc.html) to determine the value of $m$ from the observed $v$, the karyotype and tumour purity. 

## 2-class mixture

In this setup, `CNAqc` uses a statistical model with two Binomial distributions for the mutations happening before, and after aneuploudy. The density of these Binomial distributions are:

* peaked at $v_1$ for $m = 1$ (after aneuploidy) and $v_2$ for $m = 2$  (before aneuploidy). 
* computed on the domain $[0, 1]$, after translating the read counts into VAFs. The number of trials of the Binomial process is set to the median coverage $n$ of the observed mutations that map to the segments under investigation (e.g., the triploid `'2:1'` segments);

The assumptions in `CNAqc` are that 1) overdispersion of coverage is small to justify a Binomial (instead of a Beta-Binomial), and that 2) trials are well-represented by the median of the observed coverage.

`CNAqc` computes two Binomial densities $\text{Bin}_1$ and $\text{Bin}_2$ over the set of values $[0; n]$ with $n$ number of trials and success probability $v_i$, i.e.
\[
\text{Bin}_i = p(s \mid n; v_i)
\]
as the probability of finding $s$ reads with the mutant allele at out of $n$ (median depth), assuming the mutation expected frequency is $v_i$ (notice that $v_i$ includes the effect of tumour purity).

To finalize the densities of this mixed model `CNAqc` determines the absolute proportions of this mixture, which are used to scale the two theoretical Binomial densities. This process can be done in different ways; `CNAqc` uses a simple heuristic that counts how many mutations fall below the area of each Binomial density. The heuristics computes the 1% and 99% quantiles of those distributions, determing two ranges $I_1 = [v_1^1; v_1^{99}]$ and $I_2 = [v_2^1; v_2^{99}]$ so that:

* $n_1$ are the number of mutations with VAF in $I_1$; 
* $n_2$ are the number of mutations with VAF in $I_2$; 

Notice that these two intervals might overlap (i.e., it could be that $v_1^{99} > v_2^{1}$). Also, note that mutations with VAF smaller than the smallest interval point ($I_1 < v_1^1$), which are putative _neutral tail_ mutations, are not counted in $n_1$ (the mutations accrued after aneuploidy). This is correct, because they are subclonal; see more on the _analysis of evolutionary data including neutral evolution_ with the [mobster package]()). 

The counts  are normalized to create a complete 2-components Binomial density mixture
\[
M = \mu_1 * \text{Bin}_1 + (1 - \mu_1) * \text{Bin}_2
\]
where $\mu_1 = n_1/(n_1+n_2)$ and $\mu_2 = 1 - \mu_1$ are the normalized  proportions. 


## Entropy-based assignments

Assigning mutation multiplicities is particularly difficult at the crossing of the two Binomial densities. Mutations "in between" the densities $\text{Bin}_1$ and $\text{Bin}_1$ cpuld have  multiplicity $m=1$ or $m=2$, and might be misassigned. This error  determines "fake bumps" in the CCF distribution, often well above value 1; these type of errors are tricky to detect as they might look like both miscalled copy-number segments or false clonal architectures. 

`CNAqc` can use a new entropy-based heuristic to assess wether it is possible to confidently assigne $m$ to each VAF value; the heuristic _is not parametrised_.

Mutations with VAF below $v_1$ are assumed to be present in one copy ($m=1$),   those above $v_2$ in two copies ($m=2$). Note that this argument implicitely assumes that we are working with clonal copy number calls, a simplyfying assumption.

`CNAqc` uses the entropy $H(x)$ for VAF value $x$ as defined from the mixture's latent variables to determine mutation multiplicities
\[
H(x) = - p_x \log(p_x)
\]
where $p_x$ is the value of the mixture density at $x$.

Peaks and flexes in the entropy profile allow to identify those subranges of values where one of the two densities is dominated by the other, and therefore the value of $m$ can be estimated confidently. This consists in finding two peaks $\rho_1$ and $\rho_2$ and setting $m=1$ for $x < \rho_1$, $m=2$ for $x > \rho_2$, and `NA` for $\rho_1 \leq x \leq \rho_2$. Values for $\rho_i$ are found using the [peakPick: Peak Picking Methods Inspired by Biological Data](https://cran.rstudio.com/web/packages/peakPick/index.html) package.

A kayotype analysed in this way is considered `PASS` is at most `p%` (default `10%`) mutations are not assignable (`CCF = NA`). 


# Mutation multiplicity using the raw data

To get CCF values in this case we use a more rough method. With a setup similar to the above, when we have computed the
2-class Binomial mixture, we use means of the Binomial distributions to determine a hard split of the data.

We define $p_1$ and $p_2$ the two means. The midpoint would be centred at $\hat{p} = p1 + (p_2-p_1)/2$, we shift it proportionally to size ($n_i$) of the mixture components 
\[
\hat{p} = p_1 + (p_2 - p_1) * \mu_1
\]
where $\mu_1 = n_1/(n_1+n_2)$ is the normalized  proportions. 

We set $m=1$ for $x \leq \hat{p}$, and $m=2$ otherwise. Since there are no `NA` assignments, the computation is always scored `PASS` for quality check.

# Example entropy computation

We work with the template dataset. 

```{r echo=FALSE, results=TRUE, message=FALSE}
# Dataset available with the package 
data('example_dataset_CNAqc', package = 'CNAqc')
x = CNAqc::init(
  example_dataset_CNAqc$snvs, 
  example_dataset_CNAqc$cna,
  example_dataset_CNAqc$purity,
  ref = 'hg19')

print(x)
```


We run function `compute_CCF` to obtain the full analysis of CCF values. Without parameters (or with `method = 'ENTROPY'`), we use the entropy method and all supported karyotypes (default). 
```{r,fig.width=9, fig.height=5, warning=F, message=T}
x = compute_CCF(x)

# Print new object, it informs us of the available computation
print(x)
```

We can visually inspect the CCF estimates for the computed karyotypes. We can do that accessing the field `x$CCF_estimates` which contains the data and a plot reporting the result of the analysis, or using the getter functions `CCF` and two plotting functions

The tibble of the data reports the computed values  in column `"CCF"`, and in column `"mutation multiplicity"`. 

```{r,fig.width=11, fig.height=7}
# Tibble
CCF(x) %>% select(VAF, mutation_multiplicity, CCF)
```

The CCF analysis plot is a `ggpubr`-assembled figure, woith one line per karyotype.
```{r,fig.width=11, fig.height=10,warning=F}
plot_CCF(x)
```

The plot shows the quantities described above; from left to right we have

* the CCF histogram, coloured by mutation multiplicity $m$;
* the histogram of the input VAF, coloured by mutation multiplicity $m$. In this plot the range between the red dashed lines represent the areas where it is not possible to compute reliably CCF values;
* the entropy profile $H(x)$. Here notice that the points we seek to identify are those where $H(x)$ peaks higher (indeed matched). The width of the entropy peak that determines unassignable mutations is proportional to the variance of the counts, and therefore the sequencing coverage;
* the porportion of mutations assigned for each value of $m$, and those unassigned.

The rightmost CCF histogram plot is also sourrounded by a green or red square to reflect QC status of `PASS` (green) or `FAIL` (red).

The CCF histogram can be computed with `plot_data_histogram`, specifying `CCF` as parameter.
```{r,fig.width=4, fig.height=4,warning=F}
plot_data_histogram(x, which = 'CCF')
```

## Comparison with the rough computation

We work with the same data as above, `x`.

```{r,fig.width=9, fig.height=5, warning=F, message=T}
x_r = compute_CCF(x, method = 'ROUGH')

print(x_r)
```

The data can be accessed in the usual way; the layout of the CCF plots are conceptually the same, but use different colors. The cutoff is annotated in the panel with the mutation multiplicity $m$ assigned to each observed VAF value.
```{r,fig.width=11, fig.height=10,warning=F}
plot_CCF(x_r)
```

Notice that for karyotype `2:0` the difference among the mutation multiplicity histograms (second panel) obtained from the two methods is minimal. The difference is also small for triploid mutations (`'2:1'`). The results for tetraploid mutations is instead quite different, and that's because there are not many mutations happening after genome doubling (leftmost peak, $m=1$), and therefore the weighted split of the rough method tends to assign most mutations to the rightmost peak ($m=2$).

A visual plot is obtained using the internal function `CNAqc:::plot_mutation_multiplicity_rough`.
```{r,fig.width=11, fig.height=5,warning=F}
ggpubr::ggarrange(
  CNAqc:::plot_mutation_multiplicity_entropy(x, '2:2'),
  CNAqc:::plot_mutation_multiplicity_rough(x_r, '2:2'),
  nrow = 2
)
```


We can compare the two computed CCF distributions, and observe that the difference in the estimation from karyotype `'2:2'` creates a small bump at about `CCF=0.5`.
```{r,fig.width=8, fig.height=4,warning=F}
ggpubr::ggarrange(
  plot_data_histogram(x, which = 'CCF'),
  plot_data_histogram(x_r, which = 'CCF'),
  nrow = 1
)
```

## Decreasing sequencing coverage

Because these are high-quality data, the results between methods are quite similar. This is however not always true.

The uncertainty from the entropy method stems from the variance of the mixture components. Because these are Binomial distributions, the variance depends on the number of trials, so the sequencing coverage.


If we take the data of `x` and just scale by a constant the read counts, we preserve the same VAF distribution but increase the variance because we decrease coverage. Let's see this in practice retaining `70%` of the actual read counts.

```{r,fig.width=11, fig.height=10, warning=F, message=T}
x_s = init(
  x$snvs %>% mutate(DP = DP * .6, NV = NV * .6),
  x$cna,
  x$purity)

# Recompute CCF values
y_e = compute_CCF(x_s, method = 'ENTROPY', karyotypes = '2:2')
y_r = compute_CCF(x_s, method = 'ROUGH', karyotypes = '2:2')

CCF(y_e) %>% filter(!is.na(CCF))
CCF(y_r) %>% filter(!is.na(CCF))
```
Observe that there are about 400 less mutations with CCF values when we run the entropy method.

```{r,fig.width=11, fig.height=5,warning=F}
ggpubr::ggarrange(
  CNAqc:::plot_mutation_multiplicity_entropy(y_e, '2:2'),
  CNAqc:::plot_mutation_multiplicity_rough(y_r, '2:2'),
  nrow = 2
)
```

```{r,fig.width=8, fig.height=4,warning=F,message=FALSE}
ggpubr::ggarrange(
  plot_data_histogram(y_e, which = 'CCF', karyotypes = '2:2'),
  plot_data_histogram(y_r, which = 'CCF', karyotypes = '2:2'),
  nrow = 1
)
```
