---
title: "CNAqc"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CNAqc}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled=F)
```

### Requirements

To analyse a tumour sample with `CNAqc`, you need: 

* mutation calls (SNVs, indels, ...); 
* absolute CNA segments (clonal/ suclonal copy number calls);
* a tumor purity estimate.

The package uses  chromosome coordinates to map mutations to CNAs, and elaborate a number of analysis on the data.
These can be used to check the quality (QC) and consistency among the calls, or other other metrics often used to
study cancer evolution. Besides genomic coordinates, `CNAqc` requires standard information on the
type of DNA modification; _VEP, ANNOVER or other types of annotations are not required_.   

**Available vignettes:** 

* [1. Introduction (data formats, example dataset)](https://caravagn.github.io/CNAqc/articles/Introduction.html). 
* [2. Plots (data formats, example dataset)](https://caravagn.github.io/CNAqc/articles/Plots.html). 
* [3. QC purity and ploidy via peak detection](https://caravagn.github.io/CNAqc/articles/peaks_detection.html). 
* [4. Cancer Cell Fractions computation](https://caravagn.github.io/CNAqc/articles/ccf_computation.html). 
* [5. Segments smoothing](https://caravagn.github.io/CNAqc/articles/smoothing.html). 
* [6. Genome fragmentation analysis](https://caravagn.github.io/CNAqc/articles/fragmentation.html). 


### Technical details

The following concepts are used to develop `CNAqc` analyses.

The **expected Variant Allele Frequency (VAF)** for mutations present in $m$ copies of the tumour genome, when the tumour sample purity is $\pi$ and segment ploidy is $p$ (i.e., total number of copies of the minor and major allele at the mutation locus) are given by
\[
v = \dfrac{m \pi}{
2 (1 - \pi) + \pi p 
} \, .
\]

This formula gives as an expectation for the allelic frequency of somatic mutations that map to a certain karyotype (e.g., a triploid `2:1` segment, or AAB),

Given VAF, tumour purity and CNA data, **Cancer Cell Fractions (CCF)** values can be computed as

\[
\text{CCF} = 
\dfrac{
v[
(p - 2)\pi + 2
]
}
{
m \pi
}
\]
where $p$ is total segment ploidy (e.g., 3 for a `2:1` state).

