---
title: "Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```


```{r setup}
library(CNAqc)
```

To set-up your data and work with the `CNAqc` package, you need to have:

* mutation calls;
* CNA segments;
* tumor purity estimates;

The workflow of the package is simple:

1. load input data into the package;
2. use package functions to inspiect the data, and run QC analyses;
3. visualize and export QC results.

QC results can be used to determine if the estimated purity and CNA fits to the
observed mutation data are good. If they are off, QC results can be
used to adjust the actual set of calls (e.g., adjusting the purity by a certain
factor).

### Input format

`CNAqc` uses `hg19` chromosome coordinates to map mutations and CNAs. The package
comes with a template dataset available.

```{r}
# Load template data
data('example_dataset_CNAqc', package = 'CNAqc')
```


Input mutations and CNA must be a `data.frame` (or a `tibble`). Both of them are required
to have certain set of pretty standard columns; any extra column will be retained with the data. 

```{r}
# Example input SNVs
print(
  example_dataset_CNAqc$snvs %>% 
        select(
          chr, from, to, # Genomic coordinates
          ref, alt,      # Alleles (reference and alternative)
          DP, NV, VAF    # Read counts (depth, number of variant reads, tumour VAF)
          )
  )
```

Chromosome names and alleles should be in character format; chromosomes 
must be in the format `chr1`, `chr2`, etc.. `CNAqc` will use all available mutation types but we suggest to carry out QC with high-quality SNV data. 

CNA calls must be in *absolute* format, reporting the allelic imbalance of each segment.

```{r}
# Example input CNA
print(
  example_dataset_CNAqc$cna %>% 
        select(
          chr, from, to, # Genomic coordinates
          Major, minor  # Number of copies of major/ and minor allele (B-allele)
        )
  )
```

In this format, a diploid segment will have `Major = minor = 1` and will be denoted as `1:1`. `CNAqc`
distinguishes between clonal and subclonal CNA calls, if the Cancer Cell Fraction of a segment 
(i.e., its clonality) is reported.

A `CCF` column can represent the segment CCF as [0,1] real-value. If missing, 
`CCF = 1` is set assuming clonal segments; otherwise, segments with `CCF<1` are considered subclonal CNAs.

To analyze the data, a purity value in between `0` and `1` is required to
represent the proportion of actual tumour content, sometimes called 
"cellularity".
```{r}
# Example input purity
print(example_dataset_CNAqc$purity)
```

### Initialisation

To use the package, you need to initialize a `cnaqc` object. 

This can be achieved running the initialisation function `init`, and passing mutation, CNA and purity data. 
This function will perform basic checks on the input format, and will map mutations to CNA segments. This operation
can take some time, as it depends on the number of segments in the CNA calls, and the number of mutations in
the data. This call does not subset the data, unless there are errors with respect to its expected format.


```{r}
# Use SNVs, CNAs and tumour purity 
x = CNAqc::init(
  example_dataset_CNAqc$snvs, 
  example_dataset_CNAqc$cna,
  example_dataset_CNAqc$purity
  )
```

The summary of the object is available via the S3 `print` function, reporting some basic
statistics of the data. In particular, this reports the number of mutations per karyotype sorted by
size, and annotates by default is any analysis has been computed on these calls (default: no).
```{r}
print(x)
```

### Plotting segments, genome-wide mutations and CNA calls

`CNAqc` maps CNA calls to the `hg19` human genome coordinates; the references used are avaible attached to the package
```{r, fig.width=9.5, fig.height=3}
data('chr_coordinates_hg19', package = 'CNAqc')
print(chr_coordinates_hg19)
```

Segments can be visualized via function `plot_segments`. By default, this function plots all the genome,
annotating in the figure caption some summmary statistics about the data, and plots
the major and minor alleles for each segment.

```{r, fig.width=9.5, fig.height=3}
# Default plot
plot_segments(x)
```

This function, as most of the package plotting functions, returns a `ggplot` object, which makes it very easy to perform further manipulations.
```{r, fig.width=9.5, fig.height=3}
# Reduce y-axis
plot_segments(x) + ylim(-1, 6)

# Reduce x,y-axis
plot_segments(x) + xlim(1e6, 30e6) + ylim(-1, 6)
```

Genome-wide plots that follow the layout of `plot_segments` are avaiable to view the number of mutations, 
their VAF and depth. VAF and depth-plotting functions (scatterplots) can subset the input data to speed up
rendering and reduce the size of output files - by default 5000 points are randomly sampled. 
Counts plot, instead, bin the genome locations by one megabase.

```{r, fig.width=8, fig.height=1}
# All genome, all data - counts of mutations per megabase
plot_counts(x)
```

An example effect of downsampling the data is the following
```{r, fig.width=8, fig.height=4}
# Different subsamples
ggpubr::ggarrange(
  plot_vaf(x, N = 1000),
  plot_vaf(x),
  plot_vaf(x, N = 10000),
  ncol = 1
)

# Different subsamples
ggpubr::ggarrange(
  plot_depth(x, N = 1000),
  plot_depth(x),
  plot_depth(x, N = 10000),
  ncol = 1
)
```


A one-summary plot can be easily assembled combining the above functions and functions from `cowplot` or other ggplot-manipulation packages.

```{r, fig.width=9.5, fig.height=5}
# Layout a panel with segments in bottom, and all other genome-wide plots on top.
# Via cowplot we can align plots on the vertical axis, and stretch their relative
# height to obtain a nicer layout.
cowplot::plot_grid(
  plot_counts(x),
  plot_vaf(x, N = 10000),
  plot_depth(x, N = 10000),
  plot_segments(x) + ylim(-1, 6),
  align = 'v', 
  nrow = 4, 
  rel_heights = c(.15, .15, .15, .8))
```


A summary plot of CNA calls is available via function `plot_karyotypes`. This function implements two visualizations:

* `type = 'percentage'`, reporting the proportion of genome covered by each karyotypes (default);
* `type = 'number'`, reporting the counts of segments per karyotype.

The package uses a pre-set color schema to colour diploid segments in green, LOH segments in blue and amplifications in red.

```{r, fig.width=4.5, fig.height=4}
ggpubr::ggarrange(
  plot_karyotypes(x),
  plot_karyotypes(x, type = 'number'),
  common.legend = TRUE,
  legend = 'bottom'
)
```
