<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Annotate variant in a CNAqc object — annotate_variants • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Annotate variant in a CNAqc object — annotate_variants" />
<meta property="og:description" content="The function provides an easy and fast way to annotate SNVs, in particular we annotate the locations of the different mutations, the consequences for
those in coding regions and we compare them to a set of known drivers." />
<meta property="og:image" content="caravagnalab.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Drivers annotation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Annotate variant in a CNAqc object</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/annotate_variants.R'><code>R/annotate_variants.R</code></a></small>
    <div class="hidden name"><code>annotate_variants.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The function provides an easy and fast way to annotate SNVs, in particular we annotate the locations of the different mutations, the consequences for
those in coding regions and we compare them to a set of known drivers.</p>
    </div>

    <pre class="usage"><span class='fu'>annotate_variants</span><span class='op'>(</span>
  <span class='va'>x</span>,
  ref <span class='op'>=</span> <span class='st'>"hg19"</span>,
  driver_list <span class='op'>=</span> <span class='fu'>CNAqc</span><span class='fu'>::</span><span class='va'>intogen_drivers</span>,
  polyphen <span class='op'>=</span> <span class='cn'>FALSE</span>,
  filter_tumor_type <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>a data.frame with coloumns c("chr", "from", "to").</p></td>
    </tr>
    <tr>
      <th>driver_list</th>
      <td><p>a one coloumn data.frame with gene symbols</p></td>
    </tr>
    <tr>
      <th>polyphen</th>
      <td><p>whete to annotate polyphen scores</p></td>
    </tr>
    <tr>
      <th>filter_tumor_type</th>
      <td><p>filter intogen drivers by tumor type</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a data.frame with additiona coloumns `location` with the position of the variant in the genome (coding, intron, threeUTR, ...),
`consequence` with the consequence of coding mutations (synonymous, nonsynonymous, ...), `is_driver` a bool that indicate if the gene is a driver,
`gene_symbol` for the annotated corresponding gene symbol (if the variant is in a gene) and `driver_label` with the driver label written as
gene_refAA-&gt;varAA (is NA in case `is_driver` is FALSE).</p>
<p>The `chr` coloumn int the input should contain the chr prefix before the chromosome number. The annotation process is based on the package
<a href='https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html'>VariantAnnotation</a> check it for more information.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/caravagnalab/CNAqc'>CNAqc</a></span><span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'example_dataset_CNAqc'</span>, package <span class='op'>=</span> <span class='st'>'CNAqc'</span><span class='op'>)</span>

<span class='va'>snvs</span> <span class='op'>&lt;-</span> <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>snvs</span>

<span class='va'>snvs_annotated</span> <span class='op'>&lt;-</span> <span class='fu'>annotate_variants</span><span class='op'>(</span><span class='va'>snvs</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:dplyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine, intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:dplyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    first, rename</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:tidyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘IRanges’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:dplyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    collapse, desc, slice</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:purrr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    reduce</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘AnnotationDbi’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:dplyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    select</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘XVector’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:purrr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    compact</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Biostrings’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    strsplit</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Preparing annotations for a dataframe</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #BBBB00;'>!</span><span> There are existing annotations in your data, they will be cancelled.</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 3 x 13</span><span>
#&gt;   chr        from       to ref   alt   FILTER    DP    NV   VAF ANNOVAR_FUNCTION
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> chr2  179</span><span style='text-decoration: underline;'>431</span><span>633   1.79</span><span style='color: #949494;'>e</span><span>8 C     T     PASS     117    77 0.658 exonic          
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> chr16  67</span><span style='text-decoration: underline;'>646</span><span>006   6.76</span><span style='color: #949494;'>e</span><span>7 C     T     PASS     120    54 0.45  exonic          
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> chr17   7</span><span style='text-decoration: underline;'>577</span><span>106   7.58</span><span style='color: #949494;'>e</span><span>6 G     C     PASS      84    78 0.929 exonic          
#&gt; </span><span style='color: #949494;'># … with 3 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;, driver_label &lt;chr&gt;</span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Locaring variants with </span><span style='color: #BBBB00;'>VariantAnnotation</span><span></span></div><div class='output co'>#&gt; <span class='warning'>Warning: GRanges object contains 364 out-of-bound ranges located on sequences</span>
#&gt; <span class='warning'>  4529, 4631, 4632, 1653, 1654, 1655, 1656, 12192, 12362, 12363, 12364,</span>
#&gt; <span class='warning'>  12365, 12366, 12367, 12517, 12518, 12519, 12520, 13123, 13124, 13974,</span>
#&gt; <span class='warning'>  13975, 13976, 13994, 13995, 17007, 17008, 17009, 17010, 17011, 17012,</span>
#&gt; <span class='warning'>  17014, 17015, 17016, 17793, 18007, 20532, 22527, 22954, 22955, 22956,</span>
#&gt; <span class='warning'>  23347, 27275, 27276, 28075, 28085, 28893, 29046, 29047, 29242, 31362,</span>
#&gt; <span class='warning'>  32048, 32050, 33339, 33714, 33715, 32912, 32913, 40960, 40961, 44219,</span>
#&gt; <span class='warning'>  44220, 44242, 44262, 44263, 44356, 46050, 46054, 46062, 46063, 46064,</span>
#&gt; <span class='warning'>  46065, 46066, 53476, 53477, 53497, 53498, 59180, 57764, 57766, 61092,</span>
#&gt; <span class='warning'>  69815, 69930, 70330, 70331, 70332, 77129, 77130, 76239, 76240, 76241,</span>
#&gt; <span class='warning'>  77987, and 77988. Note that ranges located on a sequence whose length</span>
#&gt; <span class='warning'>  is unknown (NA) or on a circular sequence are not considered</span>
#&gt; <span class='warning'>  out-of-bound (use seqlengths() and isCircular() to get the lengths and</span>
#&gt; <span class='warning'>  circularity flags of the underlying sequences). You can use trim() to</span>
#&gt; <span class='warning'>  trim these ranges. See ?`trim,GenomicRanges-method` for more</span>
#&gt; <span class='warning'>  information.</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned many:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned many:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned many:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned many:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Locaring variants with </span><span style='color: #BBBB00;'>VariantAnnotation</span><span> ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sourcing organism</span></div><div class='output co'>#&gt; <span class='message'>using temporary cache /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpPdUtgb/BiocFileCache</span></div><div class='output co'>#&gt; <span class='message'>creating 'src_organism' database...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Ignoring remaining part of query: CREATE INDEX IF NOT EXISTS entrez_accession ON id_accession (entrez);</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Ignoring remaining part of query: CREATE INDEX IF NOT EXISTS entrez_protein ON id_protein (entrez);</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Sourcing organism ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Traslating Entrez ids</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Traslating Entrez ids ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Transforming data</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Transforming data ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Predicting coding</span></div><div class='output co'>#&gt; <span class='warning'>Warning: GRanges object contains 315 out-of-bound ranges located on sequences</span>
#&gt; <span class='warning'>  4529, 4631, 4632, 1653, 1654, 1655, 1656, 12192, 12362, 12363, 12364,</span>
#&gt; <span class='warning'>  12365, 12366, 12367, 12517, 12518, 12519, 12520, 13123, 13124, 13974,</span>
#&gt; <span class='warning'>  13975, 13976, 13994, 13995, 17007, 17008, 17009, 17010, 17011, 17012,</span>
#&gt; <span class='warning'>  17014, 17015, 17016, 17793, 18007, 20532, 22527, 22954, 22955, 22956,</span>
#&gt; <span class='warning'>  23347, 27275, 27276, 28075, 28085, 28893, 29046, 29047, 29242, 31362,</span>
#&gt; <span class='warning'>  32048, 32050, 33339, 33714, 33715, 32912, 32913, 40960, 40961, 44219,</span>
#&gt; <span class='warning'>  44220, 44242, 44262, 44263, 44356, 53476, 53477, 53497, 53498, 59180,</span>
#&gt; <span class='warning'>  57764, 57766, 61092, 69815, 69930, 70330, 70331, 70332, 77129, 77130,</span>
#&gt; <span class='warning'>  76239, 76240, 76241, 77987, and 77988. Note that ranges located on a</span>
#&gt; <span class='warning'>  sequence whose length is unknown (NA) or on a circular sequence are not</span>
#&gt; <span class='warning'>  considered out-of-bound (use seqlengths() and isCircular() to get the</span>
#&gt; <span class='warning'>  lengths and circularity flags of the underlying sequences). You can use</span>
#&gt; <span class='warning'>  trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more</span>
#&gt; <span class='warning'>  information.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Predicting coding ... done</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── Found 3 driver(s) </span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 3 x 10</span><span>
#&gt;   chr      from     to gene_symbol location consequence refAA varAA driver_label
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;fct&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> chr16  6.76</span><span style='color: #949494;'>e</span><span>7 6.76</span><span style='color: #949494;'>e</span><span>7 CTCF        coding:… nonsynonym… H     Y     CTCF_H-&gt;Y   
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> chr17  7.58</span><span style='color: #949494;'>e</span><span>6 7.58</span><span style='color: #949494;'>e</span><span>6 TP53        intron:… nonsynonym… P     A     TP53_P-&gt;A   
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> chr19  5.09</span><span style='color: #949494;'>e</span><span>7 5.09</span><span style='color: #949494;'>e</span><span>7 POLD1       coding   nonsynonym… T     M     POLD1_T-&gt;M  
#&gt; </span><span style='color: #949494;'># … with 1 more variable: is_driver &lt;lgl&gt;</span><span></div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


