<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Determines fragmentation of copy number segments — detect_arm_overfragmentation • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Determines fragmentation of copy number segments — detect_arm_overfragmentation" />
<meta property="og:description" content="This functions determines if, at the arm level, the segments are over-fragmented
using the statistical test described in the package manual." />
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Determines fragmentation of copy number segments</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/detect_arm_overfragmentation.R'><code>R/detect_arm_overfragmentation.R</code></a></small>
    <div class="hidden name"><code>detect_arm_overfragmentation.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This functions determines if, at the arm level, the segments are over-fragmented
using the statistical test described in the package manual.</p>
    </div>

    <pre class="usage"><span class='fu'>detect_arm_overfragmentation</span><span class='op'>(</span>
  <span class='va'>x</span>,
  alpha <span class='op'>=</span> <span class='fl'>0.01</span>,
  genome_percentage_cutoff <span class='op'>=</span> <span class='fl'>0.2</span>,
  minimum_segments_for_testing <span class='op'>=</span> <span class='fl'>10</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>Confidence level for the tests, for instance <code>0.05</code>.</p></td>
    </tr>
    <tr>
      <th>genome_percentage_cutoff</th>
      <td><p>Segments are considered long or short depending on whether
they are longer (in basepairs) than <code>genome_percentage_cutoff * L</code> bases, where <code>L</code>
is the arm length for the reference genome. Default is <code>0.2</code> (twenty percent).</p></td>
    </tr>
    <tr>
      <th>minimum_segments_for_testing</th>
      <td><p>Smallest number of segments required to actually test
a certain arm Default is <code>10</code> segments. This number influences the correction for mulitple
hypothesis testing.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cnaqc</code> with the results.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'example_dataset_CNAqc'</span><span class='op'>)</span>
<span class='va'>x</span> <span class='op'>=</span> <span class='fu'><a href='init.html'>init</a></span><span class='op'>(</span><span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>snvs</span>, <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>cna</span>, <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>purity</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 12963 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> Mapped n = 12963 mutations to clonal segments (100% of input)</span></div><div class='input'>
<span class='va'>x</span> <span class='op'>=</span> <span class='fu'>detect_arm_overfragmentation</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr7p</span><span>,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr1p</span><span>,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr11q</span><span>,  p = 2.4788992e-07 ~ 14 segments, 12 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr1q</span><span>,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr12q</span><span>,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr3q</span><span>,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr7q</span><span>,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✔</span><span> </span><span style='color: #00BB00;'>chr8p</span><span>,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> 8 significantly overfragmented chromosome arms (alpha level 0.01).</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $reference_genome
#&gt; [1] "GRCh38"
#&gt; 
#&gt; $snvs
#&gt; </span><span style='color: #949494;'># A tibble: 12,963 x 17</span><span>
#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> chr1  1</span><span style='text-decoration: underline;'>027</span><span>104 1</span><span style='text-decoration: underline;'>027</span><span>105 T     G     PASS      60     6 0.1    UTR5            
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> chr1  2</span><span style='text-decoration: underline;'>248</span><span>588 2</span><span style='text-decoration: underline;'>248</span><span>589 A     C     PASS     127     9 0.070</span><span style='text-decoration: underline;'>9</span><span> intergenic      
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> chr1  2</span><span style='text-decoration: underline;'>461</span><span>999 2</span><span style='text-decoration: underline;'>462</span><span>000 G     A     PASS     156    65 0.417  upstream        
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> chr1  2</span><span style='text-decoration: underline;'>727</span><span>935 2</span><span style='text-decoration: underline;'>727</span><span>936 T     C     PASS     180    90 0.5    downstream      
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> chr1  2</span><span style='text-decoration: underline;'>763</span><span>397 2</span><span style='text-decoration: underline;'>763</span><span>398 C     T     PASS     183    61 0.333  intergenic      
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> chr1  2</span><span style='text-decoration: underline;'>768</span><span>208 2</span><span style='text-decoration: underline;'>768</span><span>209 C     T     PASS     203   130 0.640  intergenic      
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> chr1  2</span><span style='text-decoration: underline;'>935</span><span>590 2</span><span style='text-decoration: underline;'>935</span><span>591 C     T     PASS     228   132 0.579  intergenic      
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> chr1  2</span><span style='text-decoration: underline;'>980</span><span>032 2</span><span style='text-decoration: underline;'>980</span><span>033 C     T     PASS     196    85 0.434  ncRNA_exonic    
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> chr1  3</span><span style='text-decoration: underline;'>387</span><span>161 3</span><span style='text-decoration: underline;'>387</span><span>162 T     G     PASS     124     6 0.048</span><span style='text-decoration: underline;'>4</span><span> intronic        
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> chr1  3</span><span style='text-decoration: underline;'>502</span><span>517 3</span><span style='text-decoration: underline;'>502</span><span>518 G     A     PASS      88    10 0.114  intronic        
#&gt; </span><span style='color: #949494;'># … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   segment_id &lt;chr&gt;</span><span>
#&gt; 
#&gt; $cna
#&gt; </span><span style='color: #949494;'># A tibble: 267 x 10</span><span>
#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> chr1   8.40</span><span style='color: #949494;'>e</span><span>5  1.69</span><span style='color: #949494;'>e</span><span>6 8.50</span><span style='color: #949494;'>e</span><span>5     1.19     3     2     1 chr1:840009:16…     1
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> chr1   1.69</span><span style='color: #949494;'>e</span><span>6  1.82</span><span style='color: #949494;'>e</span><span>6 1.25</span><span style='color: #949494;'>e</span><span>5     1.26     3     2     1 chr1:1689988:1…    </span><span style='color: #BB0000;'>NA</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> chr1   1.82</span><span style='color: #949494;'>e</span><span>6  9.80</span><span style='color: #949494;'>e</span><span>6 7.98</span><span style='color: #949494;'>e</span><span>6     1.19     3     2     1 chr1:1815016:9…    28
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> chr1   1.05</span><span style='color: #949494;'>e</span><span>7  1.21</span><span style='color: #949494;'>e</span><span>7 1.60</span><span style='color: #949494;'>e</span><span>6     1.19     3     2     1 chr1:10479910:…     1
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> chr1   1.21</span><span style='color: #949494;'>e</span><span>7  1.22</span><span style='color: #949494;'>e</span><span>7 7.51</span><span style='color: #949494;'>e</span><span>4     1.24     3     2     1 chr1:12079917:…    </span><span style='color: #BB0000;'>NA</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> chr1   1.22</span><span style='color: #949494;'>e</span><span>7  1.28</span><span style='color: #949494;'>e</span><span>7 6.85</span><span style='color: #949494;'>e</span><span>5     1.19     3     2     1 chr1:12154981:…    </span><span style='color: #BB0000;'>NA</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> chr1   1.38</span><span style='color: #949494;'>e</span><span>7  1.78</span><span style='color: #949494;'>e</span><span>7 4.01</span><span style='color: #949494;'>e</span><span>6     1.19     3     2     1 chr1:13780016:…     7
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> chr1   1.78</span><span style='color: #949494;'>e</span><span>7  2.11</span><span style='color: #949494;'>e</span><span>7 3.23</span><span style='color: #949494;'>e</span><span>6     1.19     3     2     1 chr1:17849962:…    12
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> chr1   2.11</span><span style='color: #949494;'>e</span><span>7  2.16</span><span style='color: #949494;'>e</span><span>7 4.80</span><span style='color: #949494;'>e</span><span>5     1.26     3     2     1 chr1:21080068:…    </span><span style='color: #BB0000;'>NA</span><span>
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> chr1   2.16</span><span style='color: #949494;'>e</span><span>7  2.48</span><span style='color: #949494;'>e</span><span>7 3.27</span><span style='color: #949494;'>e</span><span>6     1.19     3     2     1 chr1:21559998:…    11
#&gt; </span><span style='color: #949494;'># … with 257 more rows</span><span>
#&gt; 
#&gt; $n_snvs
#&gt; [1] 12963
#&gt; 
#&gt; $n_cna
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_clonal
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_sbclonal
#&gt; [1] 0
#&gt; 
#&gt; $n_karyotype
#&gt; 
#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 
#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 
#&gt; 
#&gt; $purity
#&gt; [1] 0.89
#&gt; 
#&gt; $l_karyotype
#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 
#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 
#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 
#&gt;     517926     225714     174002     172894     162913     130712      80103 
#&gt;       99:1 
#&gt;      75782 
#&gt; 
#&gt; $ploidy
#&gt; [1] 4
#&gt; 
#&gt; $most_prevalent_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; $basepairs_by_karyotype
#&gt; </span><span style='color: #949494;'># A tibble: 15 x 4</span><span>
#&gt;    minor Major          n karyotype
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span>     2     2 </span><span style='text-decoration: underline;'>1</span><span>482</span><span style='text-decoration: underline;'>910</span><span>013 2:2      
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span>     1     2  420</span><span style='text-decoration: underline;'>405</span><span>440 2:1      
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span>     2     3  357</span><span style='text-decoration: underline;'>172</span><span>683 3:2      
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span>     2     4  331</span><span style='text-decoration: underline;'>457</span><span>138 4:2      
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span>     0     3  137</span><span style='text-decoration: underline;'>498</span><span>772 3:0      
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span>     0     2   39</span><span style='text-decoration: underline;'>372</span><span>558 2:0      
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span>     1     3     </span><span style='text-decoration: underline;'>527</span><span>401 3:1      
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span>     2    25     </span><span style='text-decoration: underline;'>517</span><span>926 25:2     
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span>     1   106     </span><span style='text-decoration: underline;'>225</span><span>714 106:1    
#&gt; </span><span style='color: #BCBCBC;'>10</span><span>     0     5     </span><span style='text-decoration: underline;'>174</span><span>002 5:0      
#&gt; </span><span style='color: #BCBCBC;'>11</span><span>     2    69     </span><span style='text-decoration: underline;'>172</span><span>894 69:2     
#&gt; </span><span style='color: #BCBCBC;'>12</span><span>     2    26     </span><span style='text-decoration: underline;'>162</span><span>913 26:2     
#&gt; </span><span style='color: #BCBCBC;'>13</span><span>     2    16     </span><span style='text-decoration: underline;'>130</span><span>712 16:2     
#&gt; </span><span style='color: #BCBCBC;'>14</span><span>     2    70      </span><span style='text-decoration: underline;'>80</span><span>103 70:2     
#&gt; </span><span style='color: #BCBCBC;'>15</span><span>     1    99      </span><span style='text-decoration: underline;'>75</span><span>782 99:1     
#&gt; 
#&gt; $most_mutations_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; $arm_fragmentation
#&gt; $arm_fragmentation$table
#&gt; </span><span style='color: #949494;'># A tibble: 41 x 10</span><span>
#&gt;    chr   arm   n_long n_short jumps total_segments testable  p_value
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> chr7  p          0      34   206             34 TRUE     1.72</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-24</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> chr1  p          1      23     0             24 TRUE     1.63</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-15</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> chr1  q          1      12     5             13 TRUE     4.34</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 8</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> chr12 q          1      12   414             13 TRUE     4.34</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 8</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> chr11 q          2      12     0             14 TRUE     2.48</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 7</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> chr8  p          1      10     0             11 TRUE     9.22</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 7</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> chr3  q          2      10     2             12 TRUE     4.53</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 6</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> chr7  q          2      10     3             12 TRUE     4.53</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 6</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> chr11 p          1       8     0              9 FALSE    1.00</span><span style='color: #949494;'>e</span><span>+ 0
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> chr15 q          2       7     4              9 FALSE    1.00</span><span style='color: #949494;'>e</span><span>+ 0
#&gt; </span><span style='color: #949494;'># … with 31 more rows, and 2 more variables: Bonferroni_cutoff &lt;dbl&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   significant &lt;lgl&gt;</span><span>
#&gt; 
#&gt; $arm_fragmentation$alpha
#&gt; [1] 0.01
#&gt; 
#&gt; $arm_fragmentation$N_tests
#&gt; [1] 8
#&gt; 
#&gt; $arm_fragmentation$genome_percentage_cutoff
#&gt; [1] 0.2
#&gt; 
#&gt; $arm_fragmentation$minimum_segments_for_testing
#&gt; [1] 10
#&gt; 
#&gt; 
#&gt; attr(,"class")
#&gt; [1] "cnaqc"</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


