<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Computation of Cancer Cell Fractions • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Computation of Cancer Cell Fractions">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ccf_computation.html">Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/fragmentation.html">Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/smoothing.html">Smoothing copy number segments</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Computation of Cancer Cell Fractions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/ccf_computation.Rmd"><code>vignettes/ccf_computation.Rmd</code></a></small>
      <div class="hidden name"><code>ccf_computation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># To render in colour this vignette</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">old.hooks &lt;-<span class="st"> </span>fansi<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/fansi/man/set_knit_hooks.html">set_knit_hooks</a></span>(knitr<span class="op">::</span>knit_hooks)</a></code></pre></div>
<style type="text/css" scoped>
PRE.fansi SPAN {padding-top: .25em; padding-bottom: .25em};
</style>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#&gt; [32m✓[39m Loading [32m[32mCNAqc[32m[39m, [3m[3m'Copy Number Alteration quality check'[3m[23m. Support : [3m[34m[3m[34m&lt;https://caravagn.github.io/CNAqc/&gt;[34m[3m[39m[23m</span></a></code></pre></div>
<p><code>CNAqc</code> provides a function to compute mutation’s Cancer Cell Fractions (CCFs): <code>compute_CCF</code>. We suggest to use these to compare CCF estimates obtained with other tools.</p>
<p>The process of computing CCF values requires the computation of the <em>“mutation multiplicity”</em>, namely the number of copies of a mutation in a certain copy number segment. This is generally a difficult task which requires to account for tumour purity, copy number segments and tumour purity. Errors in this task propagate errors in CCF estimates, which in turn propagate in downstream analyses.</p>
<p>For this reason, <code>CNAqc</code>:</p>
<ul>
<li>limits the CCF computations to karyotypes that are usually “easier” - <code>'2:1'</code>, <code>'2:0'</code> and <code>'2:2'</code> - where a mutation is either present in one or two copies, assuming that the aneuploidy state is directly achieved from a diploid genome;</li>
<li>limits the CCF computations to clonal copy number segments.</li>
</ul>
<p>By design, computations on other karyotypes raise an error.</p>
<div id="mutation-multiplicity" class="section level3">
<h3 class="hasAnchor">
<a href="#mutation-multiplicity" class="anchor"></a>Mutation multiplicity</h3>
<p><strong>Expected VAF.</strong> Given the mutations that are mapped to a certain karyotype (e.g., <code>2:1</code>), the expected Variant Allele Frequency (VAF) for mutations in <span class="math inline">\(m\)</span> copies, with tumour purity <span class="math inline">\(\pi\)</span> and segment ploidy <span class="math inline">\(p\)</span> (number of copies of the minor and major allele) are given by <span class="math display">\[
v = \dfrac{m \pi}{
2 (1 - \pi) + \pi p 
} \, .
\]</span> This formula is also used to determine the expected peaks in the peak-detection algorithm used to QC copy number calls in <code>CNAqc</code> (see the vignette <em>“QC analysis via peaks detection”</em>).</p>
<p>In this task we determine the value of <span class="math inline">\(m\)</span> from the observed <span class="math inline">\(v\)</span>, the karyotype and tumour purity.</p>
<p><strong>2-class mixture.</strong> <code>CNAqc</code> uses a statistical model with two Binomial distributions for the mutations happening before, and after aneuploudy. The density of these Binomial distributions are:</p>
<ul>
<li>peaked at <span class="math inline">\(v_1\)</span> for <span class="math inline">\(m = 1\)</span> (after aneuploidy) and <span class="math inline">\(v_2\)</span> for <span class="math inline">\(m = 2\)</span> (before aneuploidy).</li>
<li>computed on the domain <span class="math inline">\([0, 1]\)</span>, after translating the read counts into allele frequencies. This is achieved fixing the number of trials of the Binomial process to the median coverage <span class="math inline">\(n\)</span> of the mutations that map to the segments under investigation (e.g., the triploid segments);</li>
</ul>
<p>Therefore, the assumptions in <code>CNAqc</code> are that 1) overdispersion of coverage is small to justify a Binomial instead of a Beta-Binomial model, and that 2) trials are well-represented by the median of the observed coverage.</p>
<p><code>CNAqc</code> computes two Binomial densities <span class="math inline">\(\text{Bin}_1\)</span> and <span class="math inline">\(\text{Bin}_2\)</span> over the set of values <span class="math inline">\([0; n]\)</span> with <span class="math inline">\(n\)</span> number of trials and success probability <span class="math inline">\(v_i\)</span>, i.e. <span class="math display">\[
\text{Bin}_i = p(s \mid n; v_i)
\]</span> as the probability of finding <span class="math inline">\(s\)</span> reads with the mutant allele at out of <span class="math inline">\(n\)</span> (median depth), assuming the mutation expected frequency is <span class="math inline">\(v_i\)</span> (notice that <span class="math inline">\(v_i\)</span> includes the effect of tumour purity).</p>
<p>To finalize the densities of this mixed model <code>CNAqc</code> determines the absolute proportions of this mixture, which are used to scale the two theoretical Binomial densities. This process can be done in differnt ways; <code>CNAqc</code> uses a simple heuristic that counts how many mutations fall below the area of each density. The heuristics computes the 1% and 99% quantiles of those distributions, determing two ranges <span class="math inline">\(I_1 = [v_1^1; v_1^{99}]\)</span> and <span class="math inline">\(I_2 = [v_2^1; v_2^{99}]\)</span> so that:</p>
<ul>
<li>
<span class="math inline">\(n_1\)</span> are the number of mutations with VAF in <span class="math inline">\(I_1\)</span>;</li>
<li>
<span class="math inline">\(n_2\)</span> are the number of mutations with VAF in <span class="math inline">\(I_2\)</span>;</li>
</ul>
<p>Notice that these two intervals might overlap (i.e., it could be that <span class="math inline">\(v_1^{99} &gt; v_2^{1}\)</span>). This is however not a problem because the two numbers are normalized to create a complete 2-components Binomial mixture <span class="math display">\[
M = \mu_1 * \text{Bin}_1 + (1 - \mu_1) * \text{Bin}_2
\]</span> where <span class="math inline">\(\mu_1 = n_1/(n_1+n_2)\)</span> are the normalized proportions.</p>
<p><strong>Entropy-based assignments.</strong> Assigning mutation multiplicities is particularly difficult at the crossing of the two Binomial densities, because in that case some mutations that have true multiplicity <span class="math inline">\(m=1\)</span> are actually mapped to <span class="math inline">\(m=2\)</span>, and viceversa. This creates false “bumps” in the CCF distribution, which could be interepreted either as miscalled copy-number segments or false clonal architectures.</p>
<p><code>CNAqc</code> uses a entropy-based heuristic to determine which mutations can be confidently assigned a value of <span class="math inline">\(m\)</span>, and the heuristic is controlled by a user-defined parameter that represents a quantile.</p>
<p>The mutations with VAF in between <span class="math inline">\(v_1\)</span> and <span class="math inline">\(v_2\)</span> are difficult - those with VAF below <span class="math inline">\(v_1\)</span> are clearly in one copy, those above <span class="math inline">\(v_2\)</span> are in two copies. Note that this argument implicitely assumes that we are working with clonal copy number calls, a simplyfying assumption of this analysis in <code>CNAqc</code>.</p>
<p><code>CNAqc</code> uses the entropy <span class="math inline">\(H(x)\)</span> of the mixture’s latent variables to determine mutation multiplicities, as well as its <em>numerical derivative</em> <span class="math inline">\(\partial H(x)\)</span> evaluated in the domain <span class="math inline">\([nv_1, nv_2]\)</span>. <span class="math display">\[
H(x) = - p_x \log(p_x)
\]</span> where <span class="math inline">\(p_x\)</span> is <span class="math inline">\(M\)</span>’s density at <span class="math inline">\(x\)</span>.</p>
<p>By construction, the derivative <em>grows when the uncertainty of the assignments is higher</em>, oscillating around low values in the leftmost and rightmost part of the evaluated spectrum, and peaking higher at the centre of the crossing of the two densities. This is assuming that segments calls are QCed properly via peak detection (see the Vignette).</p>
<p>The problem is therefore that of finding peaks in the derivative <span class="math inline">\(\partial H(x)\)</span>, which is done looking at a histogram of <span class="math inline">\(\partial H(x)\)</span> and determinig the quantile <span class="math inline">\(\hat{q}\)</span> (by default, the 90%). We expect most mutations to be confidently assigneable, and <span class="math inline">\(\hat{q}\)</span> defines the cutoff for the assignment:</p>
<ul>
<li>values below this quantile represent areas of the frequency spectrum where the assignments are confident towards either one of the two mixture components;</li>
<li>values above the quantile are the areas where we are not confident in the assignment.</li>
</ul>
<p><code>CNAqc</code> defines from the range <span class="math inline">\([nv_1, nv_2]\)</span> the points:</p>
<ul>
<li>
<span class="math inline">\(nv_1 \leq \hat{w}_1 &lt; nv_2\)</span> so that <span class="math inline">\(\hat{w}_1 = \min_{x}\{\partial_x H(x) &gt; \hat{q}\}\)</span>;</li>
<li>
<span class="math inline">\(nv_1 &lt; \hat{w}_2 \leq nv_2\)</span> so that <span class="math inline">\(\hat{w}_2 = \max_{x}\{\partial_x H(x) &gt; \hat{q}\}\)</span>.</li>
</ul>
<p>Then:</p>
<ul>
<li>any mutation in the range <span class="math inline">\([nv_1, n\hat{w}_1]\)</span> is considered having one mutation copy;</li>
<li>any mutation in the range <span class="math inline">\([n\hat{w}_2, nv_2]\)</span> is considered having two mutation copies;</li>
<li>any mutation in the range <span class="math inline">\([n\hat{w}_1, n\hat{w}_2]\)</span> is considered “Not Assignable” with confidence.</li>
</ul>
<p>When <code>CNAqc</code> computes this heuristic it returns a plot of the entropy and marks the ranges <span class="math inline">\([n\hat{w}_1, n\hat{w}_2]\)</span> determined by the quantile <span class="math inline">\(\hat{q}\)</span>. The user can then check if the quantile selected is too stringent or loose, and adjust the input parmeter accordingly.</p>
</div>
<div id="cancer-cell-fractions-computation" class="section level3">
<h3 class="hasAnchor">
<a href="#cancer-cell-fractions-computation" class="anchor"></a>Cancer Cell Fractions computation</h3>
<p>CCF values, are computed as</p>
<p><span class="math display">\[
\text{CCF} = 
\dfrac{
v[
(C - 2)\pi + 2
]
}
{
m \pi
}
\]</span> where <span class="math inline">\(C\)</span> is total segment ploidy (e.g., 3 for a <code>2:1</code> state).</p>
</div>
<div id="example-computation" class="section level3">
<h3 class="hasAnchor">
<a href="#example-computation" class="anchor"></a>Example computation</h3>
<p>We work with the same object shown in <code>CNAqc</code>’s Introductory Vignette.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Dataset available with the package -- see Getting Started.</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(x)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; ── [43m[30m[ CNAqc ] [39m[49m n = 13141 mutations in [32m[32m267[32m[39m segments (267 clonal + 0 subclonal) ───────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; [36mℹ[39m Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; [36mℹ[39m Mutation mapping (head): 7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0)</span></a></code></pre></div>
<p>We run function <code>compute_CCF</code> to obtain the full analysis of CCF values, first with the default parameters and focusing only one karyotype <code>2:1</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">x =<span class="st"> </span><span class="kw"><a href="../reference/compute_CCF.html">compute_CCF</a></span>(x, <span class="dt">karyotypes =</span> <span class="st">"2:1"</span>)</a></code></pre></div>
<pre class="fansi fansi-output"><code>#&gt; 
#&gt; <span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Computing mutation multiplicity with entropy H(x)</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">Binomial peaks</span><span>  </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span style="color: #555555;"># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     
#&gt; </span><span style="color: #555555;">1</span><span>                     1 2:1       0.308 One copy  
#&gt; </span><span style="color: #555555;">2</span><span>                     2 2:1       0.616 Two copies
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">H(x) quantile</span><span> 0.9 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt;  Median depth</span><span> 111 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; Mixing proportions ([1, 99] quantiles): </span><span> 0.554109589041096 0.445890410958904 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; H(x)-derived cutoffs</span><span> 0.441441441441441 0.468468468468468 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Summary CCF assignments</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #555555;"># A tibble: 3 x 3</span><span>
#&gt;   karyotype mutation_multiplicity assignments
#&gt;   </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>                     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span>       </span><span style="color: #555555;font-style: italic;">&lt;int&gt;</span><span>
#&gt; </span><span style="color: #555555;">1</span><span> 2:1                           1         886
#&gt; </span><span style="color: #555555;">2</span><span> 2:1                           2         675
#&gt; </span><span style="color: #555555;">3</span><span> 2:1                          </span><span style="color: #BB0000;">NA</span><span>           2
#&gt; Note: NA ~ mutations not confidently assignable at q = 0.9
</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co"># Print new object, it informs us of the available computation</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(x)</a></code></pre></div>
<p>We can visually inspect the CCF estimates for the computed karyotypes. We can do that accessing the field <code>x$CCF_estimates</code> which contains the data and a plot reporting the result of the analysis, or using the getter functions <code>CCF</code> and <code>plot_CCF</code>.</p>
<p>The tibble of the data reports the computed values in column <code>"CCF"</code>, and in column <code>"mutation multiplicity"</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Tibble</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="../reference/CCF.html">CCF</a></span>(x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(VAF, mutation_multiplicity, CCF)</a></code></pre></div>
<pre class="fansi fansi-output"><code>#&gt; <span style="color: #555555;"># A tibble: 1,563 x 3</span><span>
#&gt;       VAF mutation_multiplicity    CCF
#&gt;     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span>                 </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span>  </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span>
#&gt; </span><span style="color: #555555;"> 1</span><span> 0.558                      2  0.905
#&gt; </span><span style="color: #555555;"> 2</span><span> 0.467                     </span><span style="color: #BB0000;">NA</span><span> </span><span style="color: #BB0000;">NA</span><span>    
#&gt; </span><span style="color: #555555;"> 3</span><span> 0.613                      2  0.995
#&gt; </span><span style="color: #555555;"> 4</span><span> 0.529                      2  0.859
#&gt; </span><span style="color: #555555;"> 5</span><span> 0.556                      2  0.902
#&gt; </span><span style="color: #555555;"> 6</span><span> 0.336                      1  1.09 
#&gt; </span><span style="color: #555555;"> 7</span><span> 0.532                      2  0.864
#&gt; </span><span style="color: #555555;"> 8</span><span> 0.261                      1  0.848
#&gt; </span><span style="color: #555555;"> 9</span><span> 0.567                      2  0.920
#&gt; </span><span style="color: #555555;">10</span><span> 0.079</span><span style="text-decoration: underline;">4</span><span>                     1  0.258
#&gt; </span><span style="color: #555555;"># … with 1,553 more rows</span><span>
</span></code></pre>
<p>The plot is a <code>ggpubr</code>-assembled figure, which in this case contains only the estimates for one karyotype.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Tibble</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="../reference/plot_CCF.html">plot_CCF</a></span>(x)</a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-7-1.png" width="1056"></p>
<p>The plot shows all the quantities described above; from left to right the plot shows</p>
<ul>
<li>the CCF histogram, coloured by mutation multiplicity;</li>
<li>the histogram of the input VAF, coloured by mutation multiplicity;</li>
<li>the probability of being assigned to the first mixture components (<span class="math inline">\(\text{Bin}_1\)</span>), and the two Binomial peaks;</li>
<li>the entropy profile <span class="math inline">\(H(x)\)</span>, its derivative <span class="math inline">\(\partial H(x)\)</span> and the range determined by the input quantile <span class="math inline">\(\hat{q}\)</span>;</li>
<li>the derivative magnitued and its quantile <span class="math inline">\(\hat{q}\)</span>.</li>
</ul>
<p>In this case we see for instance that <span class="math inline">\(\hat{q} = 0.9\)</span> (default value) is a little bit off the required value, because the determined range <span class="math inline">\(I\)</span> is not fully including the peak in the entropy values. However, we see that beacuse the data quality is high the two Binomial distributions are well separated. Therefore even with this parameter value we still obtain perfectly reasonable mutation multiplicities and CCF estimates.</p>
<p>We inspect also the remaining karyotypes, and find a similar satisfactory result.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># Analyze all of them now</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">x =<span class="st"> </span><span class="kw"><a href="../reference/compute_CCF.html">compute_CCF</a></span>(x)</a></code></pre></div>
<pre class="fansi fansi-output"><code>#&gt; 
#&gt; <span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Computing mutation multiplicity with entropy H(x)</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">Binomial peaks</span><span>  </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span style="color: #555555;"># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     
#&gt; </span><span style="color: #555555;">1</span><span>                     1 2:1       0.308 One copy  
#&gt; </span><span style="color: #555555;">2</span><span>                     2 2:1       0.616 Two copies
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">H(x) quantile</span><span> 0.9 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt;  Median depth</span><span> 111 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; Mixing proportions ([1, 99] quantiles): </span><span> 0.554109589041096 0.445890410958904 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; H(x)-derived cutoffs</span><span> 0.441441441441441 0.468468468468468 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Computing mutation multiplicity with entropy H(x)</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">Binomial peaks</span><span>  </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span style="color: #555555;"># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     
#&gt; </span><span style="color: #555555;">1</span><span>                     1 2:0       0.445 One copy  
#&gt; </span><span style="color: #555555;">2</span><span>                     2 2:0       0.89  Two copies
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">H(x) quantile</span><span> 0.9 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt;  Median depth</span><span> 76 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; Mixing proportions ([1, 99] quantiles): </span><span> 0.0933333333333333 0.906666666666667 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; H(x)-derived cutoffs</span><span> 0.657894736842105 0.710526315789474 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Computing mutation multiplicity with entropy H(x)</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">Binomial peaks</span><span>  </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span style="color: #555555;"># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>     
#&gt; </span><span style="color: #555555;">1</span><span>                     1 2:2       0.235 One copy  
#&gt; </span><span style="color: #555555;">2</span><span>                     2 2:2       0.471 Two copies
#&gt; </span><span style="color: #000000;background-color: #BBBB00;">H(x) quantile</span><span> 0.9 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt;  Median depth</span><span> 141 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; Mixing proportions ([1, 99] quantiles): </span><span> 0.0946892325649256 0.905310767435074 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; H(x)-derived cutoffs</span><span> 0.319148936170213 0.347517730496454 </span><span style="color: #000000;background-color: #BBBB00;">
#&gt; </span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #00BBBB;"> Summary CCF assignments</span><span>
#&gt; </span><span style="color: #00BBBB;">=-=-=-=-=-=-=-=-=-=-=-=-=-</span><span>
#&gt; </span><span style="color: #555555;"># A tibble: 8 x 3</span><span>
#&gt;   karyotype mutation_multiplicity assignments
#&gt;   </span><span style="color: #555555;font-style: italic;">&lt;chr&gt;</span><span>                     </span><span style="color: #555555;font-style: italic;">&lt;dbl&gt;</span><span>       </span><span style="color: #555555;font-style: italic;">&lt;int&gt;</span><span>
#&gt; </span><span style="color: #555555;">1</span><span> 2:0                           1           9
#&gt; </span><span style="color: #555555;">2</span><span> 2:0                           2          72
#&gt; </span><span style="color: #555555;">3</span><span> 2:1                           1         886
#&gt; </span><span style="color: #555555;">4</span><span> 2:1                           2         675
#&gt; </span><span style="color: #555555;">5</span><span> 2:1                          </span><span style="color: #BB0000;">NA</span><span>           2
#&gt; </span><span style="color: #555555;">6</span><span> 2:2                           1        </span><span style="text-decoration: underline;">1</span><span>013
#&gt; </span><span style="color: #555555;">7</span><span> 2:2                           2        </span><span style="text-decoration: underline;">6</span><span>427
#&gt; </span><span style="color: #555555;">8</span><span> 2:2                          </span><span style="color: #BB0000;">NA</span><span>          38
#&gt; Note: NA ~ mutations not confidently assignable at q = 0.9
</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="../reference/plot_CCF.html">plot_CCF</a></span>(x)</a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-8-1.png" width="1056"></p>
<p>Just for example, this is how CCF values would have changed if we had used different value of <span class="math inline">\(\hat{q}\)</span>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co"># Bad estimates for q = .5</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw"><a href="../reference/plot_CCF.html">plot_CCF</a></span>(examples_low)</a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co"># Still good estimates for q = .99</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw"><a href="../reference/plot_CCF.html">plot_CCF</a></span>(examples_high)</a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-10-2.png" width="1056"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
