<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Computation of Cancer Cell Fractions • CNAqc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Computation of Cancer Cell Fractions">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ccf_computation.html">Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Computation of Cancer Cell Fractions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/ccf_computation.Rmd"><code>vignettes/ccf_computation.Rmd</code></a></small>
      <div class="hidden name"><code>ccf_computation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt;  [ CNAqc - Copy Number Alteration quality check ] </span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/CNAqc</span></a></code></pre></div>
<div id="mutation-multiplicity" class="section level3">
<h3 class="hasAnchor">
<a href="#mutation-multiplicity" class="anchor"></a>Mutation multiplicity</h3>
<p>To compute Cancer Cell Fractions (CCFs), <code>CNAqc</code> first computes the mutation multiplicity (the numbe of copies of a mutation in a certain copy number segment). This is generally a difficult task, and wrong CCF estimates in turn propagate errors in other analyses. For this reason, <code>CNAqc</code> limits this computations to karyotypes that are usually “easier” - <code>'2:1'</code>, <code>'2:0'</code> and <code>'2:2'</code> - where a mutation is either present in one or two copies, assuming that the aneuploidy state is directly achieved from a diploid genome.By design, computations on other karyotypes raise an error.</p>
<p><strong>Expected VAF.</strong> Given the mutations that are mapped to a certain karyotype (e.g., <code>2:1</code>), the expected Variant Allele Frequency (VAF) for mutations in <span class="math inline">\(m\)</span> copies, with tumour purity <span class="math inline">\(\pi\)</span> and segment ploidy <span class="math inline">\(p\)</span> (number of copies of the minor and major allele) are given by <span class="math display">\[
v = \dfrac{m \pi}{
2 (1 - \pi) + \pi p 
} \, .
\]</span> This formula is also used to determine the expected peaks in the peak-detection algorithm used to QC copy number calls in <code>CNAqc</code> (see the vignette <em>“QC analysis via peaks detection”</em>).</p>
<p>In this task we need to determine the value of <span class="math inline">\(m\)</span> from the observed <span class="math inline">\(v\)</span> and the karyotype.</p>
<p><strong>2-class mixture.</strong> <code>CNAqc</code> uses a heuristic based on the fact that we expect two Binomial distributions for the mutations happening before and after aneuploudy. The density of these Binomial distributions are:</p>
<ul>
<li>peaked at <span class="math inline">\(v_1\)</span> for <span class="math inline">\(m = 1\)</span> (after aneuploidy) and <span class="math inline">\(v_2\)</span> for <span class="math inline">\(m = 2\)</span> (before aneuploidy).</li>
<li>computed on the domain <span class="math inline">\([0, 1]\)</span>, after translating the read counts into allele frequencies. This is achieved fixing the number of trials of the Binomial process to the median coverage <span class="math inline">\(n\)</span> of the mutations that map to the segments under investigation (e.g., the triploid segments);</li>
</ul>
<p>Therefore, the assumptions in <code>CNAqc</code> are that overdispersion of coverage is small to justify a Binomial instead of a Beta-Binomial model, and that trials are well-represented with the median coverage.</p>
<p><code>CNAqc</code> computes two Binomial densities <span class="math inline">\(\text{Bin}_1\)</span> and <span class="math inline">\(\text{Bin}_2\)</span> over the set of values <span class="math inline">\([0; n]\)</span> with <span class="math inline">\(n\)</span> number of trials and success probability <span class="math inline">\(v_i\)</span>, i.e. <span class="math display">\[
\text{Bin}_i = p(s \mid n; v_i)
\]</span> as the probability of finding <span class="math inline">\(s\)</span> reads with the mutant allele at out of <span class="math inline">\(n\)</span> (median depth), assuming the mutation expected frequency is <span class="math inline">\(v_i\)</span>.</p>
<p>To finalize the densities of this mixed model is however required to determine the absolute proportions of the mixture; this quantity must be indeed used to actually scale the two theoretical densities. To do this <code>CNAqc</code> uses a simple heuristic that counts how many mutations map below the area of each of those densities. The heuristics computes the 1% and 99% quantiles of those distributions, determing two ranges <span class="math inline">\(I_1 = [v_1^1; v_1^{99}]\)</span> and <span class="math inline">\(I_2 = [v_2^1; v_2^{99}]\)</span> so that:</p>
<ul>
<li>
<span class="math inline">\(n_1\)</span> are the number of mutations with VAF in <span class="math inline">\(I_1\)</span>;</li>
<li>
<span class="math inline">\(n_2\)</span> are the number of mutations with VAF in <span class="math inline">\(I_2\)</span>;</li>
</ul>
<p>Notice that these two intervals might overlap (i.e., it could be that <span class="math inline">\(v_1^{99} &gt; v_2^{1}\)</span>). This is however not a problem because the two numbers are normalized to create a complete 2-components Binomial mixture model. <span class="math display">\[
M = \mu_1 * \text{Bin}_1 + (1 - \mu_1) * \text{Bin}_2
\]</span> where <span class="math inline">\(\mu_1 = n_1/(n_1+n_2)\)</span> are the normalized proportions.</p>
<p><strong>Entropy-based assignments.</strong> Assigning mutation multiplicities is particularly difficult at the crossing of the two Binomial densities, because in that case some mutations that have true multiplicity <span class="math inline">\(m=1\)</span> are actually mapped to <span class="math inline">\(m=2\)</span>, and viceversa. This creates false “bumps” in the CCF distribution, which could be interepreted either as miscalled copy-number segments or false clonal architectures. <code>CNAqc</code> uses a entropy-based heuristic to determine which mutations can be confidently assigned to one of the two states of <span class="math inline">\(m\)</span>, and the heuristic is controlled by a user-defined parameter that represents a quantile. Plots are returned to visually aid the user to determine the best value for the input parameter.</p>
<p>The difficult mutations to assign are those laying in between <span class="math inline">\(v_1\)</span> and <span class="math inline">\(v_2\)</span>. It is reasonable to assume that mutations with VFA below <span class="math inline">\(v_1\)</span> are in one copy, and that those above <span class="math inline">\(v_2\)</span> are in two copies. This implicitely assume that we are working with clonal copy number calls, a simplyfying assumption of this analysis in <code>CNAqc</code>.</p>
<p><code>CNAqc</code> uses the entropy <span class="math inline">\(H(x)\)</span> of the latent variables to determine mutation multiplicities, as well as its numerical derivative <span class="math inline">\(\partial_x H\)</span> evaluated in the domain <span class="math inline">\([nv_1, nv_2]\)</span>. Both quantities are computed from the template mixture <span class="math inline">\(M\)</span>. The derivative is the quantity that grows when the uncertainty of the assignments is higher. By definitiion this oscillates around low values in the leftmost and rightmost part of the evaluated spectrum, and peaks higher at the centre of the crossing of the two densities.</p>
<p>To determine which values of the derivatives are large to suggest uncertainty, the binned values of <span class="math inline">\(\partial_x H\)</span> are used to determine the quantile <span class="math inline">\(\hat{q}\)</span> (by default, the 90%); because we expect most values to be confidently assigneable, the quantile is the determinant of the cutoff for the assignment. In fact, values below this quantile represent areas of the frequency spectrum where the assignments are confident towards either one of the two mixture components. Values above the quantile are the areas where we are not confident in the assignment.</p>
<p>Therefore, define from the range <span class="math inline">\([nv_1, nv_2]\)</span> the points:</p>
<ul>
<li>
<span class="math inline">\(nv_1 \leq \hat{w}_1 &lt; nv_2\)</span> so that <span class="math inline">\(\hat{w}_1 = \min_{x}\{\partial_x H(x) &gt; \hat{q}\}\)</span>;</li>
<li>
<span class="math inline">\(nv_1 &lt; \hat{w}_2 \leq nv_2\)</span> so that <span class="math inline">\(\hat{w}_2 = \max_{x}\{\partial_x H(x) &gt; \hat{q}\}\)</span>.</li>
</ul>
<p>Then:</p>
<ul>
<li>any mutation in the range <span class="math inline">\([nv_1, n\hat{w}_1]\)</span> is considered having one mutation copy;</li>
<li>any mutation in the range <span class="math inline">\([n\hat{w}_2, nv_2]\)</span> is considered having two mutation copies;</li>
<li>any mutation in the range <span class="math inline">\([n\hat{w}_1, n\hat{w}_2]\)</span> is considered “Not Assignable” with confidence.</li>
</ul>
<p>When <code>CNAqc</code> computes this heuristic it returns a plot of the entropy and marks the ranges <span class="math inline">\([n\hat{w}_1, n\hat{w}_2]\)</span> determined by the quantile <span class="math inline">\(\hat{q}\)</span>, so that if the user sees that the quantile selected are too stringent or loose, the input parmeter can be adjusted accordingly.</p>
<div id="example-computation" class="section level4">
<h4 class="hasAnchor">
<a href="#example-computation" class="anchor"></a>Example computation</h4>
<p>We work with the same object shown in <code>CNAqc</code>’s Introductory Vignette.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Dataset available with the package -- see Getting Started.</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt;       CNAqc  n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt;      Purity  89% cellularity </span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt;  Karyotypes  7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0); 4 (16:2); 2 (25:2); 2 (3:1); 1 (106:1); 1 (26:2); 1 (99:1) </span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt;         CCF  FALSE</span></a></code></pre></div>
<p>We run function <code>compute_CCF</code> to obtain the full analysis of CCF values, first with the default parameters.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">x_default =<span class="st"> </span><span class="kw"><a href="../reference/compute_CCF.html">compute_CCF</a></span>(x)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;  Computing mutation multiplicity with entropy H(x)</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; Binomial peaks  </span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; # A tibble: 2 x 4</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;   mutation_multiplicity karyotype  peak label     </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; 1                     1 2:1       0.308 One copy  </span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; 2                     2 2:1       0.616 Two copies</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; H(x) quantile 0.9 </span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  Median depth 111 </span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt; Mixing proportions ([1, 99] quantiles):  0.554109589041096 0.445890410958904 </span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; H(x)-derived cutoffs 0.441441441441441 0.468468468468468</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; Warning in FUN(X[[i]], ...): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; Warning: Removed 2 rows containing non-finite values (stat_bin).</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt;  Computing mutation multiplicity with entropy H(x)</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; Binomial peaks  </span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt; # A tibble: 2 x 4</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt;   mutation_multiplicity karyotype  peak label     </span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt; 1                     1 2:0       0.445 One copy  </span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt; 2                     2 2:0       0.89  Two copies</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt; H(x) quantile 0.9 </span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt;  Median depth 76 </span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt; Mixing proportions ([1, 99] quantiles):  0.0933333333333333 0.906666666666667 </span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt; H(x)-derived cutoffs 0.657894736842105 0.710526315789474</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt; Warning: Removed 4 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt;  Computing mutation multiplicity with entropy H(x)</span></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">#&gt; Binomial peaks  </span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co">#&gt; # A tibble: 2 x 4</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="co">#&gt;   mutation_multiplicity karyotype  peak label     </span></a>
<a class="sourceLine" id="cb3-44" data-line-number="44"><span class="co">#&gt;                   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"><span class="co">#&gt; 1                     1 2:2       0.235 One copy  </span></a>
<a class="sourceLine" id="cb3-46" data-line-number="46"><span class="co">#&gt; 2                     2 2:2       0.471 Two copies</span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="co">#&gt; H(x) quantile 0.9 </span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="co">#&gt;  Median depth 141 </span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="co">#&gt; Mixing proportions ([1, 99] quantiles):  0.0946892325649256 0.905310767435074 </span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">#&gt; H(x)-derived cutoffs 0.319148936170213 0.347517730496454</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">#&gt; Warning in FUN(X[[i]], ...): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="co">#&gt; Warning: Removed 38 rows containing non-finite values (stat_bin).</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="co">#&gt; Warning: Removed 6 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-56" data-line-number="56"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-57" data-line-number="57"><span class="co">#&gt;  Summary CCF assignments</span></a>
<a class="sourceLine" id="cb3-58" data-line-number="58"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59"><span class="co">#&gt; # A tibble: 8 x 3</span></a>
<a class="sourceLine" id="cb3-60" data-line-number="60"><span class="co">#&gt;   karyotype mutation_multiplicity assignments</span></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="co">#&gt;   &lt;chr&gt;                     &lt;dbl&gt;       &lt;int&gt;</span></a>
<a class="sourceLine" id="cb3-62" data-line-number="62"><span class="co">#&gt; 1 2:0                           1           9</span></a>
<a class="sourceLine" id="cb3-63" data-line-number="63"><span class="co">#&gt; 2 2:0                           2          72</span></a>
<a class="sourceLine" id="cb3-64" data-line-number="64"><span class="co">#&gt; 3 2:1                           1         886</span></a>
<a class="sourceLine" id="cb3-65" data-line-number="65"><span class="co">#&gt; 4 2:1                           2         675</span></a>
<a class="sourceLine" id="cb3-66" data-line-number="66"><span class="co">#&gt; 5 2:1                          NA           2</span></a>
<a class="sourceLine" id="cb3-67" data-line-number="67"><span class="co">#&gt; 6 2:2                           1        1013</span></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"><span class="co">#&gt; 7 2:2                           2        6427</span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"><span class="co">#&gt; 8 2:2                          NA          38</span></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="co">#&gt; Note: NA ~ mutations not confidently assignable at q = 0.9</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="co"># Print new object</span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x_default)</a>
<a class="sourceLine" id="cb3-74" data-line-number="74"><span class="co">#&gt;       CNAqc  n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb3-75" data-line-number="75"><span class="co">#&gt;      Purity  89% cellularity </span></a>
<a class="sourceLine" id="cb3-76" data-line-number="76"><span class="co">#&gt;  Karyotypes  7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0); 4 (16:2); 2 (25:2); 2 (3:1); 1 (106:1); 1 (26:2); 1 (99:1) </span></a>
<a class="sourceLine" id="cb3-77" data-line-number="77"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb3-78" data-line-number="78"><span class="co">#&gt;    Peaks QC  FALSE </span></a>
<a class="sourceLine" id="cb3-79" data-line-number="79"><span class="co">#&gt;         CCF  TRUE  ~  2:1, 2:0, 2:2</span></a></code></pre></div>
<p>We can visually inspect the CCF estimates for the default karyotypes (<code>'2:1'</code>, <code>'2:0'</code> and <code>'2:2'</code>). We can do that accessing the field <code>CCF_estimates</code> which contains a plot reporting the result of the analysis.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Result for karyptype '2:1' (triploid)</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x_default<span class="op">$</span>CCF_estimates<span class="op">$</span><span class="st">`</span><span class="dt">2:1</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; $mutations</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; # A tibble: 1,563 x 15</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;  1 chr2  3.58e5 3.58e5 C     A     PASS     104    58 0.558 </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;  2 chr2  9.09e5 9.09e5 A     G     PASS      15     7 0.467 </span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;  3 chr2  1.04e6 1.04e6 C     T     PASS      93    57 0.613 </span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;  4 chr2  1.33e6 1.33e6 A     T     PASS     104    55 0.529 </span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;  5 chr2  1.52e6 1.52e6 C     T     PASS      90    50 0.556 </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;  6 chr2  2.20e6 2.20e6 G     T     PASS     125    42 0.336 </span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;  7 chr2  2.90e6 2.90e6 C     T     PASS     109    58 0.532 </span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;  8 chr2  3.13e6 3.13e6 A     G     PASS     134    35 0.261 </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;  9 chr2  3.83e6 3.83e6 A     T     PASS     120    68 0.567 </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; 10 chr2  3.88e6 3.88e6 T     A     PASS     126    10 0.0794</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; # … with 1,553 more rows, and 6 more variables: ANNOVAR_FUNCTION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; #   GENE &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; #   mutation_multiplicity &lt;dbl&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; $plot</span></a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-5-1.png" width="1056"> The plot shows all the quantities described above; in this case we see for instance that <span class="math inline">\(\hat{q} = 0.9\)</span> (default value) is a little bit off the required value, because the determined range <span class="math inline">\(I\)</span> is not fully including the peak in the entropy values. However, we see that beacuse the data quality is high the two Binomial distributions are well separated; Therefore even with this parameter value we still obtain perfectly reasonable mutation multiplicities and CCF estimates.</p>
<p>We inspect also the remaining karyotypes, and find a similar satisfactory result.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Result for karyptype '2:2' (tetraploid)</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x_default<span class="op">$</span>CCF_estimates<span class="op">$</span><span class="st">`</span><span class="dt">2:2</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; $mutations</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; # A tibble: 7,478 x 15</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt;  1 chr4  5.49e4 5.49e4 C     A     PASS     718    44 0.0613</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;  2 chr4  5.74e5 5.74e5 G     C     PASS     164    16 0.0976</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;  3 chr4  1.11e6 1.11e6 G     A     PASS     157    66 0.420 </span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  4 chr4  1.40e6 1.40e6 C     T     PASS     178    93 0.522 </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  5 chr4  1.54e6 1.54e6 G     A     PASS     155    68 0.439 </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt;  6 chr4  1.57e6 1.57e6 C     T     PASS     172    77 0.448 </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt;  7 chr4  1.60e6 1.60e6 G     A     PASS     139    61 0.439 </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt;  8 chr4  1.80e6 1.80e6 G     A     PASS     161    79 0.491 </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt;  9 chr4  2.29e6 2.29e6 C     T     PASS     168    74 0.440 </span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; 10 chr4  2.36e6 2.36e6 T     G     PASS     100     6 0.06  </span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; # … with 7,468 more rows, and 6 more variables: ANNOVAR_FUNCTION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; #   GENE &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; #   mutation_multiplicity &lt;dbl&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; $plot</span></a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-6-1.png" width="1056"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co"># Result for karyptype '2:0' (tetraploid)</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x_default<span class="op">$</span>CCF_estimates<span class="op">$</span><span class="st">`</span><span class="dt">2:0</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; $mutations</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; # A tibble: 81 x 15</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV   VAF</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt;  1 chr12 5.21e7 5.21e7 A     C     PASS      75     6 0.08 </span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt;  2 chr12 5.25e7 5.25e7 G     A     PASS      60    53 0.883</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;  3 chr12 5.34e7 5.34e7 C     T     PASS      82    75 0.915</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;  4 chr12 5.48e7 5.48e7 C     T     PASS      86    77 0.895</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;  5 chr12 5.48e7 5.48e7 A     T     PASS      72    62 0.861</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;  6 chr12 5.54e7 5.54e7 T     A     PASS      85    40 0.471</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;  7 chr12 5.78e7 5.78e7 C     T     PASS      73    62 0.849</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;  8 chr12 5.86e7 5.86e7 C     T     PASS      65    51 0.785</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;  9 chr12 5.86e7 5.86e7 T     A     PASS      64    57 0.891</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; 10 chr12 5.90e7 5.90e7 G     A     PASS      54    44 0.815</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; # … with 71 more rows, and 6 more variables: ANNOVAR_FUNCTION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; #   GENE &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; #   mutation_multiplicity &lt;dbl&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; $plot</span></a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-6-2.png" width="1056"></p>
<p>Just for example, this is how CCF values would have changed if we had used different value of <span class="math inline">\(\hat{q}\)</span>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Bad estimates for q = .5</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(examples_low<span class="op">$</span>CCF_estimates)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; $`2:1`</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; $`2:1`$mutations</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; # A tibble: 1,563 x 15</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt;  1 chr2  3.58e5 3.58e5 C     A     PASS     104    58 0.558 </span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt;  2 chr2  9.09e5 9.09e5 A     G     PASS      15     7 0.467 </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt;  3 chr2  1.04e6 1.04e6 C     T     PASS      93    57 0.613 </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt;  4 chr2  1.33e6 1.33e6 A     T     PASS     104    55 0.529 </span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt;  5 chr2  1.52e6 1.52e6 C     T     PASS      90    50 0.556 </span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt;  6 chr2  2.20e6 2.20e6 G     T     PASS     125    42 0.336 </span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt;  7 chr2  2.90e6 2.90e6 C     T     PASS     109    58 0.532 </span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt;  8 chr2  3.13e6 3.13e6 A     G     PASS     134    35 0.261 </span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt;  9 chr2  3.83e6 3.83e6 A     T     PASS     120    68 0.567 </span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; 10 chr2  3.88e6 3.88e6 T     A     PASS     126    10 0.0794</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; # … with 1,553 more rows, and 6 more variables: ANNOVAR_FUNCTION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt; #   GENE &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt; #   mutation_multiplicity &lt;dbl&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt; $`2:1`$plot</span></a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-8-1.png" width="1056"></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># Still good estimates for q = .9</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(examples_high<span class="op">$</span>CCF_estimates)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; $`2:1`</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt; $`2:1`$mutations</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt; # A tibble: 1,563 x 15</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  1 chr2  3.58e5 3.58e5 C     A     PASS     104    58 0.558 </span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt;  2 chr2  9.09e5 9.09e5 A     G     PASS      15     7 0.467 </span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt;  3 chr2  1.04e6 1.04e6 C     T     PASS      93    57 0.613 </span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;  4 chr2  1.33e6 1.33e6 A     T     PASS     104    55 0.529 </span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt;  5 chr2  1.52e6 1.52e6 C     T     PASS      90    50 0.556 </span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt;  6 chr2  2.20e6 2.20e6 G     T     PASS     125    42 0.336 </span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt;  7 chr2  2.90e6 2.90e6 C     T     PASS     109    58 0.532 </span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt;  8 chr2  3.13e6 3.13e6 A     G     PASS     134    35 0.261 </span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt;  9 chr2  3.83e6 3.83e6 A     T     PASS     120    68 0.567 </span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt; 10 chr2  3.88e6 3.88e6 T     A     PASS     126    10 0.0794</span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="co">#&gt; # … with 1,553 more rows, and 6 more variables: ANNOVAR_FUNCTION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="co">#&gt; #   GENE &lt;chr&gt;, karyotype &lt;chr&gt;, segment_id &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="co">#&gt; #   mutation_multiplicity &lt;dbl&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="co">#&gt; $`2:1`$plot</span></a></code></pre></div>
<p><img src="ccf_computation_files/figure-html/unnamed-chunk-8-2.png" width="1056"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
