<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • CNAqc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/CNAqc.Rmd"><code>vignettes/CNAqc.Rmd</code></a></small>
      <div class="hidden name"><code>CNAqc.Rmd</code></div>

    </div>

    
    
<p>This vignette describes how to set-up your data into the <code>CNAqc</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt;  [ CNAqc - Copy Number Alteration quality check ] </span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/CNAqc</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">require</a></span>(tidyverse)</a></code></pre></div>
<p>The input mutations must be a <code>data.frame</code> with:</p>
<ul>
<li>
<code>chr</code>, <code>from</code>, <code>to</code> as <code>hg19</code> chromosome coordinates.
<ul>
<li>Chromosome names must be in the format <code>chr1</code>, <code>chr2</code>, etc.;</li>
</ul>
</li>
<li>
<code>ref</code> and <code>alt</code> for the reference and alternative alleles.
<ul>
<li>Alleles should be characters, the tool will use all available mutation types but we suggest to carry out QC with high-quality SNV data;</li>
</ul>
</li>
<li>
<code>DP</code>, <code>NV</code> and <code>VAF</code> (numeric) for
<ul>
<li>the base depth - total number of reads with both reference and alternative;</li>
<li>the number of variants with the alternative allele;</li>
<li>the allele frequency (<code>VAF = NV/DP</code>).</li>
</ul>
</li>
</ul>
<p>The input CNAs must be a <code>data.frame</code> with:</p>
<ul>
<li><p><code>chr</code>, <code>from</code>, <code>to</code> as <code>hg19</code> chromosome coordinates, (as for mutations);</p></li>
<li><p><code>Major</code> and <code>minor</code> for the number of copies of the major allele, and the minor (B-allele). A diploid segment will have terefore <code>Major = minor = 1</code> and will be denoted as <code>1:1</code>.</p></li>
<li>A <code>CCF</code> column can be used to represent the Cancer Cell Fractionsas (i.e., clonality of the segment) a real-value in between 0 and 1 to.
<ul>
<li>If this is not available, <code>CCF = 1</code> is set assuming clonal segments. Otherwise, segments with <code>CCF&lt;1</code> are considered subclonal CNAs.</li>
</ul>
</li>
</ul>
<p>To analyze the data, a purity value in between <code>0</code> and <code>1</code> is required to represent the proportion of actual tumour content (sometimes called “cellularity”).</p>
<p>The package comes with a template dataset available.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">'example_dataset_CNAqc'</span>, <span class="dt">package =</span> <span class="st">'CNAqc'</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(example_dataset_CNAqc)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; $snvs</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; # A tibble: 13,141 x 51</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt;    chr   from  ID    ref   alt   QUAL  FILTER INFO  id       DP    NV</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt;  1 chr1  1027… &lt;NA&gt;  T     G     1.73… PASS   DP=8… chr1…    60     6</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt;  2 chr1  2248… &lt;NA&gt;  A     C     -0    PASS   DP=1… chr1…   127     9</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt;  3 chr1  2461… &lt;NA&gt;  G     A     225   PASS   DP=1… chr1…   156    65</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt;  4 chr1  2727… &lt;NA&gt;  T     C     225   PASS   DP=1… chr1…   180    90</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt;  5 chr1  2763… &lt;NA&gt;  C     T     225   PASS   DP=1… chr1…   183    61</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt;  6 chr1  2768… rs18… C     T     225   PASS   DP=2… chr1…   203   130</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt;  7 chr1  2935… &lt;NA&gt;  C     T     225   PASS   DP=2… chr1…   228   132</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt;  8 chr1  2980… &lt;NA&gt;  C     T     225   PASS   DP=2… chr1…   196    85</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt;  9 chr1  3387… &lt;NA&gt;  T     G     -0    PASS   DP=1… chr1…   124     6</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; 10 chr1  3502… rs11… G     A     0.00… PASS   DP=9… chr1…    88    10</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; # … with 13,131 more rows, and 40 more variables: VAF &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; #   `/icgc/dkfzlsdf/project/hipo/hipo_043/sequencing/whole_genome_sequencing/view-by-pid/H043-W99H5K/tumor/paired/merged-alignment/.merging_0/tumor_H043-W99H5K_merged.mdup.bam` &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#&gt; #   SEQUENCE_CONTEXT &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#&gt; #   `INFO_control(VAF=variant_allele_fraction;TSR=total_variant_supporting_reads_incl_lowqual)` &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">#&gt; #   ANNOTATION_control &lt;chr&gt;, DBSNP &lt;chr&gt;, `1K_GENOMES` &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">#&gt; #   ANNOVAR_FUNCTION &lt;chr&gt;, GENE &lt;chr&gt;, EXONIC_CLASSIFICATION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">#&gt; #   ANNOVAR_TRANSCRIPTS &lt;chr&gt;, SEGDUP &lt;chr&gt;, CYTOBAND &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">#&gt; #   REPEAT_MASKER &lt;chr&gt;, DAC_BLACKLIST &lt;chr&gt;, DUKE_EXCLUDED &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt; #   HISEQDEPTH &lt;chr&gt;, SELFCHAIN &lt;chr&gt;, MAPABILITY &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt; #   SIMPLE_TANDEMREPEATS &lt;chr&gt;, CONFIDENCE &lt;chr&gt;, RECLASSIFICATION &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt; #   PENALTIES &lt;chr&gt;, seqBiasPresent_1 &lt;chr&gt;, seqingBiasPresent_1 &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"><span class="co">#&gt; #   seqBiasPresent_2 &lt;chr&gt;, seqingBiasPresent_2 &lt;chr&gt;, Enhancers &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-29" data-line-number="29"><span class="co">#&gt; #   CpGislands &lt;chr&gt;, TFBScons &lt;chr&gt;, ENCODE_DNASE &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-30" data-line-number="30"><span class="co">#&gt; #   miRNAs_snoRNAs &lt;chr&gt;, miRBase18 &lt;chr&gt;, COSMIC &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-31" data-line-number="31"><span class="co">#&gt; #   miRNAtargets &lt;chr&gt;, CgiMountains &lt;chr&gt;, phastConsElem20bp &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-32" data-line-number="32"><span class="co">#&gt; #   ENCODE_TFBS &lt;chr&gt;, blacklist &lt;chr&gt;, to &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb2-33" data-line-number="33"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-34" data-line-number="34"><span class="co">#&gt; $cna</span></a>
<a class="sourceLine" id="cb2-35" data-line-number="35"><span class="co">#&gt; # A tibble: 482 x 19</span></a>
<a class="sourceLine" id="cb2-36" data-line-number="36"><span class="co">#&gt;    chr     from     to length covRatio   TCN SV.Type c1Mean c2Mean  dhEst</span></a>
<a class="sourceLine" id="cb2-37" data-line-number="37"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;    &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb2-38" data-line-number="38"><span class="co">#&gt;  1 chr1  8.40e5 1.69e6 8.50e5     1.19  5.05 &lt;NA&gt;      2.02   3.03  0.199</span></a>
<a class="sourceLine" id="cb2-39" data-line-number="39"><span class="co">#&gt;  2 chr1  1.69e6 1.82e6 1.25e5     1.26  5.38 &lt;NA&gt;      2.19   3.19  0.185</span></a>
<a class="sourceLine" id="cb2-40" data-line-number="40"><span class="co">#&gt;  3 chr1  1.82e6 9.80e6 7.98e6     1.19  5.05 &lt;NA&gt;      2.02   3.03  0.199</span></a>
<a class="sourceLine" id="cb2-41" data-line-number="41"><span class="co">#&gt;  4 chr1  9.80e6 9.87e6 7.50e4     1.26  5.38 &lt;NA&gt;     NA     NA    NA    </span></a>
<a class="sourceLine" id="cb2-42" data-line-number="42"><span class="co">#&gt;  5 chr1  9.87e6 1.05e7 6.05e5     1.24  5.29 &lt;NA&gt;     NA     NA    NA    </span></a>
<a class="sourceLine" id="cb2-43" data-line-number="43"><span class="co">#&gt;  6 chr1  1.05e7 1.21e7 1.60e6     1.19  5.05 &lt;NA&gt;      2.02   3.03  0.199</span></a>
<a class="sourceLine" id="cb2-44" data-line-number="44"><span class="co">#&gt;  7 chr1  1.21e7 1.22e7 7.51e4     1.24  5.29 &lt;NA&gt;      2.04   3.25  0.229</span></a>
<a class="sourceLine" id="cb2-45" data-line-number="45"><span class="co">#&gt;  8 chr1  1.22e7 1.28e7 6.85e5     1.19  5.05 &lt;NA&gt;      2.02   3.03  0.199</span></a>
<a class="sourceLine" id="cb2-46" data-line-number="46"><span class="co">#&gt;  9 chr1  1.38e7 1.78e7 4.01e6     1.19  5.05 &lt;NA&gt;      2.02   3.03  0.199</span></a>
<a class="sourceLine" id="cb2-47" data-line-number="47"><span class="co">#&gt; 10 chr1  1.78e7 1.78e7 5.99e4     1.26  5.38 &lt;NA&gt;     NA     NA    NA    </span></a>
<a class="sourceLine" id="cb2-48" data-line-number="48"><span class="co">#&gt; # … with 472 more rows, and 9 more variables: dhSNPs &lt;dbl&gt;, Major &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-49" data-line-number="49"><span class="co">#&gt; #   minor &lt;chr&gt;, CNA.type &lt;fct&gt;, NbrOfHetsSNPs &lt;int&gt;, minStart &lt;int&gt;,</span></a>
<a class="sourceLine" id="cb2-50" data-line-number="50"><span class="co">#&gt; #   maxStart &lt;int&gt;, minStop &lt;int&gt;, maxStop &lt;int&gt;</span></a>
<a class="sourceLine" id="cb2-51" data-line-number="51"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-52" data-line-number="52"><span class="co">#&gt; $purity</span></a>
<a class="sourceLine" id="cb2-53" data-line-number="53"><span class="co">#&gt; [1] 0.89</span></a></code></pre></div>
<p>To sue the package, you need to initialize a <code>cnaqc</code>object mapping mutations to chromosomes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Use SNVs, CNAs and tumour purity </span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">x =<span class="st"> </span>CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/init">init</a></span>(</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  example_dataset_CNAqc<span class="op">$</span>snvs, </a>
<a class="sourceLine" id="cb3-4" data-line-number="4">  example_dataset_CNAqc<span class="op">$</span>cna,</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">  example_dataset_CNAqc<span class="op">$</span>purity)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; Warning in enforce_numeric(x$from): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; 1027104, 2248588, 2461999, 2727935, 2763397, 2768208, ...</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; Warning in enforce_numeric(x$Major): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; 3, 3, 3, NA, NA, 3, ...</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; Warning in enforce_numeric(x$Major): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt; Warning in enforce_numeric(x$minor): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt; 2, 2, 2, NA, NA, 2, ...</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; Warning in enforce_numeric(x$minor): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; Input  n = 13141 mutations for 482 CNA segments (482 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt;  Mapping mutations to clonal CNA.</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; Mapping. n = 13050 mutations mapped to segments (~99% of input) </span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; [CNAqc] The following segments have Major/ minor alleles in non-numeric format or NAs</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; # A tibble: 215 x 20</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt;    chr     from     to length covRatio   TCN SV.Type c1Mean c2Mean dhEst</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt;  1 chr1  9.80e6 9.87e6  75009     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt;  2 chr1  9.87e6 1.05e7 604933     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt;  3 chr1  1.78e7 1.78e7  59936     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt;  4 chr1  3.09e7 3.09e7  14969     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt;  5 chr1  3.14e7 3.16e7 179994     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt;  6 chr1  3.45e7 3.45e7  20049     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt;  7 chr1  5.04e7 5.04e7  44976     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt;  8 chr1  5.19e7 5.19e7  24947     1.27  5.43 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt;  9 chr1  6.13e7 6.13e7  39974     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">#&gt; 10 chr1  6.84e7 6.84e7  20002     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt; # … with 205 more rows, and 10 more variables: dhSNPs &lt;dbl&gt;, Major &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt; #   minor &lt;dbl&gt;, CNA.type &lt;fct&gt;, NbrOfHetsSNPs &lt;int&gt;, minStart &lt;int&gt;,</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; #   maxStart &lt;int&gt;, minStop &lt;int&gt;, maxStop &lt;int&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt; These CNA segments will be removed.</span></a></code></pre></div>
<p>The summary of the object is available via the S3 <code>print</code> function, reporting some basic statistcs of the data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt;       CNAqc  n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt;      Purity  89 cellularity </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;  Karyotypes  7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0); 67 (NA:NA); 16 (NA:2); 4 (104:NA); 4 (16:2); 2 (25:2); 2 (3:1); 1 (106:1); 1 (26:2); 1 (99:1) </span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;    Peaks QC  FALSE</span></a></code></pre></div>
<p>To view the segments, the tool maps CNA calls to the hg19 human genome coordinates and plots the major and minor alleles for each segment. The returned <code>ggplot</code> object can be modified.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Default</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-5-1.png" width="912"></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co"># Reduce y-axis</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">ylim</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-5-2.png" width="912"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co"># Reduce x-axis</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">xlim</a></span>(<span class="fl">1e6</span>, <span class="fl">50e6</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; will replace the existing scale.</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_vline).</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_label).</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; Warning: Removed 252 rows containing missing values (geom_segment).</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; Warning: Removed 252 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-5-3.png" width="912"></p>
<p>Genome-wide plots that follow the layout of <code>plot_segments</code> are avaiable to view the number of mutations, their VAF and depth. VAF and depth-plotting functions (scatterplots) can subset the input data to speed up rendering - by default 5000 points are randomly sampled. Counts plot bin locations by one megabase.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># All genome, all data - counts of mutations per megabase</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="../reference/plot_counts.html">plot_counts</a></span>(x)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># Different subsamples</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">1000</span>),</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x),</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">  <span class="dt">ncol =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co"># Different subsamples</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">1000</span>),</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x),</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">  <span class="dt">ncol =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-7-2.png" width="768"></p>
<p>A one-summary plot can be easily assembled exploiting the above functions and functions from <code>cowplot</code> or other ggplot-manipulation packages.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_counts.html">plot_counts</a></span>(x),</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">ylim</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">6</span>),</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">  <span class="dt">align =</span> <span class="st">'v'</span>, </a>
<a class="sourceLine" id="cb11-7" data-line-number="7">  <span class="dt">nrow =</span> <span class="dv">4</span>, </a>
<a class="sourceLine" id="cb11-8" data-line-number="8">  <span class="dt">rel_heights =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(.<span class="dv">15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span>))</a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-8-1.png" width="912"></p>
<p>The total number of</p>
<p>For CNAs, it is possible to show the segments per karyotypes, reporting either their counts, or the proportion of genome covered.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">  CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/plot_karyotypes">plot_karyotypes</a></span>(x),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/plot_karyotypes">plot_karyotypes</a></span>(x, <span class="dt">type =</span> <span class="st">'number'</span>),</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">  <span class="dt">common.legend =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">  <span class="dt">legend =</span> <span class="st">'bottom'</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-9-1.png" width="384"></p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
