<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • CNAqc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/CNAqc.Rmd"><code>vignettes/CNAqc.Rmd</code></a></small>
      <div class="hidden name"><code>CNAqc.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt;  [ CNAqc - Copy Number Alteration quality check ] </span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/CNAqc</span></a></code></pre></div>
<p>To set-up your data and work with the <code>CNAqc</code> package, you need to have:</p>
<ul>
<li>mutation calls;</li>
<li>CNA segments;</li>
<li>tumor purity estimates;</li>
</ul>
<p>The workflow of the package is simple:</p>
<ol style="list-style-type: decimal">
<li>load input data into the package;</li>
<li>use package functions to inspiect the data, and run QC analyses;</li>
<li>visualize and export QC results.</li>
</ol>
<p>QC results can be used to determine if the estimated purity and CNA fits to the observed mutation data are good. If they are off, QC results can be used to adjust the actual set of calls (e.g., adjusting the purity by a certain factor).</p>
<div id="input-format" class="section level1">
<h1 class="hasAnchor">
<a href="#input-format" class="anchor"></a>Input format</h1>
<p><code>CNAqc</code> uses <code>hg19</code> chromosome coordinates to map mutations and CNAs. The package comes with a template dataset available.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Load template data</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">'example_dataset_CNAqc'</span>, <span class="dt">package =</span> <span class="st">'CNAqc'</span>)</a></code></pre></div>
<p>Input mutations and CNA must be a <code>data.frame</code>; both of them are required to have certain columns.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Example input SNVs</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  example_dataset_CNAqc<span class="op">$</span>snvs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="st">        </span><span class="kw">select</span>(</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">          chr, from, to, <span class="co"># Genomic coordinates</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">          ref, alt,      <span class="co"># Alleles (reference and alternative)</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7">          DP, NV, VAF    <span class="co"># Read counts (depth, number of variant reads, tumour VAF)</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">          )</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">  )</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; # A tibble: 13,141 x 8</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;    chr   from         to ref   alt      DP    NV    VAF</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  1 chr1  1027104 1027105 T     G        60     6 0.1   </span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  2 chr1  2248588 2248589 A     C       127     9 0.0709</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt;  3 chr1  2461999 2462000 G     A       156    65 0.417 </span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt;  4 chr1  2727935 2727936 T     C       180    90 0.5   </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt;  5 chr1  2763397 2763398 C     T       183    61 0.333 </span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt;  6 chr1  2768208 2768209 C     T       203   130 0.640 </span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt;  7 chr1  2935590 2935591 C     T       228   132 0.579 </span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt;  8 chr1  2980032 2980033 C     T       196    85 0.434 </span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt;  9 chr1  3387161 3387162 T     G       124     6 0.0484</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; 10 chr1  3502517 3502518 G     A        88    10 0.114 </span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; # … with 13,131 more rows</span></a></code></pre></div>
<p>Chromosome names and alleles should be in character format; chromosomes must be in the format <code>chr1</code>, <code>chr2</code>, etc.. <code>CNAqc</code> will use all available mutation types but we suggest to carry out QC with high-quality SNV data.</p>
<p>CNA calls must be in <em>absolute</em> format, reporting the allelic imbalance of each segment.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Example input CNA</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  example_dataset_CNAqc<span class="op">$</span>cna <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">        </span><span class="kw">select</span>(</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">          chr, from, to, <span class="co"># Genomic coordinates</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">          Major, minor,  <span class="co"># Number of copies of major/ and minor allele (B-allele)</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7">        )</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">  )</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; # A tibble: 482 x 5</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;    chr       from       to Major minor</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;    &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;chr&gt; &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;  1 chr1    840009  1689987 3     2    </span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;  2 chr1   1689988  1815015 3     2    </span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;  3 chr1   1815016  9799969 3     2    </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;  4 chr1   9799969  9874977 &lt;NA&gt;  &lt;NA&gt; </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt;  5 chr1   9874977 10479909 &lt;NA&gt;  &lt;NA&gt; </span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt;  6 chr1  10479910 12079917 3     2    </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt;  7 chr1  12079917 12154980 3     2    </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt;  8 chr1  12154981 12839977 3     2    </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt;  9 chr1  13780016 17790026 3     2    </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; 10 chr1  17790026 17849961 &lt;NA&gt;  &lt;NA&gt; </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt; # … with 472 more rows</span></a></code></pre></div>
<p>In this format, a diploid segment will have <code>Major = minor = 1</code> and will be denoted as <code>1:1</code>. <code>CNAqc</code> distinguishes between clonal and subclonal CNA calls, if the Cancer Cell Fraction of a segment (i.e., its clonality) is reported.</p>
<p>A <code>CCF</code> column can represent the segment CCF as [0,1] real-value. If missing, <code>CCF = 1</code> is set assuming clonal segments; otherwise, segments with <code>CCF&lt;1</code> are considered subclonal CNAs.</p>
<p>To analyze the data, a purity value in between <code>0</code> and <code>1</code> is required to represent the proportion of actual tumour content, sometimes called “cellularity”.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Example input purity</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(example_dataset_CNAqc<span class="op">$</span>purity)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; [1] 0.89</span></a></code></pre></div>
</div>
<div id="initialization" class="section level1">
<h1 class="hasAnchor">
<a href="#initialization" class="anchor"></a>Initialization</h1>
<p>To sue the package, you need to initialize a <code>cnaqc</code>object mapping mutations to chromosomes.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Use SNVs, CNAs and tumour purity </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">x =<span class="st"> </span>CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/init">init</a></span>(</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  example_dataset_CNAqc<span class="op">$</span>snvs, </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">  example_dataset_CNAqc<span class="op">$</span>cna,</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">  example_dataset_CNAqc<span class="op">$</span>purity)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; Warning in enforce_numeric(x$from): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; 1027104, 2248588, 2461999, 2727935, 2763397, 2768208, ...</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; [CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; Warning in enforce_numeric(x$Major): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; 3, 3, 3, NA, NA, 3, ...</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; Warning in enforce_numeric(x$Major): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; Warning in enforce_numeric(x$minor): [CNAqc] Enforcing numeric for values:</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; 2, 2, 2, NA, NA, 2, ...</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; Warning in enforce_numeric(x$minor): NAs introduced by coercion</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; Input  n = 13141 mutations for 482 CNA segments (482 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt;  Mapping mutations to clonal CNA.</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; Mapping. n = 13050 mutations mapped to segments (~99% of input) </span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; [CNAqc] The following segments have Major/ minor alleles in non-numeric format or NAs</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; # A tibble: 215 x 20</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt;    chr     from     to length covRatio   TCN SV.Type c1Mean c2Mean dhEst</span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt;  1 chr1  9.80e6 9.87e6  75009     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt;  2 chr1  9.87e6 1.05e7 604933     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt;  3 chr1  1.78e7 1.78e7  59936     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt;  4 chr1  3.09e7 3.09e7  14969     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt;  5 chr1  3.14e7 3.16e7 179994     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt;  6 chr1  3.45e7 3.45e7  20049     1.26  5.38 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt;  7 chr1  5.04e7 5.04e7  44976     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt;  8 chr1  5.19e7 5.19e7  24947     1.27  5.43 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-34" data-line-number="34"><span class="co">#&gt;  9 chr1  6.13e7 6.13e7  39974     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-35" data-line-number="35"><span class="co">#&gt; 10 chr1  6.84e7 6.84e7  20002     1.24  5.29 &lt;NA&gt;        NA     NA    NA</span></a>
<a class="sourceLine" id="cb6-36" data-line-number="36"><span class="co">#&gt; # … with 205 more rows, and 10 more variables: dhSNPs &lt;dbl&gt;, Major &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb6-37" data-line-number="37"><span class="co">#&gt; #   minor &lt;dbl&gt;, CNA.type &lt;fct&gt;, NbrOfHetsSNPs &lt;int&gt;, minStart &lt;int&gt;,</span></a>
<a class="sourceLine" id="cb6-38" data-line-number="38"><span class="co">#&gt; #   maxStart &lt;int&gt;, minStop &lt;int&gt;, maxStop &lt;int&gt;, CCF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-39" data-line-number="39"><span class="co">#&gt; These CNA segments will be removed.</span></a></code></pre></div>
<p>The summary of the object is available via the S3 <code>print</code> function, reporting some basic statistcs of the data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(x)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;       CNAqc  n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal) </span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt;      Purity  89 cellularity </span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt;  Karyotypes  7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0); 67 (NA:NA); 16 (NA:2); 4 (104:NA); 4 (16:2); 2 (25:2); 2 (3:1); 1 (106:1); 1 (26:2); 1 (99:1) </span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt;    Peaks QC  FALSE</span></a></code></pre></div>
<p>To view the segments, the tool maps CNA calls to the hg19 human genome coordinates and plots the major and minor alleles for each segment. The returned <code>ggplot</code> object can be modified.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># Default</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-8-1.png" width="912"></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co"># Reduce y-axis</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">ylim</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-8-2.png" width="912"></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co"># Reduce x-axis</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">xlim</a></span>(<span class="fl">1e6</span>, <span class="fl">50e6</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; Scale for 'x' is already present. Adding another scale for 'x', which</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; will replace the existing scale.</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_rect).</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_vline).</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; Warning: Removed 24 rows containing missing values (geom_label).</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; Warning: Removed 252 rows containing missing values (geom_segment).</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt; Warning: Removed 252 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-8-3.png" width="912"></p>
<p>Genome-wide plots that follow the layout of <code>plot_segments</code> are avaiable to view the number of mutations, their VAF and depth. VAF and depth-plotting functions (scatterplots) can subset the input data to speed up rendering - by default 5000 points are randomly sampled. Counts plot bin locations by one megabase.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># All genome, all data - counts of mutations per megabase</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="../reference/plot_counts.html">plot_counts</a></span>(x)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-9-1.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># Different subsamples</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">1000</span>),</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x),</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">  <span class="dt">ncol =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co"># Different subsamples</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">1000</span>),</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x),</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">  <span class="dt">ncol =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-10-2.png" width="768"></p>
<p>A one-summary plot can be easily assembled exploiting the above functions and functions from <code>cowplot</code> or other ggplot-manipulation packages.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">  <span class="kw"><a href="../reference/plot_counts.html">plot_counts</a></span>(x),</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">  <span class="kw"><a href="../reference/plot_vaf.html">plot_vaf</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">  <span class="kw"><a href="../reference/plot_depth.html">plot_depth</a></span>(x, <span class="dt">N =</span> <span class="dv">10000</span>),</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">  <span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot.window">ylim</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">6</span>),</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">  <span class="dt">align =</span> <span class="st">'v'</span>, </a>
<a class="sourceLine" id="cb14-7" data-line-number="7">  <span class="dt">nrow =</span> <span class="dv">4</span>, </a>
<a class="sourceLine" id="cb14-8" data-line-number="8">  <span class="dt">rel_heights =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(.<span class="dv">15</span>, <span class="fl">.15</span>, <span class="fl">.15</span>, <span class="fl">.8</span>))</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt; Warning: Removed 10 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-11-1.png" width="912"></p>
<p>The total number of</p>
<p>For CNAs, it is possible to show the segments per karyotypes, reporting either their counts, or the proportion of genome covered.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">ggpubr<span class="op">::</span><span class="kw"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span>(</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">  CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/plot_karyotypes">plot_karyotypes</a></span>(x),</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">  CNAqc<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/CNAqc/topics/plot_karyotypes">plot_karyotypes</a></span>(x, <span class="dt">type =</span> <span class="st">'number'</span>),</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">  <span class="dt">common.legend =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">  <span class="dt">legend =</span> <span class="st">'bottom'</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6">)</a></code></pre></div>
<p><img src="CNAqc_files/figure-html/unnamed-chunk-12-1.png" width="384"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#input-format">Input format</a></li>
      <li><a href="#initialization">Initialization</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
