<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>6. Genome fragmentation analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="6. Genome fragmentation analysis">
<meta property="og:description" content="">
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/smoothing.html">5. Smoothing copy number segments</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>6. Genome fragmentation analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/fragmentation.Rmd"><code>vignettes/fragmentation.Rmd</code></a></small>
      <div class="hidden name"><code>fragmentation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></a></code></pre></div>
<p>We work with the <a href="https://caravagn.github.io/CNAqc/articles/Introduction.html">template dataset</a>.</p>
<div id="fragmentation-of-individual-arms" class="section level3">
<h3 class="hasAnchor">
<a href="#fragmentation-of-individual-arms" class="anchor"></a>Fragmentation of individual arms</h3>
<p>The fragmentation of a chromosome arm is assessed with a statistical test based on counting the size of the copy number segments mapping to the arm.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># A histogram of segments' lenght</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span>(x)</a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
<p><code>CNAqc</code> counts, for every arm with lenght <span class="math inline">\(L\)</span> nucleotides:</p>
<ul>
<li>
<span class="math inline">\(n_s\)</span>, the number of mapped CNA segments shorter than <span class="math inline">\(\delta\%\)</span> of <span class="math inline">\(L\)</span>;</li>
<li>
<span class="math inline">\(n_l\)</span>, the number of mapped CNA segments longer than <span class="math inline">\(\delta\%\)</span> sof <span class="math inline">\(L\)</span>.</li>
</ul>
<p>A one-sided Binomial test is used to compute a p-value for the null hypothesis of seeing <span class="math inline">\(n_s\)</span> observations in <span class="math inline">\(n_l\)</span> trials, assuming a Binomial success probability <span class="math inline">\(p = \delta &gt; 0\)</span>. <span class="math inline">\(p\)</span> represents a model where each segment length is equally likely (uniform distribution).</p>
<p>In this way the test accounts for the difference in lenghts of the chromsome arms; a p-value per arm is reported and adjusted for multiple hyoptheses (Bonferroni).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Test with default parameters (small segments: &lt; 20% of chromosome arm)</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">x =<span class="st"> </span><span class="kw"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span>(x)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; ℹ One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; ✓ chr7p,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; ✓ chr1p,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; ✓ chr11q,  p = 2.4788992e-07 ~ 14 segments, 12 short.</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; ✓ chr1q,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; ✓ chr12q,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; ✓ chr3q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; ✓ chr7q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; ✓ chr8p,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; ℹ 8 significantly overfragmented chromosome arms (alpha level 0.01).</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(x)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; ── [ CNAqc ]  n = 13141 mutations in 267 segments (267 clonal + 0 subclonal). Genome reference: GRCh38. ──────────────────</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt;    4:2  [n = 1893] ■■■■■■■</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt;    3:2  [n = 1625] ■■■■■■</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt;    2:1  [n = 1563] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt;    3:0  [n =  312] ■</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; ✓ Arm-level fragmentation analysis: 8 segments overfragmented.</span></a></code></pre></div>
<p>You can produce a arm-level report for the fragmentation test, with:</p>
<ul>
<li>a scatter of the counts per arm, with scaled the p-values;</li>
<li>a jump statistics per arm, <span class="math inline">\(J\)</span>.</li>
</ul>
<p><span class="math inline">\(J\)</span> is the sum of the variation in total copy number profiles, evaluated among each pair of contiguous segments.</p>
<p>Significantly overfragmented arms with high <span class="math inline">\(J\)</span> have a “scattered” copy number profile. Those with low <span class="math inline">\(J\)</span> are more uniform, as they show little no copy number change, and can be possibly smoothed (see below).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span>(x, <span class="dt">zoom =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; x Increare zoom to see more chromosomes.</span></a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-5-1.png" width="672"></p>
<p>Once available, these results appear in any call to <code>plot_segments</code> as annotated purple squares sorrounding the arms.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Default plot has now segments</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="../reference/plot_segments.html">plot_segments</a></span>(x)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; will replace the existing scale.</span></a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-6-1.png" width="912"></p>
<p><a href="https://caravagn.github.io/CNAqc/articles/smoothing.html">Smoothing</a> is a good way to start cleaning up the fragmented sets of arms, as it merges segments that have no karyotype change, and are closer than a certain distance.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Smooth with default parameters</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">x =<span class="st"> </span><span class="kw"><a href="../reference/smooth_segments.html">smooth_segments</a></span>(x)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; ℹ Smoothing chr1: 37 segments.</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; [1-24] [27-30] [31-33] [34-37]</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; ℹ Smoothing chr10: 8 segments.</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; [1-2] [3-4] [5-8]</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; ℹ Smoothing chr11: 22 segments.</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; [1-7] [8-9] [10-22]</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; ℹ Smoothing chr12: 13 segments.</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; [2-3] [12-13]</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; ℹ Smoothing chr14: 2 segments.</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; ℹ Smoothing chr15: 9 segments.</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; [1-5] [7-9]</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; ℹ Smoothing chr16: 10 segments.</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; [1-6] [8-10]</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; ℹ Smoothing chr17: 10 segments.</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; [4-5] [6-7] [8-10]</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; ℹ Smoothing chr18: 8 segments.</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; [1-2] [3-8]</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; ℹ Smoothing chr19: 5 segments.</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; [2-5]</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; ℹ Smoothing chr2: 18 segments.</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt; [1-8] [10-12] [14-18]</span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt; ℹ Smoothing chr20: 9 segments.</span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt; [1-3] [4-9]</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt; ℹ Smoothing chr21: 2 segments.</span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt; ℹ Smoothing chr22: 3 segments.</span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt; ℹ Smoothing chr3: 19 segments.</span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt; [1-7] [8-13] [15-19]</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt; ℹ Smoothing chr4: 8 segments.</span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt; [1-4] [5-8]</span></a>
<a class="sourceLine" id="cb6-34" data-line-number="34"><span class="co">#&gt; ℹ Smoothing chr5: 6 segments.</span></a>
<a class="sourceLine" id="cb6-35" data-line-number="35"><span class="co">#&gt; [1-2] [3-5]</span></a>
<a class="sourceLine" id="cb6-36" data-line-number="36"><span class="co">#&gt; ℹ Smoothing chr6: 4 segments.</span></a>
<a class="sourceLine" id="cb6-37" data-line-number="37"><span class="co">#&gt; [1-2] [3-4]</span></a>
<a class="sourceLine" id="cb6-38" data-line-number="38"><span class="co">#&gt; ℹ Smoothing chr7: 46 segments.</span></a>
<a class="sourceLine" id="cb6-39" data-line-number="39"><span class="co">#&gt; [1-5] [9-27] [36-38] [39-41] [43-46]</span></a>
<a class="sourceLine" id="cb6-40" data-line-number="40"><span class="co">#&gt; ℹ Smoothing chr8: 18 segments.</span></a>
<a class="sourceLine" id="cb6-41" data-line-number="41"><span class="co">#&gt; [1-4] [5-11] [12-18]</span></a>
<a class="sourceLine" id="cb6-42" data-line-number="42"><span class="co">#&gt; ℹ Smoothing chr9: 3 segments.</span></a>
<a class="sourceLine" id="cb6-43" data-line-number="43"><span class="co">#&gt; [1-2]</span></a>
<a class="sourceLine" id="cb6-44" data-line-number="44"><span class="co">#&gt; ℹ Smoothing chrX: 6 segments.</span></a>
<a class="sourceLine" id="cb6-45" data-line-number="45"><span class="co">#&gt; [1-2] [3-6]</span></a>
<a class="sourceLine" id="cb6-46" data-line-number="46"><span class="co">#&gt; ✓ Smoothed from 267 to 87 segments with 1e+06 gap ...</span></a>
<a class="sourceLine" id="cb6-47" data-line-number="47"><span class="co">#&gt; ℹ Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span></a>
<a class="sourceLine" id="cb6-48" data-line-number="48"><span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span></a>
<a class="sourceLine" id="cb6-49" data-line-number="49"><span class="co">#&gt; ℹ Using reference genome coordinates for: GRCh38.</span></a>
<a class="sourceLine" id="cb6-50" data-line-number="50"><span class="co">#&gt; ℹ Input n = 13141 mutations for 87 CNA segments (87 clonal, 0 subclonal)</span></a>
<a class="sourceLine" id="cb6-51" data-line-number="51"><span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span></a>
<a class="sourceLine" id="cb6-52" data-line-number="52"><span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span></a>
<a class="sourceLine" id="cb6-53" data-line-number="53"><span class="co">#&gt; ✓ Mapped n = 13025 mutations to clonal segments (99% of input)</span></a>
<a class="sourceLine" id="cb6-54" data-line-number="54"></a>
<a class="sourceLine" id="cb6-55" data-line-number="55"><span class="co"># Re-compute the fragmentation</span></a>
<a class="sourceLine" id="cb6-56" data-line-number="56">x =<span class="st"> </span><span class="kw"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span>(x)</a>
<a class="sourceLine" id="cb6-57" data-line-number="57"><span class="co">#&gt; ℹ One-tailed Binomial test: 2 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></a>
<a class="sourceLine" id="cb6-58" data-line-number="58"><span class="co">#&gt; ✓ chr7p,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb6-59" data-line-number="59"><span class="co">#&gt; ✓ chr12q,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></a>
<a class="sourceLine" id="cb6-60" data-line-number="60"><span class="co">#&gt; ℹ 2 significantly overfragmented chromosome arms (alpha level 0.01).</span></a>
<a class="sourceLine" id="cb6-61" data-line-number="61"></a>
<a class="sourceLine" id="cb6-62" data-line-number="62"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(x)</a>
<a class="sourceLine" id="cb6-63" data-line-number="63"><span class="co">#&gt; ── [ CNAqc ]  n = 13141 mutations in 87 segments (87 clonal + 0 subclonal). Genome reference: GRCh38. ────────────────────</span></a>
<a class="sourceLine" id="cb6-64" data-line-number="64"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-65" data-line-number="65"><span class="co">#&gt;    2:2  [n = 7503] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { chr19:15292495:15292496:G:A }</span></a>
<a class="sourceLine" id="cb6-66" data-line-number="66"><span class="co">#&gt;    4:2  [n = 1899] ■■■■■■■</span></a>
<a class="sourceLine" id="cb6-67" data-line-number="67"><span class="co">#&gt;    3:2  [n = 1643] ■■■■■■</span></a>
<a class="sourceLine" id="cb6-68" data-line-number="68"><span class="co">#&gt;    2:1  [n = 1571] ■■■■■■  { chr2:179431633:179431634:C:T }</span></a>
<a class="sourceLine" id="cb6-69" data-line-number="69"><span class="co">#&gt;    3:0  [n =  317] ■</span></a>
<a class="sourceLine" id="cb6-70" data-line-number="70"><span class="co">#&gt; ℹ Sample Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb6-71" data-line-number="71"><span class="co">#&gt; ℹ Mutations annotated have 3 drivers.</span></a>
<a class="sourceLine" id="cb6-72" data-line-number="72"><span class="co">#&gt; ✓ These segments are smoothed; before smoothing there were 267 segments.</span></a>
<a class="sourceLine" id="cb6-73" data-line-number="73"><span class="co">#&gt; ✓ Arm-level fragmentation analysis: 2 segments overfragmented.</span></a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span>(x, <span class="dt">zoom =</span> <span class="dv">6</span>)</a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
