<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genome fragmentation analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genome fragmentation analysis">
<meta property="og:description" content="">
<meta property="og:image" content="caravagn.github.io/revolver/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/ccf_computation.html">Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/fragmentation.html">Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/smoothing.html">Smoothing copy number segments</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Genome fragmentation analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/CNAqc/blob/master/vignettes/fragmentation.Rmd"><code>vignettes/fragmentation.Rmd</code></a></small>
      <div class="hidden name"><code>fragmentation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(CNAqc)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></a></code></pre></div>
<p>We work with the <a href="https://caravagn.github.io/CNAqc/articles/Introduction.html">template dataset</a>.</p>
<div id="fragmentation-of-individual-arms" class="section level3">
<h3 class="hasAnchor">
<a href="#fragmentation-of-individual-arms" class="anchor"></a>Fragmentation of individual arms</h3>
<p>The fragmentation of a chromosome arm is assessed with a statistical test based on counting the size of the copy number segments mapping to the arm.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># A histogram of segments' lenght</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span>(x)</a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
<p><code>CNAqc</code> counts, for every arm with lenght <span class="math inline">\(L\)</span> nucleotides:</p>
<ul>
<li>
<span class="math inline">\(n_s\)</span>, the number of mapped CNA segments shorter than <span class="math inline">\(\delta\%\)</span> of <span class="math inline">\(L\)</span>;</li>
<li>
<span class="math inline">\(n_l\)</span>, the number of mapped CNA segments longer than <span class="math inline">\(\delta\%\)</span> sof <span class="math inline">\(L\)</span>.</li>
</ul>
<p>A one-sided Binomial test is used to compute a p-value for the null hypothesis of seeing <span class="math inline">\(n_s\)</span> observations in <span class="math inline">\(n_l\)</span> trials, assuming a Binomial success probability <span class="math inline">\(p = \delta &gt; 0\)</span>. <span class="math inline">\(p\)</span> represents a model where each segment length is equally likely (uniform distribution).</p>
<p>In this way the test accounts for the difference in lenghts of the chromsome arms; a p-value per arm is reported and adjusted for multiple hyoptheses (Bonferroni).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Test with default parameters (small segments: &lt; 20% of chromosome arm)</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">x =<span class="st"> </span><span class="kw"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span>(x)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; ℹ One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; ✓ chr7p,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; ✓ chr1p,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; ✓ chr1q,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; ✓ chr11q,  p = 1.0657792e-06 ~ 13 segments, 11 short.</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; ✓ chr12q,  p = 2.00704e-07 ~ 12 segments, 11 short.</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; ✓ chr3q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; ✓ chr7q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; ✓ chr8p,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; ℹ 8 significantly overfragmented chromosome arms (alpha level 0.01).</span></a></code></pre></div>
<p>You can produce a arm-level report for the fragmentation test, with:</p>
<ul>
<li>a scatter of the counts per arm, with scaled the p-values;</li>
<li>a jump statistics per arm, <span class="math inline">\(J\)</span>.</li>
</ul>
<p><span class="math inline">\(J\)</span> is the sum of the variation in total copy number profiles, evaluated among each pair of contiguous segments.</p>
<p>Significantly overfragmented arms with high <span class="math inline">\(J\)</span> - for some custom minimum value – have a “scattered” copy number profile. Those with low <span class="math inline">\(J\)</span> are more uniform, as they show little no copy number change, and can be possibly smoothed (<a href="https://caravagn.github.io/CNAqc/articles/smoothing.html">see the Vignette</a>).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span>(x)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, :</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; Segments with CN above 5 will not be plot; this is annotated in the figure.</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, : No</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; segments with CN above 3, the plot is anyway scaled up to 5 to render better.</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, :</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; Segments with CN above 5 will not be plot; this is annotated in the figure.</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, : No</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; segments with CN above 3, the plot is anyway scaled up to 5 to render better.</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, : No</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; segments with CN above 3, the plot is anyway scaled up to 5 to render better.</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; Warning in plot_segments(x, max_Y_height = 5, chromosomes = chromosomes, : No</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; segments with CN above 3, the plot is anyway scaled up to 5 to render better.</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; Warning: Transformation introduced infinite values in continuous y-axis</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; Warning: Removed 32 rows containing missing values (geom_bar).</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt; Warning: Removed 8 rows containing missing values (geom_segment).</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; Warning: Removed 2 rows containing missing values (geom_segment).</span></a></code></pre></div>
<p><img src="fragmentation_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
</div>
<div id="whole-genome-fragmentation" class="section level3">
<h3 class="hasAnchor">
<a href="#whole-genome-fragmentation" class="anchor"></a>Whole-genome fragmentation</h3>
<p>With the same principle a similar test can be carried out on the whole genome copy number profiles. Pooling segments shorter than <span class="math inline">\(\Delta\)</span> bases – default 1 megabase – and fixing a value for the Binomial probability <span class="math inline">\(p\)</span> – default <span class="math inline">\(0.2\)</span>.</p>
<p>Compared to the arm-level test where we can adjust for arm lengths, in this case using the <span class="math inline">\(\Delta\)</span> normalised by genome length leads to low values for <span class="math inline">\(p\)</span> that seem in practice unrealistic.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">x =<span class="st"> </span><span class="kw"><a href="../reference/detect_wg_overfragmentation.html">detect_wg_overfragmentation</a></span>(x)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt; ℹ One-tailed Binomial test, whole genome fragments with cutoff 1e+06 bases: 84 short, 183 long.</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; ✓ wg,  p = 6.44688822866778e-06 ~ 267 segments, 84 short.</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; ! The whole genome is overfragmented with alpha = 0.05; p = 6.44688822866778e-06</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(x)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; ── [ CNAqc ]  n = 13141 mutations in 267 segments (267 clonal + 0 subclonal) ──────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; ℹ Purity: 89% ~ Ploidy: 4.</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; ℹ Mutation mapping (head): 7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0)</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; ✓ Arm-level fragmentation analysis: 8 segments overfragmented.</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; ✓ Whole-genome fragmentation analysis: p = 1e-05: overfragmented.</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
