<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Analyze CNA calls by peak detection. — analyze_peaks • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Analyze CNA calls by peak detection. — analyze_peaks" />
<meta property="og:description" content="This function carries out a peak-detection
analysis based on KDE and the package peakPick
in order to determine if the mutations that map to a
certain karyotype have an allelic frequency that is
consistent with the Major and minor alleles of the
segment, and the tumour purity. A score is produced
as a linear combination of the distance of the actual
peak to the expected one, derived with standard equations
for CNA analysis." />
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/fragmentation.html">Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/smoothing.html">5. Smoothing copy number segments</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Analyze CNA calls by peak detection.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagn/CNAqc/blob/master/R/analyze_peaks.R'><code>R/analyze_peaks.R</code></a></small>
    <div class="hidden name"><code>analyze_peaks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function carries out a peak-detection
analysis based on KDE and the package <code>peakPick</code>
in order to determine if the mutations that map to a
certain karyotype have an allelic frequency that is
consistent with the Major and minor alleles of the
segment, and the tumour purity. A score is produced
as a linear combination of the distance of the actual
peak to the expected one, derived with standard equations
for CNA analysis.</p>
    </div>

    <pre class="usage"><span class='fu'>analyze_peaks</span>(<span class='no'>x</span>, <span class='kw'>karyotypes</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"1:0"</span>, <span class='st'>"1:1"</span>, <span class='st'>"2:1"</span>, <span class='st'>"2:0"</span>, <span class='st'>"2:2"</span>),
  <span class='kw'>min_karyotype_size</span> <span class='kw'>=</span> <span class='fl'>0.05</span>, <span class='kw'>p_binsize_peaks</span> <span class='kw'>=</span> <span class='fl'>0.005</span>,
  <span class='kw'>matching_epsilon</span> <span class='kw'>=</span> <span class='fl'>0.025</span>, <span class='kw'>kernel_adjust</span> <span class='kw'>=</span> <span class='fl'>1</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p></td>
    </tr>
    <tr>
      <th>karyotypes</th>
      <td><p>The list of karyotypes to test. By default LOH regions (A, AA),
diploid regions (AB), and amplification regions (AAB, AABB). These correspond to
<code>'1:0', '1:1', '2:1', '2:0', '2:2'</code> in "Major:minor" notation.</p></td>
    </tr>
    <tr>
      <th>min_karyotype_size</th>
      <td><p>Karyotypes are subset by their size (normalized for
the number of input mutations). Karyotypes smaller than this value are removed
from analysis.</p></td>
    </tr>
    <tr>
      <th>p_binsize_peaks</th>
      <td><p>Peaks detected will be filtered if, in a peak, we map
less than <code>p_binsize_peaks * N</code> mutations. The value <code>N</code> is obtained
couting all mutations that map in all peaks.</p></td>
    </tr>
    <tr>
      <th>matching_epsilon</th>
      <td><p>Peaks at a location are matched with a range of <code>epsilon * 0.5</code>,
left and right. By default (<code>0.025</code>) a maxium 5% tolerance is adopted. The status of
overall PASS/ FAIL for this analysis is determined with the same tolerance, given the score
computed across all karyotypes.</p></td>
    </tr>
    <tr>
      <th>kernel_adjust</th>
      <td><p>KDE adjust density parameter; see <code>density</code>. A Gaussian
kernel is used (<code>kernel = 'gaussian'</code>).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cnaqc</code>, modified to hold the results from this analysis.
See the vignette to see how to extract and plot the results.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>'example_dataset_CNAqc'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'CNAqc'</span>)
<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'><a href='init.html'>init</a></span>(<span class='no'>example_dataset_CNAqc</span>$<span class='no'>snvs</span>, <span class='no'>example_dataset_CNAqc</span>$<span class='no'>cna</span>,<span class='no'>example_dataset_CNAqc</span>$<span class='no'>purity</span>)</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 12963 mutations to clonal segments (99% of input)</span></div><div class='input'>
<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'>analyze_peaks</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Requested karyotypes </span><span style='color: #00BB00;'>1:0</span><span>, </span><span style='color: #00BB00;'>1:1</span><span>, </span><span style='color: #00BB00;'>2:1</span><span>, </span><span style='color: #00BB00;'>2:0</span><span> and </span><span style='color: #00BB00;'>2:2</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Found n = 9041 mutations in 2:2, 2:1 (skipping those with n &lt; 657 mutations).</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 4 x 12</span><span>
#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset
#&gt;              </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>        </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>                2 2:1       0.616  0.61  2.86             45 FALSE      0.005</span><span style='text-decoration: underline;'>92</span><span>
#&gt; </span><span style='color: #555555;'>2</span><span>                1 2:1       0.308  0.31  3.73             72 FALSE     -</span><span style='color: #BB0000;'>0.002</span><span style='color: #BB0000;text-decoration: underline;'>04</span><span>
#&gt; </span><span style='color: #555555;'>3</span><span>                2 2:2       0.471  0.46  8.16            612 FALSE      0.010</span><span style='text-decoration: underline;'>9</span><span> 
#&gt; </span><span style='color: #555555;'>4</span><span>                1 2:2       0.235  0.25  0.75             70 FALSE     -</span><span style='color: #BB0000;'>0.014</span><span style='color: #BB0000;text-decoration: underline;'>6</span><span> 
#&gt; </span><span style='color: #555555;'># … with 4 more variables: matched </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>, weight </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, score </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, QC </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Peak detection </span><span style='color: #00BB00;'>PASS</span><span> with </span><span style='color: #00BB00;'>r = -0.00117483319779741</span><span> and tolerance e = 0.05</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 13141 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Genome reference: </span><span style='color: #00BB00;'>GRCh38</span><span>. ──────────────────</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>chr19:15292495:15292496:G:A</span><span> }
#&gt;    4:2  [n = 1893] ■■■■■■■
#&gt;    3:2  [n = 1625] ■■■■■■
#&gt;    2:1  [n = 1563] ■■■■■■  { </span><span style='color: #BBBB00;'>chr2:179431633:179431634:C:T</span><span> }
#&gt;    3:0  [n =  312] ■
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mutations annotated have 3 drivers.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> QC via peak detection available, score: -0.00117483319779741.</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 4 x 12</span><span>
#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset
#&gt;              </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>        </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>                2 2:1       0.616  0.61  2.86             45 FALSE      0.005</span><span style='text-decoration: underline;'>92</span><span>
#&gt; </span><span style='color: #555555;'>2</span><span>                1 2:1       0.308  0.31  3.73             72 FALSE     -</span><span style='color: #BB0000;'>0.002</span><span style='color: #BB0000;text-decoration: underline;'>04</span><span>
#&gt; </span><span style='color: #555555;'>3</span><span>                2 2:2       0.471  0.46  8.16            612 FALSE      0.010</span><span style='text-decoration: underline;'>9</span><span> 
#&gt; </span><span style='color: #555555;'>4</span><span>                1 2:2       0.235  0.25  0.75             70 FALSE     -</span><span style='color: #BB0000;'>0.014</span><span style='color: #BB0000;text-decoration: underline;'>6</span><span> 
#&gt; </span><span style='color: #555555;'># … with 4 more variables: matched </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>, weight </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, score </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, QC </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>$<span class='no'>peaks_analysis</span>)</div><div class='output co'>#&gt; $score
#&gt; [1] -0.001174833
#&gt; 
#&gt; $fits
#&gt; $fits$`1:0`
#&gt; $fits$`1:0`$matching
#&gt; NULL
#&gt; 
#&gt; 
#&gt; $fits$`1:1`
#&gt; $fits$`1:1`$matching
#&gt; NULL
#&gt; 
#&gt; 
#&gt; $fits$`2:1`
#&gt; $fits$`2:1`$matching
#&gt; </span><span style='color: #555555;'># A tibble: 2 x 10</span><span>
#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset
#&gt;              </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>        </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>                2 2:1       0.616  0.61  2.86             45 FALSE      0.005</span><span style='text-decoration: underline;'>92</span><span>
#&gt; </span><span style='color: #555555;'>2</span><span>                1 2:1       0.308  0.31  3.73             72 FALSE     -</span><span style='color: #BB0000;'>0.002</span><span style='color: #BB0000;text-decoration: underline;'>04</span><span>
#&gt; </span><span style='color: #555555;'># … with 2 more variables: matched </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>, weight </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; 
#&gt; $fits$`2:1`$density
#&gt; 
#&gt; Call:
#&gt; 	density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")
#&gt; 
#&gt; Data: y (1563 obs.);	Bandwidth 'bw' = 0.03344
#&gt; 
#&gt;        x                  y          
#&gt;  Min.   :-0.05113   Min.   :0.00017  
#&gt;  1st Qu.: 0.17683   1st Qu.:0.13437  
#&gt;  Median : 0.40480   Median :0.54953  
#&gt;  Mean   : 0.40480   Mean   :1.09559  
#&gt;  3rd Qu.: 0.63276   3rd Qu.:1.98170  
#&gt;  Max.   : 0.86073   Max.   :3.73431  
#&gt; 
#&gt; $fits$`2:1`$xy_peaks
#&gt; </span><span style='color: #555555;'># A tibble: 2 x 4</span><span>
#&gt;       x     y counts_per_bin discarded
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>    
#&gt; </span><span style='color: #555555;'>1</span><span>  0.31  3.73             72 FALSE    
#&gt; </span><span style='color: #555555;'>2</span><span>  0.61  2.86             45 FALSE    
#&gt; 
#&gt; 
#&gt; $fits$`2:0`
#&gt; $fits$`2:0`$matching
#&gt; NULL
#&gt; 
#&gt; 
#&gt; $fits$`2:2`
#&gt; $fits$`2:2`$matching
#&gt; </span><span style='color: #555555;'># A tibble: 2 x 10</span><span>
#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded  offset
#&gt;              </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>                2 2:2       0.471  0.46  8.16            612 FALSE      0.010</span><span style='text-decoration: underline;'>9</span><span>
#&gt; </span><span style='color: #555555;'>2</span><span>                1 2:2       0.235  0.25  0.75             70 FALSE     -</span><span style='color: #BB0000;'>0.014</span><span style='color: #BB0000;text-decoration: underline;'>6</span><span>
#&gt; </span><span style='color: #555555;'># … with 2 more variables: matched </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>, weight </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; 
#&gt; $fits$`2:2`$density
#&gt; 
#&gt; Call:
#&gt; 	density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")
#&gt; 
#&gt; Data: y (7478 obs.);	Bandwidth 'bw' = 0.00801
#&gt; 
#&gt;        x                 y           
#&gt;  Min.   :0.02108   Min.   :0.000000  
#&gt;  1st Qu.:0.27182   1st Qu.:0.008399  
#&gt;  Median :0.52256   Median :0.181167  
#&gt;  Mean   :0.52256   Mean   :0.996088  
#&gt;  3rd Qu.:0.77329   3rd Qu.:0.717005  
#&gt;  Max.   :1.02403   Max.   :8.156854  
#&gt; 
#&gt; $fits$`2:2`$xy_peaks
#&gt; </span><span style='color: #555555;'># A tibble: 9 x 4</span><span>
#&gt;       x     y counts_per_bin discarded
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>    
#&gt; </span><span style='color: #555555;'>1</span><span>  0.11  0.24             16 TRUE     
#&gt; </span><span style='color: #555555;'>2</span><span>  0.21  0.85             57 FALSE    
#&gt; </span><span style='color: #555555;'>3</span><span>  0.25  0.75             70 FALSE    
#&gt; </span><span style='color: #555555;'>4</span><span>  0.46  8.16            612 FALSE    
#&gt; </span><span style='color: #555555;'>5</span><span>  0.67  0.01              0 TRUE     
#&gt; </span><span style='color: #555555;'>6</span><span>  0.73  0.02              0 TRUE     
#&gt; </span><span style='color: #555555;'>7</span><span>  0.85  0.01              1 TRUE     
#&gt; </span><span style='color: #555555;'>8</span><span>  0.91  0.02              2 TRUE     
#&gt; </span><span style='color: #555555;'>9</span><span>  0.97  0.03              0 TRUE     
#&gt; 
#&gt; 
#&gt; 
#&gt; $matches
#&gt; </span><span style='color: #555555;'># A tibble: 4 x 12</span><span>
#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset
#&gt;              </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>          </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span>        </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>                2 2:1       0.616  0.61  2.86             45 FALSE      0.005</span><span style='text-decoration: underline;'>92</span><span>
#&gt; </span><span style='color: #555555;'>2</span><span>                1 2:1       0.308  0.31  3.73             72 FALSE     -</span><span style='color: #BB0000;'>0.002</span><span style='color: #BB0000;text-decoration: underline;'>04</span><span>
#&gt; </span><span style='color: #555555;'>3</span><span>                2 2:2       0.471  0.46  8.16            612 FALSE      0.010</span><span style='text-decoration: underline;'>9</span><span> 
#&gt; </span><span style='color: #555555;'>4</span><span>                1 2:2       0.235  0.25  0.75             70 FALSE     -</span><span style='color: #BB0000;'>0.014</span><span style='color: #BB0000;text-decoration: underline;'>6</span><span> 
#&gt; </span><span style='color: #555555;'># … with 4 more variables: matched </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>, weight </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, score </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, QC </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; $min_karyotype_size
#&gt; [1] 0.05
#&gt; 
#&gt; $p_binsize_peaks
#&gt; [1] 0.005
#&gt; 
#&gt; $matching_epsilon
#&gt; [1] 0.025
#&gt; 
#&gt; $QC
#&gt; [1] "PASS"
#&gt; </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://sites.google.com/site/giuliocaravagna/'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


