<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compute CCF values for the available mutations. — compute_CCF • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Compute CCF values for the available mutations. — compute_CCF" />
<meta property="og:description" content="This function provides the implementation of a set of entropy-related routines that can estimate
Cancer Cell Fraction values (CCFs) for the available mutations. The implemented routine is described
in the package vignette `&quot;Computation of Cancer Cell Fractions&quot;` that is available at the URL
https://caravagn.github.io/CNAqc/articles/ccf_computation.html. This function creates a field
`CCF_estimates` inside the returned object which contains both the estimated CCF values and the
plot of the report of this analysis." />
<meta property="og:image" content="caravagn.github.io/revolver/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/ccf_computation.html">Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/fragmentation.html">Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/smoothing.html">Smoothing copy number segments</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagn/CNAqc">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute CCF values for the available mutations.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagn/CNAqc/blob/master/R/compute_CCF.R'><code>R/compute_CCF.R</code></a></small>
    <div class="hidden name"><code>compute_CCF.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function provides the implementation of a set of entropy-related routines that can estimate
Cancer Cell Fraction values (CCFs) for the available mutations. The implemented routine is described
in the package vignette `"Computation of Cancer Cell Fractions"` that is available at the URL
<a href='https://caravagn.github.io/CNAqc/articles/ccf_computation.html'>https://caravagn.github.io/CNAqc/articles/ccf_computation.html</a>. This function creates a field
`CCF_estimates` inside the returned object which contains both the estimated CCF values and the
plot of the report of this analysis.</p>
    </div>

    <pre class="usage"><span class='fu'>compute_CCF</span>(
  <span class='no'>x</span>,
  <span class='kw'>karyotypes</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"1:0"</span>, <span class='st'>"1:1"</span>, <span class='st'>"2:0"</span>, <span class='st'>"2:1"</span>, <span class='st'>"2:2"</span>),
  <span class='kw'>entropy_quantile</span> <span class='kw'>=</span> <span class='fl'>0.9</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class <code>cnaqc</code>, created by the <code>init</code> function.</p></td>
    </tr>
    <tr>
      <th>karyotypes</th>
      <td><p>The karyotypes to use, this package supports only <code><a href='https://rdrr.io/r/base/c.html'>c('2:1', '2:0', '2:2')</a></code>.</p></td>
    </tr>
    <tr>
      <th>entropy_quantile</th>
      <td><p>The entropy quantile used to determine the interval in which the CCF estimates
are not reliable because it is not possible to determine a precise value for the multiplicity of a
mutation from its allelic frequencies. See the package vignette `"Computation of Cancer Cell Fractions"`
that is available at the URL <a href='https://caravagn.github.io/CNAqc/articles/ccf_computation.html'>https://caravagn.github.io/CNAqc/articles/ccf_computation.html</a> to
see the detailed meaning of this parameter.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cnaqc</code>, with CCF values available for extraction and plotting.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Getters function <code>CCF</code> and <code>plot_CCF</code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>'example_dataset_CNAqc'</span>)
<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'><a href='init.html'>init</a></span>(<span class='no'>example_dataset_CNAqc</span>$<span class='no'>snvs</span>, <span class='no'>example_dataset_CNAqc</span>$<span class='no'>cna</span>, <span class='no'>example_dataset_CNAqc</span>$<span class='no'>purity</span>)</div><div class='output co'>#&gt; <span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 12963 mutations to clonal segments (99% of input)</span></div><div class='input'>
<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'>compute_CCF</span>(<span class='no'>x</span>, <span class='kw'>karyotypes</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>'1:0'</span>, <span class='st'>'1:1'</span>, <span class='st'>'2:1'</span>, <span class='st'>'2:0'</span>, <span class='st'>'2:2'</span>))</div><div class='output co'>#&gt; <span class='message'>── Computing mutation multiplicity for karyotype </span><span style='color: #00BB00;'>2:1</span><span> ~ q = 0.9 ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Expected Binomial peak(s) for these calls (1 and 2 copies)</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     
#&gt; </span><span style='color: #555555;'>1</span><span>                     1 2:1       0.308 One copy  
#&gt; </span><span style='color: #555555;'>2</span><span>                     2 2:1       0.616 Two copies</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mixing pre/ post aneuploidy: 0.55 and 0.45</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> H(x)-derived cutoffs: [0.441441441441441; 0.468468468468468]</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='message'>── Computing mutation multiplicity for karyotype </span><span style='color: #00BB00;'>2:0</span><span> ~ q = 0.9 ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Expected Binomial peak(s) for these calls (1 and 2 copies)</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     
#&gt; </span><span style='color: #555555;'>1</span><span>                     1 2:0       0.445 One copy  
#&gt; </span><span style='color: #555555;'>2</span><span>                     2 2:0       0.89  Two copies</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mixing pre/ post aneuploidy: 0.09 and 0.91</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> H(x)-derived cutoffs: [0.657894736842105; 0.710526315789474]</span></div><div class='output co'>#&gt; <span class='message'>── Computing mutation multiplicity for karyotype </span><span style='color: #00BB00;'>2:2</span><span> ~ q = 0.9 ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Expected Binomial peak(s) for these calls (1 and 2 copies)</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 2 x 4</span><span>
#&gt;   mutation_multiplicity karyotype  peak label     
#&gt;                   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     
#&gt; </span><span style='color: #555555;'>1</span><span>                     1 2:2       0.235 One copy  
#&gt; </span><span style='color: #555555;'>2</span><span>                     2 2:2       0.471 Two copies</div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Mixing pre/ post aneuploidy: 0.09 and 0.91</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> H(x)-derived cutoffs: [0.319148936170213; 0.347517730496454]</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Summary CCF assignments. NA: not assignable with q = </span><span style='color: #00BB00;font-weight: bold;'>0.9</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 8 x 3</span><span>
#&gt;   karyotype mutation_multiplicity assignments
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                     </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span> 2:0                           1           9
#&gt; </span><span style='color: #555555;'>2</span><span> 2:0                           2          72
#&gt; </span><span style='color: #555555;'>3</span><span> 2:1                           1         886
#&gt; </span><span style='color: #555555;'>4</span><span> 2:1                           2         675
#&gt; </span><span style='color: #555555;'>5</span><span> 2:1                          </span><span style='color: #BB0000;'>NA</span><span>           2
#&gt; </span><span style='color: #555555;'>6</span><span> 2:2                           1        </span><span style='text-decoration: underline;'>1</span><span>013
#&gt; </span><span style='color: #555555;'>7</span><span> 2:2                           2        </span><span style='text-decoration: underline;'>6</span><span>427
#&gt; </span><span style='color: #555555;'>8</span><span> 2:2                          </span><span style='color: #BB0000;'>NA</span><span>          38
#&gt; Note: NA ~ mutations not confidently assignable with q = 0.9 </div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 13141 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Genome reference: </span><span style='color: #00BB00;'>GRCh38</span><span>. ──────────────────────────────────────────────────────────────────────────</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7478] ■■■■■■■■■■■■■■■■■■■■■■■■■■■
#&gt;    4:2  [n = 1893] ■■■■■■■
#&gt;    3:2  [n = 1625] ■■■■■■
#&gt;    2:1  [n = 1563] ■■■■■■
#&gt;    3:0  [n =  312] ■
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Cancer Cell Fraction (CCF) data available for karyotypes: 2:1, 2:0 and 2:2.</span></div><div class='input'>
<span class='fu'><a href='CCF.html'>CCF</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 9,122 x 15</span><span>
#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #555555;'> 1</span><span> chr2  3.58</span><span style='color: #555555;'>e</span><span>5 3.58</span><span style='color: #555555;'>e</span><span>5 C     A     PASS     104    58 0.558  intergenic      
#&gt; </span><span style='color: #555555;'> 2</span><span> chr2  9.09</span><span style='color: #555555;'>e</span><span>5 9.09</span><span style='color: #555555;'>e</span><span>5 A     G     PASS      15     7 0.467  ncRNA_intronic  
#&gt; </span><span style='color: #555555;'> 3</span><span> chr2  1.04</span><span style='color: #555555;'>e</span><span>6 1.04</span><span style='color: #555555;'>e</span><span>6 C     T     PASS      93    57 0.613  intronic        
#&gt; </span><span style='color: #555555;'> 4</span><span> chr2  1.33</span><span style='color: #555555;'>e</span><span>6 1.33</span><span style='color: #555555;'>e</span><span>6 A     T     PASS     104    55 0.529  intronic        
#&gt; </span><span style='color: #555555;'> 5</span><span> chr2  1.52</span><span style='color: #555555;'>e</span><span>6 1.52</span><span style='color: #555555;'>e</span><span>6 C     T     PASS      90    50 0.556  intronic        
#&gt; </span><span style='color: #555555;'> 6</span><span> chr2  2.20</span><span style='color: #555555;'>e</span><span>6 2.20</span><span style='color: #555555;'>e</span><span>6 G     T     PASS     125    42 0.336  intronic        
#&gt; </span><span style='color: #555555;'> 7</span><span> chr2  2.90</span><span style='color: #555555;'>e</span><span>6 2.90</span><span style='color: #555555;'>e</span><span>6 C     T     PASS     109    58 0.532  downstream      
#&gt; </span><span style='color: #555555;'> 8</span><span> chr2  3.13</span><span style='color: #555555;'>e</span><span>6 3.13</span><span style='color: #555555;'>e</span><span>6 A     G     PASS     134    35 0.261  ncRNA_intronic  
#&gt; </span><span style='color: #555555;'> 9</span><span> chr2  3.83</span><span style='color: #555555;'>e</span><span>6 3.83</span><span style='color: #555555;'>e</span><span>6 A     T     PASS     120    68 0.567  intergenic      
#&gt; </span><span style='color: #555555;'>10</span><span> chr2  3.88</span><span style='color: #555555;'>e</span><span>6 3.88</span><span style='color: #555555;'>e</span><span>6 T     A     PASS     126    10 0.079</span><span style='text-decoration: underline;'>4</span><span> intergenic      
#&gt; </span><span style='color: #555555;'># … with 9,112 more rows, and 5 more variables: GENE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, karyotype </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   segment_id </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, mutation_multiplicity </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, CCF </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span></div><div class='input'><span class='fu'><a href='plot_CCF.html'>plot_CCF</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 2 rows containing non-finite values (stat_bin).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 2 rows containing missing values (geom_bar).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 6 rows containing missing values (geom_bar).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 2 rows containing missing values (geom_bar).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 4 rows containing missing values (geom_bar).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 38 rows containing non-finite values (stat_bin).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 2 rows containing missing values (geom_bar).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 6 rows containing missing values (geom_bar).</span></div><div class='img'><img src='compute_CCF-1.png' alt='' width='700' height='433' /></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#see-also">See also</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://sites.google.com/site/giuliocaravagna/'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


