<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create a CNAqc object. — init • CNAqc</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Create a CNAqc object. — init" />
<meta property="og:description" content="Creates a CNAqc object from a set of
mutation and CNA calls, and a tumour purity value.
The resulting object contains the mutations mapped
into CNA segments, and allows for the computation
of the QC metrics available in the CNAqc package.
CNAqc supports `hg19` coordinates." />
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a CNAqc object.</h1>
    <small class="dont-index">Source: <a href='https://github.com/caravagnalab/CNAqcR/init.R'><code>R/init.R</code></a></small>
    <div class="hidden name"><code>init.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Creates a CNAqc object from a set of
mutation and CNA calls, and a tumour purity value.
The resulting object contains the mutations mapped
into CNA segments, and allows for the computation
of the QC metrics available in the CNAqc package.
CNAqc supports `hg19` coordinates.</p>
    </div>

    <pre class="usage"><span class='fu'>init</span><span class='op'>(</span><span class='va'>snvs</span>, <span class='va'>cna</span>, <span class='va'>purity</span>, ref <span class='op'>=</span> <span class='st'>"GRCh38"</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>snvs</th>
      <td><p>A data.frame of mutations with the following fields
available: `chr`, `from`, `to` as `hg19` chromosome coordinates.
`ref` and `alt` for the reference and alternative alleles, and
`DP`, `NV` and `VAF` for the depth (total number of reads with both
reference and alternative), the number of variants with the alternative
allele and the allele frequency (VAF = NV/DP). Chromosome names must be
in the format `chr1`, `chr2`, etc.; alleles should be characters and
all other fields numeric.</p></td>
    </tr>
    <tr>
      <th>cna</th>
      <td><p>A data.frame of CNA with the following fields
available: `chr`, `from`, `to` as `hg19` chromosome coordinates.
`Major` and `minor` for the number of copies of the major allele,
and the minor (B-allele). A `CCF` column can be used as a real-value
in between 0 and 1 to represent Cancer Cell Fractions; if this is not
available `CCF = 1` is set and all calls will refer to clonal segments.
Otherwise, segments with `CCF&lt;1` would be considered subclonal CNAs.</p></td>
    </tr>
    <tr>
      <th>purity</th>
      <td><p>Value in between `0` and `1` to represent the proportion
of actual tumour content (sometimes called "cellularity").</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>The reference genome (either "hg19" or "GRCh38"); the default is "GRCh38".</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A CNAqc object of class `cnaqc`, with S3 methods for printing,
plotting and analyzing data.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='st'>'example_dataset_CNAqc'</span>, package <span class='op'>=</span> <span class='st'>'CNAqc'</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>example_dataset_CNAqc</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $snvs
#&gt; <span style='color: #555555;'># A tibble: 12,963 x 13</span><span>
#&gt;    chr     from     to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>           
#&gt; </span><span style='color: #555555;'> 1</span><span> chr1  1.03</span><span style='color: #555555;'>e</span><span>6 1.03</span><span style='color: #555555;'>e</span><span>6 T     G     PASS      60     6 0.1    UTR5            
#&gt; </span><span style='color: #555555;'> 2</span><span> chr1  2.25</span><span style='color: #555555;'>e</span><span>6 2.25</span><span style='color: #555555;'>e</span><span>6 A     C     PASS     127     9 0.070</span><span style='text-decoration: underline;'>9</span><span> intergenic      
#&gt; </span><span style='color: #555555;'> 3</span><span> chr1  2.46</span><span style='color: #555555;'>e</span><span>6 2.46</span><span style='color: #555555;'>e</span><span>6 G     A     PASS     156    65 0.417  upstream        
#&gt; </span><span style='color: #555555;'> 4</span><span> chr1  2.73</span><span style='color: #555555;'>e</span><span>6 2.73</span><span style='color: #555555;'>e</span><span>6 T     C     PASS     180    90 0.5    downstream      
#&gt; </span><span style='color: #555555;'> 5</span><span> chr1  2.76</span><span style='color: #555555;'>e</span><span>6 2.76</span><span style='color: #555555;'>e</span><span>6 C     T     PASS     183    61 0.333  intergenic      
#&gt; </span><span style='color: #555555;'> 6</span><span> chr1  2.77</span><span style='color: #555555;'>e</span><span>6 2.77</span><span style='color: #555555;'>e</span><span>6 C     T     PASS     203   130 0.640  intergenic      
#&gt; </span><span style='color: #555555;'> 7</span><span> chr1  2.94</span><span style='color: #555555;'>e</span><span>6 2.94</span><span style='color: #555555;'>e</span><span>6 C     T     PASS     228   132 0.579  intergenic      
#&gt; </span><span style='color: #555555;'> 8</span><span> chr1  2.98</span><span style='color: #555555;'>e</span><span>6 2.98</span><span style='color: #555555;'>e</span><span>6 C     T     PASS     196    85 0.434  ncRNA_exonic    
#&gt; </span><span style='color: #555555;'> 9</span><span> chr1  3.39</span><span style='color: #555555;'>e</span><span>6 3.39</span><span style='color: #555555;'>e</span><span>6 T     G     PASS     124     6 0.048</span><span style='text-decoration: underline;'>4</span><span> intronic        
#&gt; </span><span style='color: #555555;'>10</span><span> chr1  3.50</span><span style='color: #555555;'>e</span><span>6 3.50</span><span style='color: #555555;'>e</span><span>6 G     A     PASS      88    10 0.114  intronic        
#&gt; </span><span style='color: #555555;'># … with 12,953 more rows, and 3 more variables: GENE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, is_driver </span><span style='color: #555555;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #555555;'>,</span><span>
#&gt; </span><span style='color: #555555;'>#   driver_label </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; $cna
#&gt; </span><span style='color: #555555;'># A tibble: 267 x 7</span><span>
#&gt;    chr       from       to  length covRatio Major minor
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'> 1</span><span> chr1    </span><span style='text-decoration: underline;'>840</span><span>009  1</span><span style='text-decoration: underline;'>689</span><span>987  </span><span style='text-decoration: underline;'>849</span><span>979     1.19     3     2
#&gt; </span><span style='color: #555555;'> 2</span><span> chr1   1</span><span style='text-decoration: underline;'>689</span><span>988  1</span><span style='text-decoration: underline;'>815</span><span>015  </span><span style='text-decoration: underline;'>125</span><span>028     1.26     3     2
#&gt; </span><span style='color: #555555;'> 3</span><span> chr1   1</span><span style='text-decoration: underline;'>815</span><span>016  9</span><span style='text-decoration: underline;'>799</span><span>969 7</span><span style='text-decoration: underline;'>984</span><span>954     1.19     3     2
#&gt; </span><span style='color: #555555;'> 4</span><span> chr1  10</span><span style='text-decoration: underline;'>479</span><span>910 12</span><span style='text-decoration: underline;'>079</span><span>917 1</span><span style='text-decoration: underline;'>600</span><span>008     1.19     3     2
#&gt; </span><span style='color: #555555;'> 5</span><span> chr1  12</span><span style='text-decoration: underline;'>079</span><span>917 12</span><span style='text-decoration: underline;'>154</span><span>980   </span><span style='text-decoration: underline;'>75</span><span>064     1.24     3     2
#&gt; </span><span style='color: #555555;'> 6</span><span> chr1  12</span><span style='text-decoration: underline;'>154</span><span>981 12</span><span style='text-decoration: underline;'>839</span><span>977  </span><span style='text-decoration: underline;'>684</span><span>997     1.19     3     2
#&gt; </span><span style='color: #555555;'> 7</span><span> chr1  13</span><span style='text-decoration: underline;'>780</span><span>016 17</span><span style='text-decoration: underline;'>790</span><span>026 4</span><span style='text-decoration: underline;'>010</span><span>011     1.19     3     2
#&gt; </span><span style='color: #555555;'> 8</span><span> chr1  17</span><span style='text-decoration: underline;'>849</span><span>962 21</span><span style='text-decoration: underline;'>080</span><span>067 3</span><span style='text-decoration: underline;'>230</span><span>106     1.19     3     2
#&gt; </span><span style='color: #555555;'> 9</span><span> chr1  21</span><span style='text-decoration: underline;'>080</span><span>068 21</span><span style='text-decoration: underline;'>559</span><span>998  </span><span style='text-decoration: underline;'>479</span><span>931     1.26     3     2
#&gt; </span><span style='color: #555555;'>10</span><span> chr1  21</span><span style='text-decoration: underline;'>559</span><span>998 24</span><span style='text-decoration: underline;'>830</span><span>001 3</span><span style='text-decoration: underline;'>270</span><span>004     1.19     3     2
#&gt; </span><span style='color: #555555;'># … with 257 more rows</span><span>
#&gt; 
#&gt; $purity
#&gt; [1] 0.89
#&gt; 
#&gt; $reference
#&gt; [1] "hg19"
#&gt; </div><div class='input'>
<span class='va'>x</span> <span class='op'>=</span> <span class='fu'>init</span><span class='op'>(</span>snvs <span class='op'>=</span> <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>snvs</span>, cna <span class='op'>=</span> <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>cna</span>, purity <span class='op'>=</span> <span class='va'>example_dataset_CNAqc</span><span class='op'>$</span><span class='va'>purity</span><span class='op'>)</span>
</div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Using reference genome coordinates for: </span><span style='color: #00BB00;'>GRCh38</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Drivers are annotated, but 'gene' column is missing, using mutation location.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Missing CCF column from CNA calls, adding CCF = 1 assuming clonal CNA calls.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Input n = 12963 mutations for 267 CNA segments (267 clonal, 0 subclonal)</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> Mapped n = 12963 mutations to clonal segments (100% of input)</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>── </span><span style='color: #000000;background-color: #BBBB00;'>[ CNAqc ] </span><span> n = 12963 mutations in </span><span style='color: #00BB00;'>267</span><span> segments (267 clonal + 0 subclonal). Ge</span></div><div class='output co'>#&gt; 
#&gt;    2:2  [n = 7478, L = 1483 Mb] ■■■■■■■■■■■■■■■■■■■■■■■■■■■  { </span><span style='color: #BBBB00;'>CTCF</span><span> }
#&gt;    4:2  [n = 1893, L =  331 Mb] ■■■■■■■
#&gt;    3:2  [n = 1625, L =  357 Mb] ■■■■■■
#&gt;    2:1  [n = 1563, L =  420 Mb] ■■■■■■  { </span><span style='color: #BBBB00;'>TTN</span><span> }
#&gt;    3:0  [n =  312, L =  137 Mb] ■
#&gt;    2:0  [n =   81, L =   39 Mb]   { </span><span style='color: #BBBB00;'>TP53</span><span> }
#&gt;   16:2  [n =    4, L =    0 Mb] 
#&gt;   25:2  [n =    2, L =    1 Mb] 
#&gt;    3:1  [n =    2, L =    1 Mb] 
#&gt;  106:1  [n =    1, L =    0 Mb] 
#&gt;   26:2  [n =    1, L =    0 Mb] 
#&gt;   99:1  [n =    1, L =    0 Mb] 
#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Sample Purity: 89% ~ Ploidy: 4.</span></div><div class='output co'>#&gt; </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://www.caravagnalab.org'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


