<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create a CNAqc object. — init • CNAqc</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Create a CNAqc object. — init" />
<meta property="og:description" content="Creates a CNAqc object from a set of
mutation and CNA calls, and a tumour purity value.
The resulting object contains the mutations mapped
into CNA segments, and allows for the computation
of the QC metrics available in the CNAqc package.
CNAqc supports `hg19` coordinates." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/peaks_detection.html">QC analysis via peaks detection</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a CNAqc object.</h1>
    
    <div class="hidden name"><code>init.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Creates a CNAqc object from a set of
mutation and CNA calls, and a tumour purity value.
The resulting object contains the mutations mapped
into CNA segments, and allows for the computation
of the QC metrics available in the CNAqc package.
CNAqc supports `hg19` coordinates.</p>
    </div>

    <pre class="usage"><span class='fu'>init</span>(<span class='no'>snvs</span>, <span class='no'>cna</span>, <span class='no'>purity</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>snvs</th>
      <td><p>A data.frame of mutations with the following fields
available: `chr`, `from`, `to` as `hg19` chromosome coordinates.
`ref` and `alt` for the reference and alternative alleles, and
`DP`, `NV` and `VAF` for the depth (total number of reads with both
reference and alternative), the number of variants with the alternative
allele and the allele frequency (VAF = NV/DP). Chromosome names must be
in the format `chr1`, `chr2`, etc.; alleles should be characters and
all other fields numeric.</p></td>
    </tr>
    <tr>
      <th>cna</th>
      <td><p>A data.frame of CNA with the following fields
available: `chr`, `from`, `to` as `hg19` chromosome coordinates.
`Major` and `minor` for the number of copies of the major allele,
and the minor (B-allele). A `CCF` column can be used as a real-value
in between 0 and 1 to represent Cancer Cell Fractions; if this is not
available `CCF = 1` is set and all calls will refer to clonal segments.
Otherwise, segments with `CCF&lt;1` would be considered subclonal CNAs.</p></td>
    </tr>
    <tr>
      <th>purity</th>
      <td><p>Value in between `0` and `1` to represent the proportion
of actual tumour content (sometimes called "cellularity").</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A CNAqc object of class `cnaqc`, with S3 methods for printing,
plotting and analyzing data.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='st'>'example_dataset_CNAqc'</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>'CNAqc'</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/print'>print</a></span>(<span class='no'>example_dataset_CNAqc</span>)</div><div class='output co'>#&gt; $snvs
#&gt; <span style='color: #555555;'># A tibble: 13,141 x 51</span><span>
#&gt;    chr   from  ID    ref   alt   QUAL  FILTER INFO  id       DP    NV    VAF
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'> 1</span><span> chr1  1027… </span><span style='color: #BB0000;'>NA</span><span>    T     G     1.73… PASS   DP=8… chr1…    60     6 0.1   
#&gt; </span><span style='color: #555555;'> 2</span><span> chr1  2248… </span><span style='color: #BB0000;'>NA</span><span>    A     C     -0    PASS   DP=1… chr1…   127     9 0.070</span><span style='text-decoration: underline;'>9</span><span>
#&gt; </span><span style='color: #555555;'> 3</span><span> chr1  2461… </span><span style='color: #BB0000;'>NA</span><span>    G     A     225   PASS   DP=1… chr1…   156    65 0.417 
#&gt; </span><span style='color: #555555;'> 4</span><span> chr1  2727… </span><span style='color: #BB0000;'>NA</span><span>    T     C     225   PASS   DP=1… chr1…   180    90 0.5   
#&gt; </span><span style='color: #555555;'> 5</span><span> chr1  2763… </span><span style='color: #BB0000;'>NA</span><span>    C     T     225   PASS   DP=1… chr1…   183    61 0.333 
#&gt; </span><span style='color: #555555;'> 6</span><span> chr1  2768… rs18… C     T     225   PASS   DP=2… chr1…   203   130 0.640 
#&gt; </span><span style='color: #555555;'> 7</span><span> chr1  2935… </span><span style='color: #BB0000;'>NA</span><span>    C     T     225   PASS   DP=2… chr1…   228   132 0.579 
#&gt; </span><span style='color: #555555;'> 8</span><span> chr1  2980… </span><span style='color: #BB0000;'>NA</span><span>    C     T     225   PASS   DP=2… chr1…   196    85 0.434 
#&gt; </span><span style='color: #555555;'> 9</span><span> chr1  3387… </span><span style='color: #BB0000;'>NA</span><span>    T     G     -0    PASS   DP=1… chr1…   124     6 0.048</span><span style='text-decoration: underline;'>4</span><span>
#&gt; </span><span style='color: #555555;'>10</span><span> chr1  3502… rs11… G     A     0.00… PASS   DP=9… chr1…    88    10 0.114 
#&gt; </span><span style='color: #555555;'># … with 13,131 more rows, and 39 more variables:
#&gt; #   `/icgc/dkfzlsdf/project/hipo/hipo_043/sequencing/whole_genome_sequencing/view-by-pid/H043-W99H5K/tumor/paired/merged-alignment/.merging_0/tumor_H043-W99H5K_merged.mdup.bam` </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   SEQUENCE_CONTEXT </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   `INFO_control(VAF=variant_allele_fraction;TSR=total_variant_supporting_reads_incl_lowqual)` </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   ANNOTATION_control </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DBSNP </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, `1K_GENOMES` </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   ANNOVAR_FUNCTION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, GENE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, EXONIC_CLASSIFICATION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   ANNOVAR_TRANSCRIPTS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, SEGDUP </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, CYTOBAND </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   REPEAT_MASKER </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DAC_BLACKLIST </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, DUKE_EXCLUDED </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   HISEQDEPTH </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, SELFCHAIN </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, MAPABILITY </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   SIMPLE_TANDEMREPEATS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, CONFIDENCE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, RECLASSIFICATION </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   PENALTIES </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, seqBiasPresent_1 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, seqingBiasPresent_1 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   seqBiasPresent_2 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, seqingBiasPresent_2 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, Enhancers </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   CpGislands </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, TFBScons </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, ENCODE_DNASE </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, miRNAs_snoRNAs </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   miRBase18 </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, COSMIC </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, miRNAtargets </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, CgiMountains </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   phastConsElem20bp </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, ENCODE_TFBS </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, blacklist </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, to </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; 
#&gt; $cna
#&gt; </span><span style='color: #555555;'># A tibble: 482 x 19</span><span>
#&gt;    chr     from     to length covRatio   TCN SV.Type c1Mean c2Mean  dhEst dhSNPs
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;fct&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'> 1</span><span> chr1  8.40</span><span style='color: #555555;'>e</span><span>5 1.69</span><span style='color: #555555;'>e</span><span>6 8.50</span><span style='color: #555555;'>e</span><span>5     1.19  5.05 </span><span style='color: #BB0000;'>NA</span><span>        2.02   3.03  0.199  0.192
#&gt; </span><span style='color: #555555;'> 2</span><span> chr1  1.69</span><span style='color: #555555;'>e</span><span>6 1.82</span><span style='color: #555555;'>e</span><span>6 1.25</span><span style='color: #555555;'>e</span><span>5     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>        2.19   3.19  0.185  0.182
#&gt; </span><span style='color: #555555;'> 3</span><span> chr1  1.82</span><span style='color: #555555;'>e</span><span>6 9.80</span><span style='color: #555555;'>e</span><span>6 7.98</span><span style='color: #555555;'>e</span><span>6     1.19  5.05 </span><span style='color: #BB0000;'>NA</span><span>        2.02   3.03  0.199  0.190
#&gt; </span><span style='color: #555555;'> 4</span><span> chr1  9.80</span><span style='color: #555555;'>e</span><span>6 9.87</span><span style='color: #555555;'>e</span><span>6 7.50</span><span style='color: #555555;'>e</span><span>4     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>       </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #555555;'> 5</span><span> chr1  9.87</span><span style='color: #555555;'>e</span><span>6 1.05</span><span style='color: #555555;'>e</span><span>7 6.05</span><span style='color: #555555;'>e</span><span>5     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>       </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>      0.228
#&gt; </span><span style='color: #555555;'> 6</span><span> chr1  1.05</span><span style='color: #555555;'>e</span><span>7 1.21</span><span style='color: #555555;'>e</span><span>7 1.60</span><span style='color: #555555;'>e</span><span>6     1.19  5.05 </span><span style='color: #BB0000;'>NA</span><span>        2.02   3.03  0.199  0.181
#&gt; </span><span style='color: #555555;'> 7</span><span> chr1  1.21</span><span style='color: #555555;'>e</span><span>7 1.22</span><span style='color: #555555;'>e</span><span>7 7.51</span><span style='color: #555555;'>e</span><span>4     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>        2.04   3.25  0.229  0.207
#&gt; </span><span style='color: #555555;'> 8</span><span> chr1  1.22</span><span style='color: #555555;'>e</span><span>7 1.28</span><span style='color: #555555;'>e</span><span>7 6.85</span><span style='color: #555555;'>e</span><span>5     1.19  5.05 </span><span style='color: #BB0000;'>NA</span><span>        2.02   3.03  0.199  0.200
#&gt; </span><span style='color: #555555;'> 9</span><span> chr1  1.38</span><span style='color: #555555;'>e</span><span>7 1.78</span><span style='color: #555555;'>e</span><span>7 4.01</span><span style='color: #555555;'>e</span><span>6     1.19  5.05 </span><span style='color: #BB0000;'>NA</span><span>        2.02   3.03  0.199  0.193
#&gt; </span><span style='color: #555555;'>10</span><span> chr1  1.78</span><span style='color: #555555;'>e</span><span>7 1.78</span><span style='color: #555555;'>e</span><span>7 5.99</span><span style='color: #555555;'>e</span><span>4     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>       </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>      0.158
#&gt; </span><span style='color: #555555;'># … with 472 more rows, and 8 more variables: Major </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>, minor </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span style='color: #555555;'>,
#&gt; #   CNA.type </span><span style='color: #555555;font-style: italic;'>&lt;fct&gt;</span><span style='color: #555555;'>, NbrOfHetsSNPs </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, minStart </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, maxStart </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>,
#&gt; #   minStop </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, maxStop </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; 
#&gt; $purity
#&gt; [1] 0.89
#&gt; </div><div class='input'>
<span class='no'>x</span> <span class='kw'>=</span> <span class='fu'>init</span>(<span class='no'>example_dataset_CNAqc</span>$<span class='no'>snvs</span>, <span class='no'>example_dataset_CNAqc</span>$<span class='no'>cna</span>,<span class='no'>example_dataset_CNAqc</span>$<span class='no'>purity</span>)</div><div class='output co'>#&gt; </span><span style='color: #BBBBBB;background-color: #0000BB;'> [ CNAqc - CNA Quality Check ] 
#&gt; </span><span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] Enforcing numeric for values: 1027104, 2248588, 2461999, 2727935, 2763397, 2768208, ...</span></div><div class='output co'>#&gt; <span class='message'>[CNAqc] Missing CCF column from CNA calls, adding CCF = 1 assuming all calls to be clonal.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] Enforcing numeric for values: 3, 3, 3, NA, NA, 3, ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; <span class='warning'>Warning: [CNAqc] Enforcing numeric for values: 2, 2, 2, NA, NA, 2, ...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: NAs introduced by coercion</span></div><div class='output co'>#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Input </span><span> n = 13141 mutations for 482 CNA segments (482 clonal, 0 subclonal)
#&gt; 
#&gt;  </span><span style='color: #00BB00;'>Mapping mutations to clonal CNA.
#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; Mapping.</span><span> n = 13050 mutations mapped to segments (~99% of input) </span><span style='color: #BB0000;'>
#&gt; [CNAqc] The following segments have Major/ minor alleles in non-numeric format or NAs
#&gt; </span><span style='color: #555555;'># A tibble: 215 x 20</span><span>
#&gt;    chr     from     to length covRatio   TCN SV.Type c1Mean c2Mean dhEst dhSNPs
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;fct&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'> 1</span><span> chr1  9.80</span><span style='color: #555555;'>e</span><span>6 9.87</span><span style='color: #555555;'>e</span><span>6  </span><span style='text-decoration: underline;'>75</span><span>009     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span> </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #555555;'> 2</span><span> chr1  9.87</span><span style='color: #555555;'>e</span><span>6 1.05</span><span style='color: #555555;'>e</span><span>7 </span><span style='text-decoration: underline;'>604</span><span>933     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.228
#&gt; </span><span style='color: #555555;'> 3</span><span> chr1  1.78</span><span style='color: #555555;'>e</span><span>7 1.78</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>59</span><span>936     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.158
#&gt; </span><span style='color: #555555;'> 4</span><span> chr1  3.09</span><span style='color: #555555;'>e</span><span>7 3.09</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>14</span><span>969     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.229
#&gt; </span><span style='color: #555555;'> 5</span><span> chr1  3.14</span><span style='color: #555555;'>e</span><span>7 3.16</span><span style='color: #555555;'>e</span><span>7 </span><span style='text-decoration: underline;'>179</span><span>994     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.102
#&gt; </span><span style='color: #555555;'> 6</span><span> chr1  3.45</span><span style='color: #555555;'>e</span><span>7 3.45</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>20</span><span>049     1.26  5.38 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span> </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #555555;'> 7</span><span> chr1  5.04</span><span style='color: #555555;'>e</span><span>7 5.04</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>44</span><span>976     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.216
#&gt; </span><span style='color: #555555;'> 8</span><span> chr1  5.19</span><span style='color: #555555;'>e</span><span>7 5.19</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>24</span><span>947     1.27  5.43 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span> </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #555555;'> 9</span><span> chr1  6.13</span><span style='color: #555555;'>e</span><span>7 6.13</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>39</span><span>974     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.233
#&gt; </span><span style='color: #555555;'>10</span><span> chr1  6.84</span><span style='color: #555555;'>e</span><span>7 6.84</span><span style='color: #555555;'>e</span><span>7  </span><span style='text-decoration: underline;'>20</span><span>002     1.24  5.29 </span><span style='color: #BB0000;'>NA</span><span>          </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>    </span><span style='color: #BB0000;'>NA</span><span>  0.235
#&gt; </span><span style='color: #555555;'># … with 205 more rows, and 9 more variables: Major </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>, minor </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #555555;'>,
#&gt; #   CNA.type </span><span style='color: #555555;font-style: italic;'>&lt;fct&gt;</span><span style='color: #555555;'>, NbrOfHetsSNPs </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, minStart </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, maxStart </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>,
#&gt; #   minStop </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, maxStop </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span style='color: #555555;'>, CCF </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #BB0000;'>These CNA segments will be removed.
#&gt; </span><span></div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/print'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; </span><span style='color: #000000;background-color: #BBBB00;'>      CNAqc </span><span> n = 13141 mutations for 267 CNA segments (267 clonal, 0 subclonal) </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;      Purity </span><span> 89 cellularity </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;  Karyotypes </span><span> 7478 (2:2); 1893 (4:2); 1625 (3:2); 1563 (2:1); 312 (3:0); 81 (2:0); 67 (NA:NA); 16 (NA:2); 4 (104:NA); 4 (16:2); 2 (25:2); 2 (3:1); 1 (106:1); 1 (26:2); 1 (99:1) 
#&gt;  </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt;    Peaks QC </span><span> </span><span style='color: #BB0000;'>FALSE</span><span> </span><span style='color: #000000;background-color: #BBBB00;'>
#&gt; </span><span></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

