<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. QC analysis via peaks detection • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. QC analysis via peaks detection">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a3_peaks_detection_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. QC analysis via peaks detection</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a3_peaks_detection.Rmd"><code>vignettes/a3_peaks_detection.Rmd</code></a></small>
      <div class="hidden name"><code>a3_peaks_detection.Rmd</code></div>

    </div>

    
    
<p><code>CNAqc</code> can determine if purity and CNA segments “fit well” to mutation data, an hortogonal quality check (QC) metric that can be used to adjust purity/ ploidy.</p>
<p><code>CNAqc</code> matches allelic imbalance of CNA segments to the allelic frequencies of somatic mutations. The package implements a karyotype-weighted linear score which uses the distance between data peaks (empirical), and their theoretical expectation. This score accouts for normal plodiy (germline), and tumour purity/ ploidy. The peaks are determined via kernel density estimation and peak-detection heuristics.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagn.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></code></pre></div>
<p>We work with the template dataset.</p>
<pre><code>#&gt;  [ CNAqc - CNA Quality Check ]
#&gt; $reference_genome
#&gt; [1] "hg19"
#&gt; 
#&gt; $snvs
#&gt; # A tibble: 12,963 x 17
#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION
#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           
#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            
#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      
#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        
#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      
#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      
#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      
#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      
#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    
#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        
#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        
#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,
#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,
#&gt; #   segment_id &lt;chr&gt;
#&gt; 
#&gt; $cna
#&gt; # A tibble: 267 x 10
#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n
#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           &lt;int&gt;
#&gt;  1 chr1   8.40e5  1.69e6 8.50e5     1.19     3     2     1 chr1:840009:16…     1
#&gt;  2 chr1   1.69e6  1.82e6 1.25e5     1.26     3     2     1 chr1:1689988:1…    NA
#&gt;  3 chr1   1.82e6  9.80e6 7.98e6     1.19     3     2     1 chr1:1815016:9…    28
#&gt;  4 chr1   1.05e7  1.21e7 1.60e6     1.19     3     2     1 chr1:10479910:…     1
#&gt;  5 chr1   1.21e7  1.22e7 7.51e4     1.24     3     2     1 chr1:12079917:…    NA
#&gt;  6 chr1   1.22e7  1.28e7 6.85e5     1.19     3     2     1 chr1:12154981:…    NA
#&gt;  7 chr1   1.38e7  1.78e7 4.01e6     1.19     3     2     1 chr1:13780016:…     7
#&gt;  8 chr1   1.78e7  2.11e7 3.23e6     1.19     3     2     1 chr1:17849962:…    12
#&gt;  9 chr1   2.11e7  2.16e7 4.80e5     1.26     3     2     1 chr1:21080068:…    NA
#&gt; 10 chr1   2.16e7  2.48e7 3.27e6     1.19     3     2     1 chr1:21559998:…    11
#&gt; # … with 257 more rows
#&gt; 
#&gt; $n_snvs
#&gt; [1] 12963
#&gt; 
#&gt; $n_cna
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_clonal
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_sbclonal
#&gt; [1] 0
#&gt; 
#&gt; $n_karyotype
#&gt; 
#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 
#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 
#&gt; 
#&gt; $purity
#&gt; [1] 0.89
#&gt; 
#&gt; $l_karyotype
#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 
#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 
#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 
#&gt;     517926     225714     174002     172894     162913     130712      80103 
#&gt;       99:1 
#&gt;      75782 
#&gt; 
#&gt; $ploidy
#&gt; [1] 4
#&gt; 
#&gt; $most_prevalent_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; $basepairs_by_karyotype
#&gt; # A tibble: 15 x 4
#&gt;    minor Major          n karyotype
#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    
#&gt;  1     2     2 1482910013 2:2      
#&gt;  2     1     2  420405440 2:1      
#&gt;  3     2     3  357172683 3:2      
#&gt;  4     2     4  331457138 4:2      
#&gt;  5     0     3  137498772 3:0      
#&gt;  6     0     2   39372558 2:0      
#&gt;  7     1     3     527401 3:1      
#&gt;  8     2    25     517926 25:2     
#&gt;  9     1   106     225714 106:1    
#&gt; 10     0     5     174002 5:0      
#&gt; 11     2    69     172894 69:2     
#&gt; 12     2    26     162913 26:2     
#&gt; 13     2    16     130712 16:2     
#&gt; 14     2    70      80103 70:2     
#&gt; 15     1    99      75782 99:1     
#&gt; 
#&gt; $most_mutations_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; attr(,"class")
#&gt; [1] "cnaqc"</code></pre>
<p>Peak detection routines are implemented in function <code>analyze_peaks</code>.</p>
<p>The idea is as follows: a segment with <span class="math inline">\(m\)</span> and <span class="math inline">\(M\)</span> copies of the minor and major alleles is consistent to mutation data if the VAF distribution peaks at certain values, which we can compute. Profiling then of the VAF distribution with a peak-detection heuristic allows <code>CNAqc</code> to measure the if peaks match. Large deviance suggests usually wrong purity estimation.</p>
<div id="peak-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#peak-analysis" class="anchor"></a>Peak analysis</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Run default analysis</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/analyze_peaks.html">analyze_peaks</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ Requested karyotypes 1:0, 1:1, 2:1, 2:0, and 2:2. Matching strategy closest.</span>
<span class="co">#&gt; ℹ Found n = 9122 mutations in 2:2, 2:1, 2:0 (skipping those with n &lt; 0 mutations).</span>
<span class="co">#&gt;  [ BMix fit ]</span>
<span class="co">#&gt; ℹ Binomials k_B = 1, 2, 3, and 4, Beta-Binomials k_BB = 0.</span>
<span class="co">#&gt; ✔ Fits to run, n = 8.</span>
<span class="co">#&gt; ℹ Bmix best fit completed in 0.05 mins</span>
<span class="co">#&gt; ── [ BMix ] My BMix model n = 1563 with k = 3 component(s) (3 + 0) ─────────────</span>
<span class="co">#&gt; ● Clusters: π = 53% [Bin 3], 43% [Bin 1], and 4% [Bin 2], with π &gt; 0.</span>
<span class="co">#&gt; ● Binomial Bin 1 with mean = 0.601114062229919.</span>
<span class="co">#&gt; ● Binomial Bin 2 with mean = 0.0957596171130631.</span>
<span class="co">#&gt; ● Binomial Bin 3 with mean = 0.305251093630488.</span>
<span class="co">#&gt; ℹ Score(s): ICL = 12253.31.</span>
<span class="co">#&gt;  [ BMix fit ]</span>
<span class="co">#&gt; ℹ Binomials k_B = 1, 2, 3, and 4, Beta-Binomials k_BB = 0.</span>
<span class="co">#&gt; ✔ Fits to run, n = 8.</span>
<span class="co">#&gt; ℹ Bmix best fit completed in 0 mins</span>
<span class="co">#&gt; ── [ BMix ] My BMix model n = 81 with k = 3 component(s) (3 + 0) ───────────────</span>
<span class="co">#&gt; ● Clusters: π = 89% [Bin 2], 9% [Bin 3], and 2% [Bin 1], with π &gt; 0.</span>
<span class="co">#&gt; ● Binomial Bin 1 with mean = 0.100778278373059.</span>
<span class="co">#&gt; ● Binomial Bin 2 with mean = 0.885928705870122.</span>
<span class="co">#&gt; ● Binomial Bin 3 with mean = 0.422641352917061.</span>
<span class="co">#&gt; ℹ Score(s): ICL = 506.43.</span>
<span class="co">#&gt;  [ BMix fit ]</span>
<span class="co">#&gt; ℹ Binomials k_B = 1, 2, 3, and 4, Beta-Binomials k_BB = 0.</span>
<span class="co">#&gt; ✔ Fits to run, n = 8.</span>
<span class="co">#&gt; ℹ Bmix best fit completed in 0.31 mins</span>
<span class="co">#&gt; ── [ BMix ] My BMix model n = 7478 with k = 4 component(s) (4 + 0) ─────────────</span>
<span class="co">#&gt; ● Clusters: π = 53% [Bin 4], 34% [Bin 2], 9% [Bin 1], and 5% [Bin 3], with π</span>
<span class="co">#&gt;   &gt; 0.</span>
<span class="co">#&gt; ● Binomial Bin 1 with mean = 0.227490006584454.</span>
<span class="co">#&gt; ● Binomial Bin 2 with mean = 0.493349008034858.</span>
<span class="co">#&gt; ● Binomial Bin 3 with mean = 0.0810077215512083.</span>
<span class="co">#&gt; ● Binomial Bin 4 with mean = 0.441642803552547.</span>
<span class="co">#&gt; ℹ Score(s): ICL = 59650.77.</span>
<span class="co">#&gt; # A tibble: 6 x 12</span>
<span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span>
<span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt; 1                1 2:1       0.308 0.306 3.60          831.   FALSE      0.00245</span>
<span class="co">#&gt; 2                2 2:1       0.616 0.600 2.94          677.   FALSE      0.0156 </span>
<span class="co">#&gt; 3                1 2:0       0.445 0.423 0.172           7.00 FALSE      0.0215 </span>
<span class="co">#&gt; 4                2 2:0       0.89  0.885 7.99           72.0  FALSE      0.00476</span>
<span class="co">#&gt; 5                1 2:2       0.235 0.24  0.86           65    FALSE     -0.00455</span>
<span class="co">#&gt; 6                2 2:2       0.471 0.46  8.14          625    FALSE      0.0109 </span>
<span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; ✔ Peak detection PASS with r = 0.00426886204162835 and tolerance e = 0.06</span></code></pre></div>
<p><code>CNAqc</code> analyze LOH regions (A, AA), diploid regions (AB), and amplification regions (AAB, AABB). These correspond to <code>'1:1'</code>, <code>'2:1'</code>, <code>'2:0'</code> and <code>'2:2'</code> in “Major:minor” notation. Karyotypes are subset by their size (normalized for the number of input mutations), and by default karyotypes smaller than 5% of the actual mutational burden are not considered</p>
<p>A Kernel-density method (unitary adjustment, Gaussian kernel) is used to smooth the data distribution, and the <a href="https://cran.rstudio.com/web/packages/peakPick/index.html">peakPick: Peak Picking Methods Inspired by Biological Data</a> package is used to determine peaks in the smoothed density. Peaks below a mnimum density (noise and miscalled segments) are filtered, and the others are matched to a predefined location with some tolerance (e.g., plus/minus <code>0.5 * epsilon</code>, where <code>epsilon = 0.015</code> is the default).</p>
<p>Theis analysis produces a set of scores that account for karyotype size and peaks offset, and that can be combined into a linear model to compute an overall QC metrics, called <code>rho</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; $reference_genome</span>
<span class="co">#&gt; [1] "hg19"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $snvs</span>
<span class="co">#&gt; # A tibble: 12,963 x 17</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      </span>
<span class="co">#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      </span>
<span class="co">#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      </span>
<span class="co">#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      </span>
<span class="co">#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    </span>
<span class="co">#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,</span>
<span class="co">#&gt; #   segment_id &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cna</span>
<span class="co">#&gt; # A tibble: 267 x 10</span>
<span class="co">#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   8.40e5  1.69e6 8.50e5     1.19     3     2     1 chr1:840009:16…     1</span>
<span class="co">#&gt;  2 chr1   1.69e6  1.82e6 1.25e5     1.26     3     2     1 chr1:1689988:1…    NA</span>
<span class="co">#&gt;  3 chr1   1.82e6  9.80e6 7.98e6     1.19     3     2     1 chr1:1815016:9…    28</span>
<span class="co">#&gt;  4 chr1   1.05e7  1.21e7 1.60e6     1.19     3     2     1 chr1:10479910:…     1</span>
<span class="co">#&gt;  5 chr1   1.21e7  1.22e7 7.51e4     1.24     3     2     1 chr1:12079917:…    NA</span>
<span class="co">#&gt;  6 chr1   1.22e7  1.28e7 6.85e5     1.19     3     2     1 chr1:12154981:…    NA</span>
<span class="co">#&gt;  7 chr1   1.38e7  1.78e7 4.01e6     1.19     3     2     1 chr1:13780016:…     7</span>
<span class="co">#&gt;  8 chr1   1.78e7  2.11e7 3.23e6     1.19     3     2     1 chr1:17849962:…    12</span>
<span class="co">#&gt;  9 chr1   2.11e7  2.16e7 4.80e5     1.26     3     2     1 chr1:21080068:…    NA</span>
<span class="co">#&gt; 10 chr1   2.16e7  2.48e7 3.27e6     1.19     3     2     1 chr1:21559998:…    11</span>
<span class="co">#&gt; # … with 257 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_snvs</span>
<span class="co">#&gt; [1] 12963</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_clonal</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_sbclonal</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_karyotype</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 </span>
<span class="co">#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $purity</span>
<span class="co">#&gt; [1] 0.89</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $l_karyotype</span>
<span class="co">#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 </span>
<span class="co">#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 </span>
<span class="co">#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 </span>
<span class="co">#&gt;     517926     225714     174002     172894     162913     130712      80103 </span>
<span class="co">#&gt;       99:1 </span>
<span class="co">#&gt;      75782 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ploidy</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_prevalent_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $basepairs_by_karyotype</span>
<span class="co">#&gt; # A tibble: 15 x 4</span>
<span class="co">#&gt;    minor Major          n karyotype</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1     2     2 1482910013 2:2      </span>
<span class="co">#&gt;  2     1     2  420405440 2:1      </span>
<span class="co">#&gt;  3     2     3  357172683 3:2      </span>
<span class="co">#&gt;  4     2     4  331457138 4:2      </span>
<span class="co">#&gt;  5     0     3  137498772 3:0      </span>
<span class="co">#&gt;  6     0     2   39372558 2:0      </span>
<span class="co">#&gt;  7     1     3     527401 3:1      </span>
<span class="co">#&gt;  8     2    25     517926 25:2     </span>
<span class="co">#&gt;  9     1   106     225714 106:1    </span>
<span class="co">#&gt; 10     0     5     174002 5:0      </span>
<span class="co">#&gt; 11     2    69     172894 69:2     </span>
<span class="co">#&gt; 12     2    26     162913 26:2     </span>
<span class="co">#&gt; 13     2    16     130712 16:2     </span>
<span class="co">#&gt; 14     2    70      80103 70:2     </span>
<span class="co">#&gt; 15     1    99      75782 99:1     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_mutations_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis</span>
<span class="co">#&gt; $peaks_analysis$score</span>
<span class="co">#&gt; [1] 0.004268862</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits</span>
<span class="co">#&gt; $peaks_analysis$fits$`1:0`</span>
<span class="co">#&gt; $peaks_analysis$fits$`1:0`$matching</span>
<span class="co">#&gt; NULL</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`1:1`</span>
<span class="co">#&gt; $peaks_analysis$fits$`1:1`$matching</span>
<span class="co">#&gt; NULL</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:1`</span>
<span class="co">#&gt; $peaks_analysis$fits$`2:1`$matching</span>
<span class="co">#&gt; # A tibble: 2 x 10</span>
<span class="co">#&gt;   mutation_multipl… karyotype  peak     x     y counts_per_bin discarded  offset</span>
<span class="co">#&gt;               &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt; 1                 1 2:1       0.308 0.306  3.60           831. FALSE     0.00245</span>
<span class="co">#&gt; 2                 2 2:1       0.616 0.600  2.94           677. FALSE     0.0156 </span>
<span class="co">#&gt; # … with 2 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:1`$density</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt;  density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Data: y (1563 obs.); Bandwidth 'bw' = 0.0341</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        x                  y          </span>
<span class="co">#&gt;  Min.   :-0.05578   Min.   :0.00016  </span>
<span class="co">#&gt;  1st Qu.: 0.17384   1st Qu.:0.21293  </span>
<span class="co">#&gt;  Median : 0.40346   Median :0.51690  </span>
<span class="co">#&gt;  Mean   : 0.40346   Mean   :1.08768  </span>
<span class="co">#&gt;  3rd Qu.: 0.63308   3rd Qu.:1.99965  </span>
<span class="co">#&gt;  Max.   : 0.86270   Max.   :3.60365  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:1`$xy_peaks</span>
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;        x     y counts_per_bin discarded</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;    </span>
<span class="co">#&gt; 1 0.08   0.27             5   FALSE    </span>
<span class="co">#&gt; 2 0.3    3.6             78   FALSE    </span>
<span class="co">#&gt; 3 0.6    2.94            60   FALSE    </span>
<span class="co">#&gt; 4 0.600  2.94           677.  FALSE    </span>
<span class="co">#&gt; 5 0.0952 0.259           54.9 FALSE    </span>
<span class="co">#&gt; 6 0.306  3.60           831.  FALSE    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:0`</span>
<span class="co">#&gt; $peaks_analysis$fits$`2:0`$matching</span>
<span class="co">#&gt; # A tibble: 2 x 10</span>
<span class="co">#&gt;   mutation_multipl… karyotype  peak     x     y counts_per_bin discarded  offset</span>
<span class="co">#&gt;               &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt; 1                 1 2:0       0.445 0.423 0.172           7.00 FALSE     0.0215 </span>
<span class="co">#&gt; 2                 2 2:0       0.89  0.885 7.99           72.0  FALSE     0.00476</span>
<span class="co">#&gt; # … with 2 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:0`$density</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt;  density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Data: y (81 obs.);   Bandwidth 'bw' = 0.0165</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        x                y           </span>
<span class="co">#&gt;  Min.   :0.0305   Min.   :0.000000  </span>
<span class="co">#&gt;  1st Qu.:0.2815   1st Qu.:0.000732  </span>
<span class="co">#&gt;  Median :0.5325   Median :0.193553  </span>
<span class="co">#&gt;  Mean   :0.5325   Mean   :0.994921  </span>
<span class="co">#&gt;  3rd Qu.:0.7836   3rd Qu.:0.556356  </span>
<span class="co">#&gt;  Max.   :1.0346   Max.   :9.001287  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:0`$xy_peaks</span>
<span class="co">#&gt; # A tibble: 8 x 4</span>
<span class="co">#&gt;       x     y counts_per_bin discarded</span>
<span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;    </span>
<span class="co">#&gt; 1 0.13  0.3             1    FALSE    </span>
<span class="co">#&gt; 2 0.32  0.3             0    FALSE    </span>
<span class="co">#&gt; 3 0.38  0.56            1    FALSE    </span>
<span class="co">#&gt; 4 0.47  0.79            1    FALSE    </span>
<span class="co">#&gt; 5 0.87  9               5    FALSE    </span>
<span class="co">#&gt; 6 0.101 0.199           2.00 FALSE    </span>
<span class="co">#&gt; 7 0.885 7.99           72.0  FALSE    </span>
<span class="co">#&gt; 8 0.423 0.172           7.00 FALSE    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:2`</span>
<span class="co">#&gt; $peaks_analysis$fits$`2:2`$matching</span>
<span class="co">#&gt; # A tibble: 2 x 10</span>
<span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span>
<span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt; 1                1 2:2       0.235  0.24  0.86             65 FALSE     -0.00455</span>
<span class="co">#&gt; 2                2 2:2       0.471  0.46  8.14            625 FALSE      0.0109 </span>
<span class="co">#&gt; # … with 2 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:2`$density</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt;  density.default(x = y, adjust = kernel_adjust, kernel = "gaussian")</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Data: y (7478 obs.); Bandwidth 'bw' = 0.008033</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        x                 y           </span>
<span class="co">#&gt;  Min.   :0.02101   Min.   :0.000244  </span>
<span class="co">#&gt;  1st Qu.:0.27179   1st Qu.:0.007045  </span>
<span class="co">#&gt;  Median :0.52256   Median :0.203133  </span>
<span class="co">#&gt;  Mean   :0.52256   Mean   :0.995946  </span>
<span class="co">#&gt;  3rd Qu.:0.77333   3rd Qu.:0.787272  </span>
<span class="co">#&gt;  Max.   :1.02410   Max.   :8.138867  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$fits$`2:2`$xy_peaks</span>
<span class="co">#&gt; # A tibble: 17 x 4</span>
<span class="co">#&gt;         x     y counts_per_bin discarded</span>
<span class="co">#&gt;     &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;    </span>
<span class="co">#&gt;  1 0.11   0.27             20  FALSE    </span>
<span class="co">#&gt;  2 0.22   0.84             69  FALSE    </span>
<span class="co">#&gt;  3 0.24   0.86             65  FALSE    </span>
<span class="co">#&gt;  4 0.46   8.14            625  FALSE    </span>
<span class="co">#&gt;  5 0.67   0.01              0  TRUE     </span>
<span class="co">#&gt;  6 0.72   0.02              2  FALSE    </span>
<span class="co">#&gt;  7 0.75   0.01              2  TRUE     </span>
<span class="co">#&gt;  8 0.79   0.01              1  TRUE     </span>
<span class="co">#&gt;  9 0.82   0.01              1  TRUE     </span>
<span class="co">#&gt; 10 0.85   0.01              1  TRUE     </span>
<span class="co">#&gt; 11 0.91   0.01              1  TRUE     </span>
<span class="co">#&gt; 12 0.93   0.01              0  TRUE     </span>
<span class="co">#&gt; 13 0.97   0.02              0  FALSE    </span>
<span class="co">#&gt; 14 0.227  0.816           641. FALSE    </span>
<span class="co">#&gt; 15 0.494  5.89           2541. FALSE    </span>
<span class="co">#&gt; 16 0.0819 0.400           364. FALSE    </span>
<span class="co">#&gt; 17 0.441  6.82           3932. FALSE    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$matches</span>
<span class="co">#&gt; # A tibble: 6 x 12</span>
<span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span>
<span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt; 1                1 2:1       0.308 0.306 3.60          831.   FALSE      0.00245</span>
<span class="co">#&gt; 2                2 2:1       0.616 0.600 2.94          677.   FALSE      0.0156 </span>
<span class="co">#&gt; 3                1 2:0       0.445 0.423 0.172           7.00 FALSE      0.0215 </span>
<span class="co">#&gt; 4                2 2:0       0.89  0.885 7.99           72.0  FALSE      0.00476</span>
<span class="co">#&gt; 5                1 2:2       0.235 0.24  0.86           65    FALSE     -0.00455</span>
<span class="co">#&gt; 6                2 2:2       0.471 0.46  8.14          625    FALSE      0.0109 </span>
<span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$matching_strategy</span>
<span class="co">#&gt; [1] "closest"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$min_karyotype_size</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$p_binsize_peaks</span>
<span class="co">#&gt; [1] 0.005</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$matching_epsilon</span>
<span class="co">#&gt; [1] 0.03</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $peaks_analysis$QC</span>
<span class="co">#&gt; [1] "PASS"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "cnaqc"</span></code></pre></div>
<p>A summary table is stored inside the named list <code>peaks_analysis</code>.</p>
</div>
<div id="plotting-results" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-results" class="anchor"></a>Plotting results</h1>
<p>For every karyotype, a plot with the estimated KDE and the matched peaks is available (assembled <code>ggpubr</code> figure with); gray panels represent karyotypes that have not been used.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-5-1.png" width="1056"></p>
<p>Individual karyotypes plots are available inside the named lists <code>peaks_analysis$plots</code>, named afte each required karyotype.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Tetraploid segments</span>
<span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">plots</span><span class="op">$</span><span class="va">`2:2`</span>
<span class="co">#&gt; NULL</span></code></pre></div>
<p>You can integrate plots from QC analysis with data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_peaks_analysis.html">plot_peaks_analysis</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,
  rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.8</span>, <span class="fl">1</span><span class="op">)</span>,
  nrow <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-7-1.png" width="1056"></p>
</div>
<div id="summary-results-and-quality-check-qc" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-results-and-quality-check-qc" class="anchor"></a>Summary results and Quality Check (QC)</h1>
<p>The summary technical table is available as <code>peaks_analysis$matches</code> inside <code>x</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">matches</span>
<span class="co">#&gt; # A tibble: 6 x 12</span>
<span class="co">#&gt;   mutation_multip… karyotype  peak     x     y counts_per_bin discarded   offset</span>
<span class="co">#&gt;              &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt; 1                1 2:1       0.308 0.306 3.60          831.   FALSE      0.00245</span>
<span class="co">#&gt; 2                2 2:1       0.616 0.600 2.94          677.   FALSE      0.0156 </span>
<span class="co">#&gt; 3                1 2:0       0.445 0.423 0.172           7.00 FALSE      0.0215 </span>
<span class="co">#&gt; 4                2 2:0       0.89  0.885 7.99           72.0  FALSE      0.00476</span>
<span class="co">#&gt; 5                1 2:2       0.235 0.24  0.86           65    FALSE     -0.00455</span>
<span class="co">#&gt; 6                2 2:2       0.471 0.46  8.14          625    FALSE      0.0109 </span>
<span class="co">#&gt; # … with 4 more variables: matched &lt;lgl&gt;, weight &lt;dbl&gt;, score &lt;dbl&gt;, QC &lt;chr&gt;</span></code></pre></div>
<p>The table reports several information. The following might be of some use after the analysis:</p>
<ul>
<li>
<code>mutation_multiplicity</code> and <code>karyotype</code>, which are the number of copies of the mutation, and the segment’s karyotype. Each segment with <code>discarded = FALSE</code> has been analysed;</li>
<li>
<code>peak</code>, <code>x</code>, <code>y</code> report where the expected peak should be found, and where it has been found (<code>x</code> and <code>y</code>);</li>
<li>
<code>offset</code>, <code>weight</code> and <code>score</code> that report the factors of the final <code>score</code>;</li>
<li>
<code>QC</code> that reports if the peak is considered matched.</li>
</ul>
<p>The overall <code>QC</code> result - <code>"PASS"/"FAIL"</code> - is available in <code>x</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span><span class="op">$</span><span class="va">peaks_analysis</span><span class="op">$</span><span class="va">QC</span>
<span class="co">#&gt; [1] "PASS"</span></code></pre></div>
<p>You can visually summarise the result of the QC analysis, for both peaks and <a href="https://caravagn.github.io/CNAqc/articles/ccf_computation.html">CCF computation</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_qc.html">plot_qc</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a3_peaks_detection_files/figure-html/unnamed-chunk-10-1.png" width="288"></p>
<p>Here since CCF have not been computed, the only available QC tests are for peak-detection.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
