<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>6. Genome fragmentation analysis • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="6. Genome fragmentation analysis">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a6_fragmentation_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>6. Genome fragmentation analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a6_fragmentation.Rmd"><code>vignettes/a6_fragmentation.Rmd</code></a></small>
      <div class="hidden name"><code>a6_fragmentation.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagn.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span></code></pre></div>
<p>We work with the template dataset.</p>
<pre><code>#&gt;  [ CNAqc - CNA Quality Check ]
#&gt; $reference_genome
#&gt; [1] "hg19"
#&gt; 
#&gt; $snvs
#&gt; # A tibble: 12,963 x 17
#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION
#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           
#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            
#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      
#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        
#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      
#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      
#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      
#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      
#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    
#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        
#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        
#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,
#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,
#&gt; #   segment_id &lt;chr&gt;
#&gt; 
#&gt; $cna
#&gt; # A tibble: 267 x 10
#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n
#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           &lt;int&gt;
#&gt;  1 chr1   8.40e5  1.69e6 8.50e5     1.19     3     2     1 chr1:840009:16…     1
#&gt;  2 chr1   1.69e6  1.82e6 1.25e5     1.26     3     2     1 chr1:1689988:1…    NA
#&gt;  3 chr1   1.82e6  9.80e6 7.98e6     1.19     3     2     1 chr1:1815016:9…    28
#&gt;  4 chr1   1.05e7  1.21e7 1.60e6     1.19     3     2     1 chr1:10479910:…     1
#&gt;  5 chr1   1.21e7  1.22e7 7.51e4     1.24     3     2     1 chr1:12079917:…    NA
#&gt;  6 chr1   1.22e7  1.28e7 6.85e5     1.19     3     2     1 chr1:12154981:…    NA
#&gt;  7 chr1   1.38e7  1.78e7 4.01e6     1.19     3     2     1 chr1:13780016:…     7
#&gt;  8 chr1   1.78e7  2.11e7 3.23e6     1.19     3     2     1 chr1:17849962:…    12
#&gt;  9 chr1   2.11e7  2.16e7 4.80e5     1.26     3     2     1 chr1:21080068:…    NA
#&gt; 10 chr1   2.16e7  2.48e7 3.27e6     1.19     3     2     1 chr1:21559998:…    11
#&gt; # … with 257 more rows
#&gt; 
#&gt; $n_snvs
#&gt; [1] 12963
#&gt; 
#&gt; $n_cna
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_clonal
#&gt; [1] 267
#&gt; 
#&gt; $n_cna_sbclonal
#&gt; [1] 0
#&gt; 
#&gt; $n_karyotype
#&gt; 
#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 
#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 
#&gt; 
#&gt; $purity
#&gt; [1] 0.89
#&gt; 
#&gt; $l_karyotype
#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 
#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 
#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 
#&gt;     517926     225714     174002     172894     162913     130712      80103 
#&gt;       99:1 
#&gt;      75782 
#&gt; 
#&gt; $ploidy
#&gt; [1] 4
#&gt; 
#&gt; $most_prevalent_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; $basepairs_by_karyotype
#&gt; # A tibble: 15 x 4
#&gt;    minor Major          n karyotype
#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    
#&gt;  1     2     2 1482910013 2:2      
#&gt;  2     1     2  420405440 2:1      
#&gt;  3     2     3  357172683 3:2      
#&gt;  4     2     4  331457138 4:2      
#&gt;  5     0     3  137498772 3:0      
#&gt;  6     0     2   39372558 2:0      
#&gt;  7     1     3     527401 3:1      
#&gt;  8     2    25     517926 25:2     
#&gt;  9     1   106     225714 106:1    
#&gt; 10     0     5     174002 5:0      
#&gt; 11     2    69     172894 69:2     
#&gt; 12     2    26     162913 26:2     
#&gt; 13     2    16     130712 16:2     
#&gt; 14     2    70      80103 70:2     
#&gt; 15     1    99      75782 99:1     
#&gt; 
#&gt; $most_mutations_karyotype
#&gt; [1] "2:2"
#&gt; 
#&gt; attr(,"class")
#&gt; [1] "cnaqc"</code></pre>
<div id="fragmentation-of-individual-arms" class="section level3">
<h3 class="hasAnchor">
<a href="#fragmentation-of-individual-arms" class="anchor"></a>Fragmentation of individual arms</h3>
<p>The fragmentation of a chromosome arm is assessed with a statistical test based on counting the size of the copy number segments mapping to the arm.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A histogram of segments' lenght</span>
<span class="fu"><a href="../reference/plot_segment_size_distribution.html">plot_segment_size_distribution</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
<p><code>CNAqc</code> counts, for every arm with lenght <span class="math inline">\(L\)</span> nucleotides:</p>
<ul>
<li>
<span class="math inline">\(n_s\)</span>, the number of mapped CNA segments shorter than <span class="math inline">\(\delta\%\)</span> of <span class="math inline">\(L\)</span>;</li>
<li>
<span class="math inline">\(n_l\)</span>, the number of mapped CNA segments longer than <span class="math inline">\(\delta\%\)</span> sof <span class="math inline">\(L\)</span>.</li>
</ul>
<p>A one-sided Binomial test is used to compute a p-value for the null hypothesis of seeing <span class="math inline">\(n_s\)</span> observations in <span class="math inline">\(n_l\)</span> trials, assuming a Binomial success probability <span class="math inline">\(p = \delta &gt; 0\)</span>. <span class="math inline">\(p\)</span> represents a model where each segment length is equally likely (uniform distribution).</p>
<p>In this way the test accounts for the difference in lenghts of the chromsome arms; a p-value per arm is reported and adjusted for multiple hyoptheses (Bonferroni).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Test with default parameters (small segments: &lt; 20% of chromosome arm)</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ One-tailed Binomial test: 8 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span>
<span class="co">#&gt; ✔ chr7p,  p = 1.71798691840001e-24 ~ 34 segments, 34 short.</span>
<span class="co">#&gt; ✔ chr1p,  p = 1.62738995200002e-15 ~ 24 segments, 23 short.</span>
<span class="co">#&gt; ✔ chr1q,  p = 4.34176000000001e-08 ~ 13 segments, 12 short.</span>
<span class="co">#&gt; ✔ chr11q,  p = 1.0657792e-06 ~ 13 segments, 11 short.</span>
<span class="co">#&gt; ✔ chr12q,  p = 2.00704e-07 ~ 12 segments, 11 short.</span>
<span class="co">#&gt; ✔ chr3q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr7q,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr8p,  p = 9.21599999999998e-07 ~ 11 segments, 10 short.</span>
<span class="co">#&gt; ℹ 8 significantly overfragmented chromosome arms (alpha level 0.01).</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; $reference_genome</span>
<span class="co">#&gt; [1] "hg19"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $snvs</span>
<span class="co">#&gt; # A tibble: 12,963 x 17</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      </span>
<span class="co">#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      </span>
<span class="co">#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      </span>
<span class="co">#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      </span>
<span class="co">#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    </span>
<span class="co">#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,</span>
<span class="co">#&gt; #   segment_id &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cna</span>
<span class="co">#&gt; # A tibble: 267 x 10</span>
<span class="co">#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   8.40e5  1.69e6 8.50e5     1.19     3     2     1 chr1:840009:16…     1</span>
<span class="co">#&gt;  2 chr1   1.69e6  1.82e6 1.25e5     1.26     3     2     1 chr1:1689988:1…    NA</span>
<span class="co">#&gt;  3 chr1   1.82e6  9.80e6 7.98e6     1.19     3     2     1 chr1:1815016:9…    28</span>
<span class="co">#&gt;  4 chr1   1.05e7  1.21e7 1.60e6     1.19     3     2     1 chr1:10479910:…     1</span>
<span class="co">#&gt;  5 chr1   1.21e7  1.22e7 7.51e4     1.24     3     2     1 chr1:12079917:…    NA</span>
<span class="co">#&gt;  6 chr1   1.22e7  1.28e7 6.85e5     1.19     3     2     1 chr1:12154981:…    NA</span>
<span class="co">#&gt;  7 chr1   1.38e7  1.78e7 4.01e6     1.19     3     2     1 chr1:13780016:…     7</span>
<span class="co">#&gt;  8 chr1   1.78e7  2.11e7 3.23e6     1.19     3     2     1 chr1:17849962:…    12</span>
<span class="co">#&gt;  9 chr1   2.11e7  2.16e7 4.80e5     1.26     3     2     1 chr1:21080068:…    NA</span>
<span class="co">#&gt; 10 chr1   2.16e7  2.48e7 3.27e6     1.19     3     2     1 chr1:21559998:…    11</span>
<span class="co">#&gt; # … with 257 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_snvs</span>
<span class="co">#&gt; [1] 12963</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_clonal</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_sbclonal</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_karyotype</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 </span>
<span class="co">#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $purity</span>
<span class="co">#&gt; [1] 0.89</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $l_karyotype</span>
<span class="co">#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 </span>
<span class="co">#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 </span>
<span class="co">#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 </span>
<span class="co">#&gt;     517926     225714     174002     172894     162913     130712      80103 </span>
<span class="co">#&gt;       99:1 </span>
<span class="co">#&gt;      75782 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ploidy</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_prevalent_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $basepairs_by_karyotype</span>
<span class="co">#&gt; # A tibble: 15 x 4</span>
<span class="co">#&gt;    minor Major          n karyotype</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1     2     2 1482910013 2:2      </span>
<span class="co">#&gt;  2     1     2  420405440 2:1      </span>
<span class="co">#&gt;  3     2     3  357172683 3:2      </span>
<span class="co">#&gt;  4     2     4  331457138 4:2      </span>
<span class="co">#&gt;  5     0     3  137498772 3:0      </span>
<span class="co">#&gt;  6     0     2   39372558 2:0      </span>
<span class="co">#&gt;  7     1     3     527401 3:1      </span>
<span class="co">#&gt;  8     2    25     517926 25:2     </span>
<span class="co">#&gt;  9     1   106     225714 106:1    </span>
<span class="co">#&gt; 10     0     5     174002 5:0      </span>
<span class="co">#&gt; 11     2    69     172894 69:2     </span>
<span class="co">#&gt; 12     2    26     162913 26:2     </span>
<span class="co">#&gt; 13     2    16     130712 16:2     </span>
<span class="co">#&gt; 14     2    70      80103 70:2     </span>
<span class="co">#&gt; 15     1    99      75782 99:1     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_mutations_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation</span>
<span class="co">#&gt; $arm_fragmentation$table</span>
<span class="co">#&gt; # A tibble: 41 x 10</span>
<span class="co">#&gt;    chr   arm   n_long n_short jumps total_segments testable  p_value</span>
<span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr7  p          0      34   206             34 TRUE     1.72e-24</span>
<span class="co">#&gt;  2 chr1  p          1      23     0             24 TRUE     1.63e-15</span>
<span class="co">#&gt;  3 chr1  q          1      12     5             13 TRUE     4.34e- 8</span>
<span class="co">#&gt;  4 chr12 q          1      11   414             12 TRUE     2.01e- 7</span>
<span class="co">#&gt;  5 chr8  p          1      10     0             11 TRUE     9.22e- 7</span>
<span class="co">#&gt;  6 chr11 q          2      11     0             13 TRUE     1.07e- 6</span>
<span class="co">#&gt;  7 chr3  q          2      10     2             12 TRUE     4.53e- 6</span>
<span class="co">#&gt;  8 chr7  q          2      10     3             12 TRUE     4.53e- 6</span>
<span class="co">#&gt;  9 chr11 p          1       8     0              9 FALSE    1.00e+ 0</span>
<span class="co">#&gt; 10 chr15 q          2       7     4              9 FALSE    1.00e+ 0</span>
<span class="co">#&gt; # … with 31 more rows, and 2 more variables: Bonferroni_cutoff &lt;dbl&gt;,</span>
<span class="co">#&gt; #   significant &lt;lgl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$alpha</span>
<span class="co">#&gt; [1] 0.01</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$N_tests</span>
<span class="co">#&gt; [1] 8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$genome_percentage_cutoff</span>
<span class="co">#&gt; [1] 0.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$minimum_segments_for_testing</span>
<span class="co">#&gt; [1] 10</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "cnaqc"</span></code></pre></div>
<p>You can produce a arm-level report for the fragmentation test, with:</p>
<ul>
<li>a scatter of the counts per arm, with scaled the p-values;</li>
<li>a jump statistics per arm, <span class="math inline">\(J\)</span>.</li>
</ul>
<p><span class="math inline">\(J\)</span> is the sum of the variation in total copy number profiles, evaluated among each pair of contiguous segments.</p>
<p>Significantly overfragmented arms with high <span class="math inline">\(J\)</span> have a “scattered” copy number profile. Those with low <span class="math inline">\(J\)</span> are more uniform, as they show little no copy number change, and can be possibly smoothed (see below).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; ✖ Increare zoom to see more chromosomes.</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-5-1.png" width="672"></p>
<p>Once available, these results appear in any call to <code>plot_segments</code> as annotated purple squares sorrounding the arms.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot has now segments</span>
<span class="fu"><a href="../reference/plot_segments.html">plot_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-6-1.png" width="912"></p>
<p><a href="https://caravagn.github.io/CNAqc/articles/smoothing.html">Smoothing</a> is a good way to start cleaning up the fragmented sets of arms, as it merges segments that have no karyotype change, and are closer than a certain distance.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Smooth with default parameters</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/smooth_segments.html">smooth_segments</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; Smoothing chr1 with 37 segments: [1-24] [27-30] [31-33] [34-37] </span>
<span class="co">#&gt; Smoothing chr10 with 8 segments: [1-2] [3-4] [5-8] </span>
<span class="co">#&gt; Smoothing chr11 with 22 segments: [1-7] [8-9] [10-22] </span>
<span class="co">#&gt; Smoothing chr12 with 13 segments: [2-3] [12-13] </span>
<span class="co">#&gt; Smoothing chr14 with 2 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chr15 with 9 segments: [1-5] [7-9] </span>
<span class="co">#&gt; Smoothing chr16 with 10 segments: [1-6] [8-10] </span>
<span class="co">#&gt; Smoothing chr17 with 10 segments: [4-5] [6-7] [8-10] </span>
<span class="co">#&gt; Smoothing chr18 with 8 segments: [1-2] [3-8] </span>
<span class="co">#&gt; Smoothing chr19 with 5 segments: [2-5] </span>
<span class="co">#&gt; Smoothing chr2 with 18 segments: [1-8] [10-12] [14-18] </span>
<span class="co">#&gt; Smoothing chr20 with 9 segments: [1-3] [4-9] </span>
<span class="co">#&gt; Smoothing chr21 with 2 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chr22 with 3 segments: </span>
<span class="co">#&gt; Smoothing chr3 with 19 segments: [1-7] [8-13] [15-19] </span>
<span class="co">#&gt; Smoothing chr4 with 8 segments: [1-4] [5-8] </span>
<span class="co">#&gt; Smoothing chr5 with 6 segments: [1-2] [3-5] </span>
<span class="co">#&gt; Smoothing chr6 with 4 segments: [1-2] [3-4] </span>
<span class="co">#&gt; Smoothing chr7 with 46 segments: [1-5] [9-27] [36-38] [39-41] [43-46] </span>
<span class="co">#&gt; Smoothing chr8 with 18 segments: [1-4] [5-11] [12-18] </span>
<span class="co">#&gt; Smoothing chr9 with 3 segments: [1-2] </span>
<span class="co">#&gt; Smoothing chrX with 6 segments: [1-2] [3-6]</span>
<span class="co">#&gt; ✔ Smoothed from 267 to 87 segments with 1e+06 gap (bases).</span>
<span class="co">#&gt; ℹ Creating a new CNAqc object. The old object will be retained in the $before_smoothing field.</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>
<span class="co">#&gt; ℹ Using reference genome coordinates for: hg19.</span>
<span class="co">#&gt; ℹ Input n = 12963 mutations for 87 CNA segments (87 clonal, 0 subclonal)</span>
<span class="co">#&gt; Warning in map_mutations_to_segments(snvs, cna %&gt;% filter(CCF == 1)): [CNAqc] a</span>
<span class="co">#&gt; karyotype column is present in CNA calls, and will be overwritten</span>
<span class="co">#&gt; ✔ Mapped n = 12963 mutations to clonal segments (100% of input)</span>

<span class="co"># Re-compute the fragmentation</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/detect_arm_overfragmentation.html">detect_arm_overfragmentation</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; ℹ One-tailed Binomial test: 2 tests, alpha 0.01. Short segments: 0.2% of the reference arm.</span>
<span class="co">#&gt; ✔ chr7p,  p = 4.52608e-06 ~ 12 segments, 10 short.</span>
<span class="co">#&gt; ✔ chr12q,  p = 4.19839999999999e-06 ~ 10 segments, 9 short.</span>
<span class="co">#&gt; ℹ 2 significantly overfragmented chromosome arms (alpha level 0.01).</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="co">#&gt; $reference_genome</span>
<span class="co">#&gt; [1] "hg19"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $snvs</span>
<span class="co">#&gt; # A tibble: 12,963 x 17</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      </span>
<span class="co">#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      </span>
<span class="co">#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      </span>
<span class="co">#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      </span>
<span class="co">#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    </span>
<span class="co">#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,</span>
<span class="co">#&gt; #   segment_id &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cna</span>
<span class="co">#&gt; # A tibble: 87 x 10</span>
<span class="co">#&gt;    chr      from      to  length covRatio Major minor   CCF segment_id         n</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;   &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;          &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   8.40e5  1.21e8  1.20e8    1.19      3     2     1 chr1:840009:1…   402</span>
<span class="co">#&gt;  2 chr1   1.45e8  1.46e8  8.20e5    1.04      2     2     1 chr1:14492008…    NA</span>
<span class="co">#&gt;  3 chr1   1.47e8  1.48e8  1.20e6    1.19      3     2     1 chr1:14663001…     2</span>
<span class="co">#&gt;  4 chr1   1.50e8  1.86e8  3.57e7    1.19      3     2     1 chr1:14981506…   124</span>
<span class="co">#&gt;  5 chr1   1.86e8  1.95e8  9.59e6    0.729     3     0     1 chr1:18555902…    20</span>
<span class="co">#&gt;  6 chr1   1.95e8  2.49e8  5.38e7    1.19      3     2     1 chr1:19537336…   169</span>
<span class="co">#&gt;  7 chr10  1.30e5  3.86e7  3.85e7    0.729     3     0     1 chr10:130013:…    95</span>
<span class="co">#&gt;  8 chr10  4.24e7  4.62e7  3.75e6    0.729     3     0     1 chr10:4241000…    12</span>
<span class="co">#&gt;  9 chr10  4.83e7  1.35e8  8.71e7    0.729     3     0     1 chr10:4831017…   185</span>
<span class="co">#&gt; 10 chr11  1.90e5  4.89e7  4.87e7    0.731     2     1     1 chr11:190010:…   183</span>
<span class="co">#&gt; # … with 77 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_snvs</span>
<span class="co">#&gt; [1] 12963</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna</span>
<span class="co">#&gt; [1] 87</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_clonal</span>
<span class="co">#&gt; [1] 87</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_sbclonal</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_karyotype</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 </span>
<span class="co">#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $purity</span>
<span class="co">#&gt; [1] 0.89</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $l_karyotype</span>
<span class="co">#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 </span>
<span class="co">#&gt; 1493370778  424084287  362333832  332861761  138936099   39404909     527400 </span>
<span class="co">#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 </span>
<span class="co">#&gt;     517923     225713     174001     172892     162912     130711      80102 </span>
<span class="co">#&gt;       99:1 </span>
<span class="co">#&gt;      75781 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ploidy</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_prevalent_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $basepairs_by_karyotype</span>
<span class="co">#&gt; # A tibble: 15 x 4</span>
<span class="co">#&gt;    minor Major          n karyotype</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1     2     2 1493370778 2:2      </span>
<span class="co">#&gt;  2     1     2  424084287 2:1      </span>
<span class="co">#&gt;  3     2     3  362333832 3:2      </span>
<span class="co">#&gt;  4     2     4  332861761 4:2      </span>
<span class="co">#&gt;  5     0     3  138936099 3:0      </span>
<span class="co">#&gt;  6     0     2   39404909 2:0      </span>
<span class="co">#&gt;  7     1     3     527400 3:1      </span>
<span class="co">#&gt;  8     2    25     517923 25:2     </span>
<span class="co">#&gt;  9     1   106     225713 106:1    </span>
<span class="co">#&gt; 10     0     5     174001 5:0      </span>
<span class="co">#&gt; 11     2    69     172892 69:2     </span>
<span class="co">#&gt; 12     2    26     162912 26:2     </span>
<span class="co">#&gt; 13     2    16     130711 16:2     </span>
<span class="co">#&gt; 14     2    70      80102 70:2     </span>
<span class="co">#&gt; 15     1    99      75781 99:1     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_mutations_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $before_smoothing</span>
<span class="co">#&gt; $reference_genome</span>
<span class="co">#&gt; [1] "hg19"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $snvs</span>
<span class="co">#&gt; # A tibble: 12,963 x 17</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      </span>
<span class="co">#&gt;  5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      </span>
<span class="co">#&gt;  6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      </span>
<span class="co">#&gt;  7 chr1  2935590 2935591 C     T     PASS     228   132 0.579  intergenic      </span>
<span class="co">#&gt;  8 chr1  2980032 2980033 C     T     PASS     196    85 0.434  ncRNA_exonic    </span>
<span class="co">#&gt;  9 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt; 10 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt; # … with 12,953 more rows, and 7 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, gene &lt;chr&gt;, type &lt;chr&gt;, karyotype &lt;chr&gt;,</span>
<span class="co">#&gt; #   segment_id &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $cna</span>
<span class="co">#&gt; # A tibble: 267 x 10</span>
<span class="co">#&gt;    chr      from      to length covRatio Major minor   CCF segment_id          n</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;   &lt;int&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           &lt;int&gt;</span>
<span class="co">#&gt;  1 chr1   8.40e5  1.69e6 8.50e5     1.19     3     2     1 chr1:840009:16…     1</span>
<span class="co">#&gt;  2 chr1   1.69e6  1.82e6 1.25e5     1.26     3     2     1 chr1:1689988:1…    NA</span>
<span class="co">#&gt;  3 chr1   1.82e6  9.80e6 7.98e6     1.19     3     2     1 chr1:1815016:9…    28</span>
<span class="co">#&gt;  4 chr1   1.05e7  1.21e7 1.60e6     1.19     3     2     1 chr1:10479910:…     1</span>
<span class="co">#&gt;  5 chr1   1.21e7  1.22e7 7.51e4     1.24     3     2     1 chr1:12079917:…    NA</span>
<span class="co">#&gt;  6 chr1   1.22e7  1.28e7 6.85e5     1.19     3     2     1 chr1:12154981:…    NA</span>
<span class="co">#&gt;  7 chr1   1.38e7  1.78e7 4.01e6     1.19     3     2     1 chr1:13780016:…     7</span>
<span class="co">#&gt;  8 chr1   1.78e7  2.11e7 3.23e6     1.19     3     2     1 chr1:17849962:…    12</span>
<span class="co">#&gt;  9 chr1   2.11e7  2.16e7 4.80e5     1.26     3     2     1 chr1:21080068:…    NA</span>
<span class="co">#&gt; 10 chr1   2.16e7  2.48e7 3.27e6     1.19     3     2     1 chr1:21559998:…    11</span>
<span class="co">#&gt; # … with 257 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_snvs</span>
<span class="co">#&gt; [1] 12963</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_clonal</span>
<span class="co">#&gt; [1] 267</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_cna_sbclonal</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $n_karyotype</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   2:2   4:2   3:2   2:1   3:0   2:0  16:2  25:2   3:1 106:1  26:2  99:1 </span>
<span class="co">#&gt;  7478  1893  1625  1563   312    81     4     2     2     1     1     1 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $purity</span>
<span class="co">#&gt; [1] 0.89</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $l_karyotype</span>
<span class="co">#&gt;        2:2        2:1        3:2        4:2        3:0        2:0        3:1 </span>
<span class="co">#&gt; 1482910013  420405440  357172683  331457138  137498772   39372558     527401 </span>
<span class="co">#&gt;       25:2      106:1        5:0       69:2       26:2       16:2       70:2 </span>
<span class="co">#&gt;     517926     225714     174002     172894     162913     130712      80103 </span>
<span class="co">#&gt;       99:1 </span>
<span class="co">#&gt;      75782 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ploidy</span>
<span class="co">#&gt; [1] 4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_prevalent_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $basepairs_by_karyotype</span>
<span class="co">#&gt; # A tibble: 15 x 4</span>
<span class="co">#&gt;    minor Major          n karyotype</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1     2     2 1482910013 2:2      </span>
<span class="co">#&gt;  2     1     2  420405440 2:1      </span>
<span class="co">#&gt;  3     2     3  357172683 3:2      </span>
<span class="co">#&gt;  4     2     4  331457138 4:2      </span>
<span class="co">#&gt;  5     0     3  137498772 3:0      </span>
<span class="co">#&gt;  6     0     2   39372558 2:0      </span>
<span class="co">#&gt;  7     1     3     527401 3:1      </span>
<span class="co">#&gt;  8     2    25     517926 25:2     </span>
<span class="co">#&gt;  9     1   106     225714 106:1    </span>
<span class="co">#&gt; 10     0     5     174002 5:0      </span>
<span class="co">#&gt; 11     2    69     172894 69:2     </span>
<span class="co">#&gt; 12     2    26     162913 26:2     </span>
<span class="co">#&gt; 13     2    16     130712 16:2     </span>
<span class="co">#&gt; 14     2    70      80103 70:2     </span>
<span class="co">#&gt; 15     1    99      75782 99:1     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $most_mutations_karyotype</span>
<span class="co">#&gt; [1] "2:2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation</span>
<span class="co">#&gt; $arm_fragmentation$table</span>
<span class="co">#&gt; # A tibble: 41 x 10</span>
<span class="co">#&gt;    chr   arm   n_long n_short jumps total_segments testable  p_value</span>
<span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr7  p          0      34   206             34 TRUE     1.72e-24</span>
<span class="co">#&gt;  2 chr1  p          1      23     0             24 TRUE     1.63e-15</span>
<span class="co">#&gt;  3 chr1  q          1      12     5             13 TRUE     4.34e- 8</span>
<span class="co">#&gt;  4 chr12 q          1      11   414             12 TRUE     2.01e- 7</span>
<span class="co">#&gt;  5 chr8  p          1      10     0             11 TRUE     9.22e- 7</span>
<span class="co">#&gt;  6 chr11 q          2      11     0             13 TRUE     1.07e- 6</span>
<span class="co">#&gt;  7 chr3  q          2      10     2             12 TRUE     4.53e- 6</span>
<span class="co">#&gt;  8 chr7  q          2      10     3             12 TRUE     4.53e- 6</span>
<span class="co">#&gt;  9 chr11 p          1       8     0              9 FALSE    1.00e+ 0</span>
<span class="co">#&gt; 10 chr15 q          2       7     4              9 FALSE    1.00e+ 0</span>
<span class="co">#&gt; # … with 31 more rows, and 2 more variables: Bonferroni_cutoff &lt;dbl&gt;,</span>
<span class="co">#&gt; #   significant &lt;lgl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$alpha</span>
<span class="co">#&gt; [1] 0.01</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$N_tests</span>
<span class="co">#&gt; [1] 8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$genome_percentage_cutoff</span>
<span class="co">#&gt; [1] 0.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$minimum_segments_for_testing</span>
<span class="co">#&gt; [1] 10</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "cnaqc"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation</span>
<span class="co">#&gt; $arm_fragmentation$table</span>
<span class="co">#&gt; # A tibble: 41 x 10</span>
<span class="co">#&gt;    chr   arm   n_long n_short jumps total_segments testable    p_value</span>
<span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt; &lt;lgl&gt;         &lt;dbl&gt;</span>
<span class="co">#&gt;  1 chr12 q          1       9   414             10 TRUE     0.00000420</span>
<span class="co">#&gt;  2 chr7  p          2      10   206             12 TRUE     0.00000453</span>
<span class="co">#&gt;  3 chr1  q          2       3     5              5 FALSE    1         </span>
<span class="co">#&gt;  4 chr17 p          3       2     8              5 FALSE    1         </span>
<span class="co">#&gt;  5 chr7  q          2       3     3              5 FALSE    1         </span>
<span class="co">#&gt;  6 chr15 q          1       2     4              3 FALSE    1         </span>
<span class="co">#&gt;  7 chr2  q          2       1     2              3 FALSE    1         </span>
<span class="co">#&gt;  8 chr22 q          2       1     2              3 FALSE    1         </span>
<span class="co">#&gt;  9 chr3  q          2       1     2              3 FALSE    1         </span>
<span class="co">#&gt; 10 chr10 q          1       1     0              2 FALSE    1         </span>
<span class="co">#&gt; # … with 31 more rows, and 2 more variables: Bonferroni_cutoff &lt;dbl&gt;,</span>
<span class="co">#&gt; #   significant &lt;lgl&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$alpha</span>
<span class="co">#&gt; [1] 0.01</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$N_tests</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$genome_percentage_cutoff</span>
<span class="co">#&gt; [1] 0.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $arm_fragmentation$minimum_segments_for_testing</span>
<span class="co">#&gt; [1] 10</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "cnaqc"</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_arm_fragmentation.html">plot_arm_fragmentation</a></span><span class="op">(</span><span class="va">x</span>, zoom <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="a6_fragmentation_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
