<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>8. Annotatation of SNVs • CNAqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="8. Annotatation of SNVs">
<meta property="og:description" content="CNAqc">
<meta property="og:image" content="caravagn.github.io/CNAqc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNAqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CNAqc.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_Introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_Plots.html">2. Plotting data</a>
    </li>
    <li>
      <a href="../articles/a3_peaks_detection.html">3. QC analysis via peaks detection</a>
    </li>
    <li>
      <a href="../articles/a4_ccf_computation.html">4. Computation of Cancer Cell Fractions</a>
    </li>
    <li>
      <a href="../articles/a5_smoothing.html">5. Smoothing copy number segments</a>
    </li>
    <li>
      <a href="../articles/a6_fragmentation.html">6. Genome fragmentation analysis</a>
    </li>
    <li>
      <a href="../articles/a7_example_MSeq.html">7. Example data analysis</a>
    </li>
    <li>
      <a href="../articles/a8_annotate_variants.html">8. Annotatation of SNVs</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/CNAqc/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a8_annotate_variants_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>8. Annotatation of SNVs</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/CNAqcvignettes/a8_annotate_variants.Rmd"><code>vignettes/a8_annotate_variants.Rmd</code></a></small>
      <div class="hidden name"><code>a8_annotate_variants.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/CNAqc">CNAqc</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✔ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; ✔ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>

<span class="co"># Extra packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="co">#if you use RStudio you will probably have to run this line</span>
<span class="co"># options(connectionObserver = NULL)</span></code></pre></div>
<p>CNAqc also has a number of simple annotation functions that can be useful for giving a preliminary idea of what mutations might be driver events. However, it should be noted that this function provides an overview but does not replace more complex and in-depth analyzes of SNVs. Given a SNVs table we in turn annotate: * Position of the mutation (coding, intron, fiveUTR, threeUTR, intron, intergenic, spliceSite, promoter) * Consequence on the protein for coding mutations (nonsynonymous, synonymous, frameshift, stop) with eventual change in the AA * Synonymous mutations are then compared to known driver genes from the intogen database and flagged as drivers * Optionally if the dbSNP id is present the PolyPhen score can also be annotated</p>
<p>The annotation phase is heavily based on the <a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html">VariantAnnotation</a> package</p>
<div id="install-required-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#install-required-packages" class="anchor"></a>Install Required Packages</h1>
<p>We will show the functioning with tha example file shipped in CNAqc</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load template data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'example_dataset_CNAqc'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>

<span class="va">snvs</span> <span class="op">&lt;-</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">snvs</span>


<span class="va">snvs</span> <span class="op">%&gt;%</span>  <span class="va">head</span>
<span class="co">#&gt; # A tibble: 6 x 13</span>
<span class="co">#&gt;   chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt; 1 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt; 2 chr1  2248588 2248589 A     C     PASS     127     9 0.0709 intergenic      </span>
<span class="co">#&gt; 3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt; 4 chr1  2727935 2727936 T     C     PASS     180    90 0.5    downstream      </span>
<span class="co">#&gt; 5 chr1  2763397 2763398 C     T     PASS     183    61 0.333  intergenic      </span>
<span class="co">#&gt; 6 chr1  2768208 2768209 C     T     PASS     203   130 0.640  intergenic      </span>
<span class="co">#&gt; # … with 3 more variables: GENE &lt;chr&gt;, is_driver &lt;lgl&gt;, driver_label &lt;chr&gt;</span></code></pre></div>
<p>As we can see we already have driver annotated from an external source (ANNOVAR) here, let’s get rid of them and then compare what we will found wi our approach</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">drivers_true</span> <span class="op">&lt;-</span> <span class="va">snvs</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">is_driver</span> <span class="op">==</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">driver_label</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="va">unique</span>

<span class="va">snvs</span> <span class="op">&lt;-</span> <span class="va">snvs</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">is_driver</span>, <span class="op">-</span><span class="va">driver_label</span><span class="op">)</span></code></pre></div>
<p>The function needs some databases from Bioconductor to annotate correctly the variants</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">

<span class="co"># First we have to find the reference against which we mapped the reads</span>
<span class="va">reference_genome</span> <span class="op">&lt;-</span> <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">reference</span>

<span class="co"># All those packages are distributed in Bioconductor</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; The downloaded binary packages are in</span>
<span class="co">#&gt;  /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//Rtmpdr8WE5/downloaded_packages</span>


<span class="co">#We have to install the corresponding txdb package for transcript annotations</span>
<span class="va">tx_pkg</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"TxDb.Hsapiens.UCSC."</span>,<span class="va">reference_genome</span>, <span class="st">".knownGene"</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">tx_pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="va">tx_pkg</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; The downloaded binary packages are in</span>
<span class="co">#&gt;  /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//Rtmpdr8WE5/downloaded_packages</span>

<span class="co">#We have to install also the BS database for the sequences (it may take some time)</span>
<span class="va">bs_pkg</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC."</span>,<span class="va">reference_genome</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">bs_pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="va">bs_pkg</span><span class="op">)</span>

<span class="co"># Then those two packages provide usefull utilities to deal with biological databases</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">"Organism.dplyr"</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"Organism.dplyr"</span><span class="op">)</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></code></pre></div>
</div>
<div id="annotation-phase" class="section level1">
<h1 class="hasAnchor">
<a href="#annotation-phase" class="anchor"></a>Annotation phase</h1>
<p>Now we have everything we can just run the function andget back our annotated file</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">snvs_annotated</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_variants.html">annotate_variants</a></span><span class="op">(</span><span class="va">snvs</span>, ref <span class="op">=</span> <span class="va">reference_genome</span><span class="op">)</span>

<span class="va">snvs_annotated</span>
<span class="co">#&gt; # A tibble: 13,129 x 18</span>
<span class="co">#&gt;    chr       from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr16 67646006  6.76e7 C     T     PASS     120    54 0.45   exonic          </span>
<span class="co">#&gt;  2 chr17  7577106  7.58e6 G     C     PASS      84    78 0.929  exonic          </span>
<span class="co">#&gt;  3 chr19 50919900  5.09e7 C     T     PASS     145    61 0.421  exonic          </span>
<span class="co">#&gt;  4 chr1   1027104  1.03e6 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  5 chr1   2461999  2.46e6 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  6 chr1   3387161  3.39e6 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt;  7 chr1   3502517  3.50e6 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt;  8 chr1   4722897  4.72e6 G     A     PASS     186    75 0.403  intronic        </span>
<span class="co">#&gt;  9 chr1   6175539  6.18e6 C     T     PASS     155    54 0.348  intronic        </span>
<span class="co">#&gt; 10 chr1   6213216  6.21e6 G     A     PASS     184    98 0.533  intronic        </span>
<span class="co">#&gt; # … with 13,119 more rows, and 8 more variables: GENE &lt;chr&gt;, gene_symbol &lt;chr&gt;,</span>
<span class="co">#&gt; #   location &lt;chr&gt;, consequence &lt;chr&gt;, refAA &lt;chr&gt;, varAA &lt;chr&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, is_driver &lt;lgl&gt;</span></code></pre></div>
<p>One thing to note is the presence of multiple locations and consequences for a single variant. This happens as we try to annotate the mutations in a transcript-agnostic manner, consequently we report all possible effects and locations for any transcript (separated by a : ). The drivers largely correspond to those previously noted (TP53 and CTCF) with some differences. Obviously using multiple annotations and validation guarantees more precise results.</p>
<p>We use the intogen database to annotate drivers, but you can costumize this behaviour in two way. First, drivers are annotated by tumor types, by default we use all the drivers but it is possible to specify the cancer type id. A list of tumor types and their codes is available in the package</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'intogen_ttypes'</span>, package <span class="op">=</span> <span class="st">'CNAqc'</span><span class="op">)</span>

<span class="va">intogen_ttypes</span>
<span class="co">#&gt; # A tibble: 66 x 2</span>
<span class="co">#&gt;    cancer_name                           intogen_code</span>
<span class="co">#&gt;    &lt;chr&gt;                                 &lt;chr&gt;       </span>
<span class="co">#&gt;  1 Adrenocortical carcinoma              ACC         </span>
<span class="co">#&gt;  2 Adenoid cystic carcinoma              ACY         </span>
<span class="co">#&gt;  3 Acute lymphoblastic leukemia          ALL         </span>
<span class="co">#&gt;  4 Acute myeloid leukemia                AML         </span>
<span class="co">#&gt;  5 Anus cancer                           AN          </span>
<span class="co">#&gt;  6 Angiosarcoma                          ANGS        </span>
<span class="co">#&gt;  7 Atypical teratoid/rhabdoid tumor      ATRT        </span>
<span class="co">#&gt;  8 Bowel cancer                          BC          </span>
<span class="co">#&gt;  9 Small intestine cancer neuroendocrine BCNET       </span>
<span class="co">#&gt; 10 Bladder cancer                        BLCA        </span>
<span class="co">#&gt; # … with 56 more rows</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snvs_annotated_OV</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_variants.html">annotate_variants</a></span><span class="op">(</span><span class="va">snvs</span>, ref <span class="op">=</span> <span class="va">reference_genome</span>, filter_tumor_type <span class="op">=</span> <span class="st">"OV"</span><span class="op">)</span>

<span class="va">snvs_annotated_OV</span>
<span class="co">#&gt; # A tibble: 13,129 x 18</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr17 7577106 7577107 G     C     PASS      84    78 0.929  exonic          </span>
<span class="co">#&gt;  2 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt;  5 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt;  6 chr1  4722897 4722898 G     A     PASS     186    75 0.403  intronic        </span>
<span class="co">#&gt;  7 chr1  6175539 6175540 C     T     PASS     155    54 0.348  intronic        </span>
<span class="co">#&gt;  8 chr1  6213216 6213217 G     A     PASS     184    98 0.533  intronic        </span>
<span class="co">#&gt;  9 chr1  6421856 6421857 C     T     PASS     174    56 0.322  intronic        </span>
<span class="co">#&gt; 10 chr1  6751814 6751815 T     A     PASS     190    38 0.2    intronic        </span>
<span class="co">#&gt; # … with 13,119 more rows, and 8 more variables: GENE &lt;chr&gt;, gene_symbol &lt;chr&gt;,</span>
<span class="co">#&gt; #   location &lt;chr&gt;, consequence &lt;chr&gt;, refAA &lt;chr&gt;, varAA &lt;chr&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, is_driver &lt;lgl&gt;</span></code></pre></div>
<p>Another option is to supply your list of drivers as a one coloumn <code>data.frame</code> using the argument <code>driver_list</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">snvs_annotated_custom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_variants.html">annotate_variants</a></span><span class="op">(</span><span class="va">snvs</span>, ref <span class="op">=</span> <span class="va">reference_genome</span>, driver_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"CHDS"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">snvs_annotated_custom</span>
<span class="co">#&gt; # A tibble: 13,129 x 18</span>
<span class="co">#&gt;    chr      from      to ref   alt   FILTER    DP    NV    VAF ANNOVAR_FUNCTION</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;           </span>
<span class="co">#&gt;  1 chr17 7577106 7577107 G     C     PASS      84    78 0.929  exonic          </span>
<span class="co">#&gt;  2 chr1  1027104 1027105 T     G     PASS      60     6 0.1    UTR5            </span>
<span class="co">#&gt;  3 chr1  2461999 2462000 G     A     PASS     156    65 0.417  upstream        </span>
<span class="co">#&gt;  4 chr1  3387161 3387162 T     G     PASS     124     6 0.0484 intronic        </span>
<span class="co">#&gt;  5 chr1  3502517 3502518 G     A     PASS      88    10 0.114  intronic        </span>
<span class="co">#&gt;  6 chr1  4722897 4722898 G     A     PASS     186    75 0.403  intronic        </span>
<span class="co">#&gt;  7 chr1  6175539 6175540 C     T     PASS     155    54 0.348  intronic        </span>
<span class="co">#&gt;  8 chr1  6213216 6213217 G     A     PASS     184    98 0.533  intronic        </span>
<span class="co">#&gt;  9 chr1  6421856 6421857 C     T     PASS     174    56 0.322  intronic        </span>
<span class="co">#&gt; 10 chr1  6751814 6751815 T     A     PASS     190    38 0.2    intronic        </span>
<span class="co">#&gt; # … with 13,119 more rows, and 8 more variables: GENE &lt;chr&gt;, gene_symbol &lt;chr&gt;,</span>
<span class="co">#&gt; #   location &lt;chr&gt;, consequence &lt;chr&gt;, refAA &lt;chr&gt;, varAA &lt;chr&gt;,</span>
<span class="co">#&gt; #   driver_label &lt;chr&gt;, is_driver &lt;lgl&gt;</span></code></pre></div>
</div>
<div id="plot" class="section level1">
<h1 class="hasAnchor">
<a href="#plot" class="anchor"></a>Plot</h1>
<p>We can finally perform the QC and see how the drivers are actually plotted together with the data distribution</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>
  <span class="va">snvs_annotated</span>, 
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">cna</span>,
  <span class="va">example_dataset_CNAqc</span><span class="op">$</span><span class="va">purity</span>,
  ref <span class="op">=</span> <span class="st">'hg19'</span>
  <span class="op">)</span>
<span class="co">#&gt;  [ CNAqc - CNA Quality Check ]</span>

<span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html">ggarrange</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'VAF'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'DP'</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_data_histogram.html">plot_data_histogram</a></span><span class="op">(</span><span class="va">x</span>, which <span class="op">=</span> <span class="st">'NV'</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">3</span>,
  nrow <span class="op">=</span> <span class="fl">1</span>
  <span class="op">)</span></code></pre></div>
<p><img src="a8_annotate_variants_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.caravagnalab.org">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
